---
title: "Vignette Title"
author: "Jason T. Serviss"
date: "`r Sys.Date()`"
output:
  html_document:
    theme: flatly
    toc: yes
    code_folding: hide
vignette: >
  %\VignetteIndexEntry{Vignette Title}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
liftr:
  maintainer: "Jason Serviss"
  email: "jason.serviss@ki.se"
  from: "rocker/tidyverse:3.4.2"
  pandoc: false
  cran:
    - knitr/1.17
    - rmarkdown/1.6
    - printr/0.1
    - ggthemes/3.4.0
    - repmis/0.5
    - extrafont/0.17
    - scales/0.5.0
  remotes:
    - "GranderLab/miR34a_asRNA_project"
---

```{r, echo = FALSE, message = FALSE}
packages <- c(
    "tidyverse",
    "printr",
    "ggthemes",
    "readr",
    "miR34AasRNAproject"
)
purrr::walk(packages, library, character.only = TRUE)
rm(packages)
```

```{r}
projectUrl <- "https://github.com/GranderLab/miR34a_asRNA_project/raw/master/inst/"
dataUrl <- "https://github.com/GranderLab/miR34a_asRNA_project/raw/master/data"
```

##Figure 1 {.tabset}

### Figure 1a

####Introduction

An antisense transcript to the miR34a HG was detected using the UCSC genome browser.

<br></br>

####Methods

The USCS genome browser was utilized for the bioinformatic evaluation of antisense transcription utilizing the Human EST and ENCODE RNAseq tracks.

<br></br>

####Results

```{r, out.width = "850px"}
url <- "miR34aLocus/miR34aLocus_hg38.png"
knitr::include_graphics(file.path(projectUrl, url))
```

<br></br>

A schematic picture of the miR34a locus from UCSC genome browser (hg38) including miR34a HG and mature miR34a, and LINC01759. H3K4me3 ChIP-seq data and conservation over the locus is also shown.

<br></br>

####Conclusion

GENCODE annotation supports the presence of a transcript orininating in the antisense orientation from the miR34a gene. 

<br></br><br></br>

###Figure 1b

####Introduction

We wished to examine expression of miR34a asRNA in different cancer tissues and, therefore, utilized a panel of common cancer cell lines to detect miR34a asRNA expression, as well as, miR34a HG. We included both p53 wild type, p53 mutated, and p53 null cell lines in the cell line panel due to the fact that miR34a is a known downstream target of p53.

<br></br>

####Methods

Cells were cultured with the appropriate growth medium and seeded at a desity between 50-75% confluency. The following day, RNA was extracted using the RNeasy mini kit (Qiagen) and subsequently treated with DNase (Ambion Turbo DNA-free, Life Technologies). 500ng RNA was used for cDNA synthesis using MuMLV (Life Technologies) and a 1:1 mix of oligo(dT) and random nanomers. PCR was run for 35 or 38 or 16 cycles for miR34a asRNA, miR34a HG, and beta-actin, respectivley, with 58˚C annealing temperature. PCR products were analyzed on a 2% agarose gel.

<br></br>

#####Primers

```{r}
primers <- data.frame(
    name=c(
        "miR34a asRNA F1",
        "miR34a asRNA R1",
        "miR34aHG_F",
        "miR34aHG_R",
        "B-actin_F",
        "B-actin_R"
    ),
    sequence=c(
        "AGC GGC ATC TCC TCC ACC TGA AA",
        "TTG CCT CGT GAG TCC AAG GAG AAT",
        "TCT GCT CCA GTG GCT GAT GAG AAA",
        "GTT CAC TGG CCT CAA AGT TGG CAT",
        "AGG TCA TCA CCA TTG GCA ATG AG",
        "CTT TGC GGA TGT CCA CGT CA"
    )
)
primers
```

<br></br>

####Results

```{r, out.width = "850px"}
url <- "miR34aASExpressionCellPanel/cellPanelLarge.png"
knitr::include_graphics(file.path(projectUrl, url))
```

<br></br>

Semi-quantitative PCR data from the screening of a panel of cell lines. 
* Indicates that, although the cell line is p53wt, other mechanisms are present which inhibit p53 function. 

<br></br>

####Conclusions

We were able to detect co-expression of both miR34a HG and miR34a AS in a variety of cell lines, including HCT116 human colon cancer cells and 293T human embryonic kidney cells.

<br></br><br></br>

###Figure 1c

####Introduction

<br></br>

####Methods

<br></br>

####Results

See Supplementary Figure 1a for R-squared and p-values corersponding with the analysis.

Insert code here from Erik and write Intro Methods etc

```{r, out.width = "850px"}
url <- "TCGAcorrelation/miR34a_RP3-510D11.2_diploid_only_edited.png"
knitr::include_graphics(file.path(projectUrl, url))
```

<br></br>

####Conclusions

<br></br><br></br>


###Supplementary Figure 1a

<br></br><br></br>

```{r, out.width = "850px"}
url <- "TCGAcorrelation/table.png"
knitr::include_graphics(file.path(projectUrl, url))
```

Add a description.

<br></br><br></br>

###Figure 1d

####Introduction

In order to define the 3' transcription termination site for the miR34a asRNA, 3'-RACE was performed.

<br></br>

####Methods

All cell lines were cultured at 5% CO2 and 37° C with U2OS cells grown in McCoy’s 5a (Life Technologies). All growth mediums were supplemented with 10% heat-inactivated FBS and 50 μg/ml of streptomycin and 50 μg/ml of penicillin. U2OS cell RNA was polyA-tailed using yeast polyA polymerase after which cDNA was synthesized using oligo(dT) primers. Nested-PCR was performed first using a forward primer in miR34a asRNA exon 1 (miR34a_asRNA_F10) and a tailed oligo(dT) primer (polyT_T7FAM) followed by a second PCR using an alternate miR34a asRNA exon 1 primer (miR34a_asRNA_F1) and a reverse primer binding to the tail of the previously used oligo(dT) primer (FAMprimer). PCR products were gel purified and cloned the Strata Clone Kit (Agilent Technologies), and sequenced.

<br></br>

#####Primers

```{r}
primers <- data.frame(
  name = c("miR34a_asRNA_F10", "polyT_T7FAM", "miR34a_asRNA_F1", "FAMprimer"),
  sequence = c(
    "ACG CGT CTC TCC AGC CCG GGA T",
    "CAG TGA ATT GTA ATA CGA CTC ACT ATA GGG ACA TCC GTA GCT CGT CCA GGA CCC TTT TTT TTT TTT TTT TTT VN",
    "AGC GGC ATC TCC TCC ACC TGA AA",
    "CCG TAG CTC GTC CAG GAC CC"
  )
)
primers
```

<br></br>

####Results

```{r, out.width = "850px"}
url <- "3primeRACE/results_hg38.png"
knitr::include_graphics(file.path(projectUrl, url))
```

<br></br>

3’-RACE sequencing results displayed in the UCSC genome browser together with the annotated miR34a asRNA transcript from GENCODE.

<br></br>

####Conclusions

Sequencing of the resulting cloned cDNA indicated the transcripts 3’ transcription termination site to be 125bp upstream of the RP3-510D11.2 transcript’s annotated termination site.

<br></br><br></br>

###Figure 1e

####Introduction

We desired to determine the 5'-start site for the miR34a asRNA.

<br></br>

####Methods

To accomplish this we performed the primer walk assay. All cell lines were cultured at 5% CO2 and 37° C with HEK293T cells cultured in DMEM high glucose (Hyclone). All growth mediums were supplemented with 10% heat-inactivated FBS and 50 μg/ml of streptomycin and 50 μg/ml of penicillin. RNA was extracted using the RNeasy mini kit (Qiagen) and subsequently treated with DNase (Ambion Turbo DNA-free, Life Technologies). 500ng RNA was used for cDNA synthesis using MuMLV (Life Technologies) and a 1:1 mix of oligo(dT) and random nanomers. PCR was performed for 35 cycles at 62°C annealing temperature and the resulting products were analyzed on a 2% agarose gel.

<br></br>

#####Primers

```{r}
primers <- data.frame(
    name=c(
        "miR34aAS_F10",
        "miR34aAS_F11",
        "miR34aAS_F12",
        "miR34aAS_F13",
        "miR34aAS_F14",
        "miR34aAS_F15",
        "miR34aAS_R1"
    ),
    sequence=c(
        "ACG CGT CTC TCC AGC CCG GGA T",
        "ATC TGC GTG GTC ACC GAG AAG CA",
        "CGC ACG GAC TGA GAA ACA CAA G",
        "ACG GAG GCT ACA CAA TTG AAC AGG",
        "AGG GAA GAA AGA ACT AGC CGA GCA",
        "CAT TTG CTG CAA TAT CAC CGT GGC",
        "TTG CCT CGT GAG TCC AAG GAG AAT"
    )
)
primers
```

<br></br>

####Results

Primer placement at the locus can be viewed in Supplementary Figure 1b.

```{r, out.width = "500px"}
url <- "primerWalk/gel.png"
knitr::include_graphics(file.path(projectUrl, url))
```

<br></br>

The semi-quantitative PCR results from the primer walk assay performed using HEK293T cells.

<br></br>

####Conclusion

It appears as though the miR34a asRNA transcript start site is somewhat upstream of the annotated start site. Although there is some product present for the F12 primer, the majority of transcription seems to be originating from the F11 primer. This indicates that the start site is between 71-224bp upstream of the annotated start site.

<br></br><br></br>


###Supplementary Figure 1b

<br></br><br></br>


```{r, out.width = "850px"}
url <- "primerWalk/schematic_hg38.png"
knitr::include_graphics(file.path(projectUrl, url))
```

<br></br>

A schematic picture of the miR34a asRNA locus including the location of the primers used in the primer walk assay.

<br></br>

###Figure 1f

####Introduction

We utilized a bioinformatic approach to evaluate the coding potential of the miR34a asRNA transcript. The Coding-potential assessment tool uses a linear regression model to evaluate coding-potential by examining ORF length, ORF coverage, Fickett score and hexamer score. We further confirmed these results using the Coding-potential Calculator which utilizes a support based machine-based classifier and accesses an alternate set of discriminatory features.

<br></br>

####Methods

Protein-coding capacity was evaluated using the [Coding-potential Assessment Tool](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3616698/) and [Coding-potential Calculator](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1933232/) with default settings. Transcript sequences for use with Coding-potential Assessment Tool were downloaded from the UCSC genome browser using the following IDs: HOTAIR (ENST00000455246.1), XIST (ENST00000429829.1), β-actin (ENST00000331789.5), Tubulin (ENST00000427480.1), and MYC (ENST00000377970). Transcript sequences for use with Coding-potential Calculator were downloaded from the UCSC genome browser using the following IDs: HOTAIR (uc031qho.1), β-actin (uc003soq.4).

<br></br>

####Results

#####Coding Potential Assesment Tool

```{r}
url <- "codingPotential/CodingPotentialAssessmentTool/results.txt"
cpat <-read.table(url(file.path(projectUrl, url)), header = TRUE, sep = "\t")
cpat
```

<br></br>

Coding potential analysis results from the Coding-potential Assessment Tool including miR34a asRNA and two characterized non-coding transcripts (HOTAIR and XIST) and 3 known coding transcripts (β-actin, tubulin, and MYC). 

<br></br>

####Conclusions

Results indicated that miR34a asRNA has a similar lack of coding capacity to the known non-coding transcripts HOTAIR and XIST and differs greatly when examining these parameters to the known coding transcripts beta-actin, tubulin, and MYC. However, to fully evaluate coding potential methods such as mass spectrometry or ribosome profiling must be used. 

<br></br><br></br>


###Supplementary Figure 1g

<br></br><br></br>

```{r}
url <- "codingPotential/CodingPotentialCalculator/results.txt"
cpc <- read.table(url(file.path(projectUrl, url)), header = TRUE, sep = "\t")
cpc
```

<br></br>

Coding potential analysis results using the Coding-potential calculator for miR34a asRNA, HOTAIR, and β-actin.

<br></br>


###Supplementary Figure 1c

<br></br><br></br>

```{r, out.width = "500px"}
url <- "polyadenylation/polyadenylation.png"
knitr::include_graphics(file.path(projectUrl, url))
```

<br></br>

Add description.

<br></br><br></br>

###Supplementary Figure 1d

<br></br><br></br>

```{r, out.width = "850px"}
url <- "isoforms/isoforms.png"
knitr::include_graphics(file.path(projectUrl, url))
```

<br></br>

Add description.

<br></br><br></br>

###Supplementary Figure 1e

####Introduction
AsRNAs are known to be capable of exerting regulatory functions on both a transcriptional and post-transcriptional level and therefore, the cellular localization of the asRNA can be informative concerning the method of regulation. 

####Methods
All cell lines were cultured at 5% CO2 and 37° C with HEK293T cells cultured in DMEM high glucose (Hyclone). All growth mediums were supplemented with 10% heat-inactivated FBS and 50 μg/ml of streptomycin and 50 μg/ml of penicillin. Cells were plated at 300,000 cells per well in a 6-well plate and cultured overnight. Fractionation was performed using the PARIS kit according to the manufacturer's recommendations (Life Technologies). 0.5ul RNA was used for cDNA synthesis using MuMLV (Life Technologies) and a 1:1 mix of oligo(dT) and random nanomers. QPCR was carried out using KAPA 2G SYBRGreen (Kapa Biosystems) using the Applied Biosystems 7900HT machine with the cycling conditions: 95 °C for 3 min, 95 °C for 3 s, 60 °C for 30 s.

```{r}
library(printr)
primers <- data.frame(
    name=c(
        "ß-actin Fwd",
        "ß-actin Rev",
        "miR34a HG_F",
        "miR34a HG_R",
        "miR34a_asF1",
        "miR34a_asR1",
        "U48_F",
        "U48_R"
        
    ),
    sequence=c(
        "AGG TCA TCA CCA TTG GCA ATG AG",
        "CTT TCG GGA TGT CCA CGT CA",
        "TCT GCT CCA GTG GCT GAT GAG AAA",
        "GTT CAC TGG CCT CAA AGT TGG CAT",
        "AGC GGC ATC TCC TCC ACC TGA AA",
        "TTG CCT CGT GAG TCC AAG GAG AAT",
        "AGT GAT GAT GAC CCC AGG TA",
        "GGT CAG AGC GCT GCG GTG AT"
        
    )
)
primers
```

####Results
```{r, message = FALSE}
#download and format data
url <- "cellularLocalization.rda"
invisible(repmis::source_data(file.path(dataUrl, url)))
data <- .

#calculate dct
groups <- c("fraction", "Cell line", "gene")
gois <- c("miR34a\nasRNA", "miR34a\nHG")

data <- data %>%
  technicalMeans(., groups) %>%
  mutate(antiLog = 2^-ctMean) %>%
  group_by(gene) %>%
  mutate(
    total = sum(antiLog),
    percent = 100 * (antiLog / total)
  )
```

```{r, fig.align='center', fig.height=8, fig.width=12, message=FALSE, warning=FALSE}
ggplot(data = NULL) +
  geom_bar(
    data = data,
      aes(
        x = gene,
        y = percent,
        fill = fraction
      ),
      position = "fill",
      stat = "identity",
      alpha = 0.6
    ) +
    labs(
      x = "Gene",
      y = "",
      title = "miR34a asRNA cellular localization",
      caption = 
            "Q-PCR results from fractionated HEK293T cells investigating the cellular localization of the miR34a asRNA transcript."
    ) +
    theme_few() +
    scale_y_continuous(labels = scales::percent_format()) +
    scale_fill_ptol() +
    scale_colour_ptol() +
    theme(
        plot.title = element_text(
            hjust = 0.5,
            face = "bold",
            size = 20
        ),
        plot.caption = element_text(
            hjust = 0, 
            margin = margin(t = 15),
            family = "Arial Unicode MS"
        ),
        legend.position = "top",
        legend.title = element_text(size = 17),
        legend.text = element_text(size = 15),
        axis.title = element_text(size = 17),
        axis.title.x = element_text(margin = margin(t = 10)),
        axis.text = element_text(size = 15),
        strip.text.x = element_text(size = 17)
    )+
    guides(fill = guide_legend(title = "Fraction", override.aes = list(size = 3)))
```

####Conclusion
Semi-quantitative PCR revealed that the miR34a asRNA transcript localizes to both the nucleus and cytoplasm although the majority is located in the cytoplasmic fraction.

###Supplementary Figure 1f

####Introduction
We desired to deduce the relative stability of the miR34a asRNA and HG transcripts.

####Methods
All cell lines were cultured at 5% CO2 and 37° C with U2OS cells grown in McCoy’s 5a (Life Technologies). All growth mediums were supplemented with 10% heat-inactivated FBS and 50 μg/ml of streptomycin and 50 μg/ml of penicillin. Cells were treated with 10nM actinomycin D (Sigma) for the indicated times. RNA was extracted using the RNeasy mini kit (Qiagen) and subsequently treated with DNase (Ambion Turbo DNA-free, Life Technologies). 500ng RNA was used for cDNA synthesis using MuMLV (Life Technologies) and a 1:1 mix of oligo(dT) and random nanomers. QPCR was carried out using KAPA 2G SYBRGreen (Kapa Biosystems) using the Applied Biosystems 7900HT machine with the cycling conditions: 95 °C for 3 min, 95 °C for 3 s, 60 °C for 30 s.

```{r}
library(printr)
primers <- data.frame(
    name=c(
        "ß-actin Fwd",
        "ß-actin Rev",
        "miR34a HG_F",
        "miR34a HG_R",
        "miR34a_asF1",
        "miR34a_asR1"
        
    ),
    sequence=c(
        "AGGTCATCACCATTGGCAATGAG",
        "CTTTCGGGATGTCCACGTCA",
        "TCTGCTCCAGTGGCTGATGAGAAA",
        "GTTCACTGGCCTCAAAGTTGGCAT",
        "AGCGGCATCTCCTCCACCTGAAA",
        "TTGCCTCGTGAGTCCAAGGAGAAT"
        
    )
)
primers
```

####Results

```{r, message = FALSE}
#download and format data
url <- "transcriptStability.rda"
invisible(repmis::source_data(file.path(dataUrl, url)))
data <- .

#calculate dct
groups <- c("treatment", "Biological Replicate", "gene")
gois <- c("miR34a asRNA", "miR34a HG")

data <- technicalMeans(data, groups) %>%
  dct(., gois, Actin, groups, F)

#calculate ddct
groups[groups == "gene"] <- "GOI"
logical <- tibble(
  treatment = data$treatment == 0
)

data <- data %>%
  ddct(., logical, groups, F) %>%
  folds(.)

#calculate stats
stats <- data %>%
  group_by(GOI, treatment) %>%
  summarize(mean = mean(log2fold), sd = sd(log2fold)) %>%
  ungroup()
```

```{r, fig.align='center', fig.height=8, fig.width=12, eval=TRUE, message=FALSE, warning=FALSE}
ggplot(data = NULL)+
  geom_point(
    data = data,
    aes(
      x = treatment,
      y = log2fold,
      colour = GOI
    ),
    fill = "white",
    size = 4,
    alpha = 0.6,
    position = position_dodge(width = 0.9)
  ) +
  geom_linerange(
    data = stats,
    aes(
      x = treatment,
      ymin = mean - sd,
      ymax = mean + sd,
      group = GOI
    ),
    colour = "black",
    position = position_dodge(width = 0.9),
    show.legend = FALSE
  ) +
  geom_point(
    data = stats,
    aes(
      x = treatment,
      y = mean,
      group = GOI
    ),
    colour = "black",
    position = position_dodge(width = 0.9),
    show.legend = FALSE
  ) +
  scale_y_continuous("\u0394\u0394Ct") +
  labs(
    x = "Treatment (hours)",
    title = "miR34a asRNA and HG stability",
    caption = 
            "Q-PCR showing miR34a asRNA and miR34a HG levels in U2OS cells after treatment with 10nM actinomycin D.\nError bars (black vertical) indicate standard deviation and means are represented by black dots."
  ) +
  theme_few() +
  scale_fill_ptol() +
  scale_colour_ptol() +
  theme(
    plot.title = element_text(
      hjust = 0.5,
      face = "bold",
      size = 20
    ),
    plot.caption = element_text(
      hjust = 0, 
      margin = margin(t = 15),
      family = "Arial Unicode MS"
    ),
    legend.position = "top",
    legend.title = element_text(size = 17),
    legend.text = element_text(size = 15),
    axis.title = element_text(size = 17),
    axis.title.x = element_text(margin = margin(t = 10)),
    axis.text = element_text(size = 15),
    strip.text.x = element_text(size = 17)
  ) +
  guides(color = guide_legend(title = "Gene", override.aes = list(size = 3)))
```

####Conclusions
While the levels of miR34a HG fall quickly over the 4 hour treatment period, levels of miR34a asRNA remain relativley stable.

##Figure 2 {.tabset}

###Figure 2a

####Introduction

miR34a is a known downstream target of p53 and has been previously shown to exhibit increased expression upon cellular apoptotic signaling. We hypothesized that miR34a asRNA may be regulated in a similar fashion whereby transcription is stimulated by activation of p53. To test this we treated p53 wild type and p53 null HCT116 cells with increasing concentrations of the DNA damaging agent doxorubicin and monitored miR34a asRNA expression.

<br></br>

####Methods

HCT116 cells were cultured in DMEM high modified (Hyclone, GE healthcare) supplemented with supplemented with 2mM L-glutamine, 50ug/ml Penicillin-Streptomycin and 10% Fetal Calf Serum. 25 x 10^5 HCT116 cells were plated in 6 well plates. 24 hours later media was exchanged and doxorubicin was added to a final concentration of 100, 200 or 500 ng/ml. Cells were harvested for RNA extraction 24 hours later using trypsin. RNA was extracted using Nucleospin RNA kit (Machery-Nagel Ref. 740955) according to manufacturer‟s protocol and DNase treated using Ambion Turbo DNA-free according to manufacturer‟s protocol (Life Technologies Ref. AM1907). cDNA was synthesized using ~500 ng RNA with M-MLV (Life Technologies Ref 28025013) and a mixture of oligo(dT) with nanomers in accordance with the manufacturer's protocol. qPCR quantification was carried out using the PowerUp SYBR Green Master Mix (Thermo Fisher Scientific, Ref. A25777) on the CFX96 Touch Real-Time PCR Detection System: 50°C for 2 min, 95°C for 2min, and 95°C for 1 sec followed by 60°C for 30 sec repeated for 40 cycles. qPCR primers are shown below. Two experimental (technical) replicates were included in each QPCR run and delta ct was calculated for each sample using the mean of the gene of interest's technical replicates and the house keeping gene's technical replicates. delta-delta ct was calculated for each sample by subtracting the dct value for the corresponding untreated sample. Three independant experiments (biological replicates) were performed in total. The Student's t-test was used to compare the p-53 wild type and p-53 null groups in each treatment concentration and for both genes.  

<br></br>

#####Primers

```{r}

primers <- data.frame(
    name=c(
        "ß-actin Fwd",
        "ß-actin Rev",
        "miR34a HG_F",
        "miR34a HG_R",
        "miR34a_asF1",
        "miR34a_asR1"
        
    ),
    sequence=c(
        "AGGTCATCACCATTGGCAATGAG",
        "CTTTCGGGATGTCCACGTCA",
        "TCTGCTCCAGTGGCTGATGAGAAA",
        "GTTCACTGGCCTCAAAGTTGGCAT",
        "AGCGGCATCTCCTCCACCTGAAA",
        "TTGCCTCGTGAGTCCAAGGAGAAT"
        
    )
)
primers
```

<br></br>

####Results

```{r, message = FALSE}
#download and format data
url <- "HCT116p53null.rda"
invisible(repmis::source_data(file.path(dataUrl, url)))
data <- .

#calculate dct
groups <- c("Condition", "Treatment", "Biological Replicate", "Cell line", "gene")
gois <- c("miR34a asRNA", "miR34a HG")

data <- technicalMeans(data, groups) %>%
  dct(., gois, Actin, groups)

#calculate ddct
groups[groups == "gene"] <- "GOI"
logical <- tibble(
  Treatment = data$Treatment == 0
)

data <- data %>%
  ddct(., logical, groups) %>%
  folds(.)

#calculate stats
stats <- calcStats(data, Condition, "p53-wt", groups) %>%
  pFormat(.)

#setup plotting variables
stats <- stats %>%
  group_by(GOI, Treatment) %>%
  mutate(
    max = max(CI95h),
    x = case_when(
      Treatment ==   0 ~ 1 + 0.25,
      Treatment == 100 ~ 2 + 0.25,
      Treatment == 200 ~ 3 + 0.25,
      Treatment == 500 ~ 4 + 0.25
    ),
    xend = case_when(
      Treatment ==   0 ~ 1 - 0.25,
      Treatment == 100 ~ 2 - 0.25,
      Treatment == 200 ~ 3 - 0.25,
      Treatment == 500 ~ 4 - 0.25
    )
  ) %>%
  ungroup()
```

```{r, fig.align='center', fig.height=8, fig.width=12}
#plot for Rmarkdown
ggplot(data = NULL) +
  geom_violin(
    data = data,
    aes(
      x = Treatment, 
      y = log2fold,
      group = interaction(Condition, Treatment)
    ),
    trim = TRUE,
    alpha = 0.3,
    fill = "gray",
    colour = NA,
    scale = "width",
    show.legend = FALSE
  ) +
  geom_dotplot(
    data = data, 
    aes(x = Treatment, y = log2fold, fill = Condition),
    colour = NA,
    binwidth = 1/10,
    alpha = 0.5,
    position = position_dodge(width = 0.9), 
    binaxis='y', 
    stackdir='center',
    dotsize = 2,
    show.legend = TRUE
  ) +
  geom_point(
    data = stats,
    aes(x = Treatment, y = mean, group = Condition), 
    position = position_dodge(width = 0.9),
    colour = "gray20",
    size = 6,
    shape = 95
  ) +
  facet_grid(. ~ GOI) +
  geom_linerange(
    data = stats,
    aes(x = Treatment, ymin = CI95l, ymax = CI95h, group = Condition),
    colour = "gray20",
    position = position_dodge(width = 0.9),
    show.legend = FALSE
  ) +
  geom_label(
    data = stats,
    aes(x = Treatment, y = max + 0.28, label = pFormat),
    label.size = 0,
    label.padding = unit(0.01, "lines"),
    show.legend = FALSE,
    fill = "white",
    size = 5,
    family = "Arial Unicode MS",
    position = position_dodge(width = 0.9)
  ) +
  geom_segment(
    data = stats,
    aes(x = x, y = max + 0.1, xend = xend, yend = max + 0.1),
    colour="grey43",
    show.legend = FALSE
  ) +
  scale_y_continuous("\u0394\u0394Ct", breaks = c(-1,0,1,2,3,4,5))+
  labs(
    x = "Treatment",
    title = "miR34a asRNA regulation by p53 induction",
    caption = 
            "Monitoring the effects of 24 hours doxorubicin treatment on miR34a asRNA and HG in P53+/+ and p53-/- HCT116 cells."
  ) +
  theme_few() +
  scale_fill_ptol() +
  scale_colour_ptol() +
  theme(
    plot.title = element_text(
      hjust = 0.5,
      face = "bold",
      size = 20
    ),
    plot.caption = element_text(
      hjust = 0, 
      margin = margin(t = 15),
      family = "Arial Unicode MS"
    ),
    legend.position = "top",
    legend.title = element_text(size = 17),
    legend.text = element_text(size = 15),
    axis.title = element_text(size = 17),
    axis.title.x = element_text(margin = margin(t = 10)),
    axis.text = element_text(size = 15),
    strip.text.x = element_text(size = 17)
  )+
  guides(
    fill = guide_legend(title = "Cell line", override.aes = list(size=3)),
    colour = FALSE  
  )
```

<br></br>

####Conclusions

We observed increasing miR34a HG and asRNA expression with increasing doses of doxorubicin indicating that these two transcripts are co-regulated by p53. Although p53 wild type cells showed a 25-fold increase over untreated cells in miR34a AS expression at the highest tested doxorubicin concentration, this effect was largely abrogated, 11 fold over untreated, in p53 null cells, indicating that p53 is a major regulator of miR34a asRNA expression. 

<br></br><br></br>

###Figure 2b

####Introduction

miR34a is a known downstream target of p53 and has been previously shown to exhibit increased expression with cellular apoptotic signaling. We hypothesized that miR34a asRNA may be regulated in a similar fashion whereby transcription is stimulated by activation of p53. To test this we treated HCT116 and HEK293t cells with 200 ng/ml of the DNA damaging agent doxorubicin for 24 hours and monitored miR34a asRNA expression. 

<br></br>

####Methods

All cell lines were cultured at 5% CO2 and 37° C with HEK293T cells cultured in DMEM high glucose (Hyclone) and HCT116 cells in McCoy’s 5a (Life Technologies). All growth mediums were supplemented with 10% heat-inactivated FBS and 50 μg/ml of streptomycin and 50 μg/ml of penicillin. Cells were plated at 300,000 cells per well in a 6-well plate and cultured overnight. The following day cells were treated with 0, 100, 200, or 500 ng/ml doxorubicin for 24hrs. RNA was extracted using the RNeasy mini kit (Qiagen) and subsequently treated with DNase (Ambion Turbo DNA-free, Life Technologies). 500ng RNA was used for cDNA synthesis using MuMLV (Life Technologies) and a 1:1 mix of oligo(dT) and random nanomers. QPCR was carried out using KAPA 2G SYBRGreen (Kapa Biosystems) using the Applied Biosystems 7900HT machine with the cycling conditions: 95 °C for 3 min, 95 °C for 3 s, 60 °C for 30 s.

Two experimental (technical) replicates were included in each QPCR run and delta ct was calculated for each sample using the mean of the gene of interest's technical replicates and the house keeping gene's technical replicates. delta-delta ct was calculated for each sample by subtracting the dct value for the corresponding untreated sample. Three independant experiments (biological replicates) were performed in total. The Student's t-test was used to compare the p-53 wild type and p-53 null groups in each treatment concentration and for both genes. 

<br></br>

#####Primers

```{r}
primers <- data.frame(
    name=c(
        "ß-actin Fwd",
        "ß-actin Rev",
        "miR34a HG_F",
        "miR34a HG_R",
        "miR34a_asF1",
        "miR34a_asR1"
        
    ),
    sequence=c(
        "AGGTCATCACCATTGGCAATGAG",
        "CTTTCGGGATGTCCACGTCA",
        "TCTGCTCCAGTGGCTGATGAGAAA",
        "GTTCACTGGCCTCAAAGTTGGCAT",
        "AGCGGCATCTCCTCCACCTGAAA",
        "TTGCCTCGTGAGTCCAAGGAGAAT"
        
    )
)
primers
```

<br></br>

####Results

```{r, message = FALSE}
url <- "HctHekDox.rda"
invisible(repmis::source_data(file.path(dataUrl, url)))
data <- .

#calculate dct
groups <- c("Treatment", "Biological Replicate", "Cell line", "gene")
gois <- c("miR34a asRNA", "miR34a HG")

data <- technicalMeans(data, groups) %>%
  dct(., gois, Actin, groups)

#calculate ddct
groups[groups == "gene"] <- "GOI"
logical <- tibble(
  `Biological Replicate` = pull(data, `Biological Replicate`) == 1,
  Treatment = pull(data, Treatment) == "untreated"
)

data <- data %>%
  ddct(., logical, groups) %>%
  folds(.)

#calculate stats
stats <- calcStats(data, Treatment, "untreated", groups) %>%
  pFormat(.)

#prepare for plotting

#vertical lines and labels
vl <- stats %>%
  group_by(`Cell line`, GOI) %>%
  summarize(
    max = max(CI95h),
    pValue = as.numeric(min(pValue))
  ) %>%
  ungroup() %>%
  pFormat(.) %>%
  mutate(
    x = case_when(
      GOI == "miR34a asRNA" ~ 1 - 0.25,
      GOI ==    "miR34a HG" ~ 1 + 0.25
    ),
    xend = case_when(
      GOI == "miR34a asRNA" ~ 2 - 0.25,
      GOI ==    "miR34a HG" ~ 2 + 0.25
    ),
    pFormat = pFormat,
    labelX = 1.5
  )
```

```{r, fig.align='center', fig.height=8, fig.width=12}
#plot for Rmarkdown
ggplot(data = NULL) +
  geom_violin(
    data = data,
    aes(
      x = Treatment,
      y = log2fold,
      group = interaction(GOI, Treatment)
    ),
    trim = TRUE,
    fill = "gray",
    alpha = 0.3,
    color = NA,
    scale = "width",
    show.legend = FALSE
  ) +
  geom_dotplot(
    data = data, 
    aes(
      x = Treatment,
      y = log2fold,
      fill = GOI
    ),
    binwidth = 1/10,
    colour = NA,
    alpha = 0.5,
    position = position_dodge(width = 0.9), 
    binaxis='y', 
    stackdir='center',
    dotsize = 2,
    show.legend = TRUE
  ) +
  geom_point(
    data = stats,
    aes(
      x = Treatment,
      y = mean,
      group = GOI
    ),
    size = 6,
    position = position_dodge(width = 0.9),
    show.legend = FALSE,
    shape = 95,
    colour = "gray20"
  ) +
  facet_grid(. ~ `Cell line`) +
  geom_linerange(
    data = stats,
    aes(
      x = Treatment,
      ymin = CI95l,
      ymax = CI95h,
      group = GOI
    ),
    colour = "gray20",
    position = position_dodge(width = 0.9),
    show.legend = FALSE
  ) +
  geom_label(
    data = vl,
    aes(
      x = labelX,
      y = max + 0.25,
      label = pFormat,
      group = GOI
    ),
    label.size = 0,
    label.padding = unit(0.01, "lines"),
    show.legend = FALSE,
    fill = "white",
    size = 5,
    family = "Arial Unicode MS",
    position = position_dodge(width = 0.9)
  ) +
  geom_segment(
    data = vl,
    aes(
      x = x,
      y = max + 0.1,
      xend = xend,
      yend = max + 0.1,
      group = GOI
    ),
    colour="grey43",
    show.legend = FALSE
  ) +
  scale_y_continuous(
    "\u0394\u0394Ct",
    breaks=c(-1,0,1,2,3,4,5)
  ) +
  labs(
    x = "Treatment",
    title = "miR34a asRNA response to p53 induction",
    caption = 
            "Q-PCR showing miR34a asRNA and miR34a HG levels in HCT116 and HEK293t cells after treatment with 200 ng/ml doxorubicin for 24hrs.\nError bars (black vertical) indicate 95% CI and means are represented by black dots."
  ) +
  theme_few() +
  scale_fill_ptol() +
  scale_colour_ptol() +
  theme(
    plot.title = element_text(
      hjust = 0.5,
      face = "bold",
      size = 20
    ),
    plot.caption = element_text(
      hjust = 0, 
      margin = margin(t = 15),
      family = "Arial Unicode MS"
    ),
    legend.position = "top",
    legend.title = element_text(size = 17),
    legend.text = element_text(size = 15),
    axis.title = element_text(size = 17),
    axis.title.x = element_text(margin = margin(t = 10)),
    axis.text = element_text(size = 15),
    strip.text.x = element_text(size = 17)
  ) +
  guides(
    color = FALSE,
    fill = guide_legend(title = "Gene", override.aes = list(size = 3))
  )
```

<br></br>

####Conclusions

We observed increases in miR34a HG and asRNA expression upon doxorubicin treatment indicating that these two transcripts are co-regulated. 

<br></br><br></br>

###Figure 2c

####Introduction

It is likely, due to the head-to head orientation of miR34a HG and asRNA, that transcription initiation may be activated from a single promoter in a bi-directional manner. To investigate whether miR34a HG and asRNA are transcribed from the same promoter as convergent transcripts, we cloned the miR34a HG promoter, including the p53 binding site, into a luciferase/renilla dual reporter vector which we hereafter refer to as p1. The p1 sequence was previously published in Raver-Shapira, N., et al., Transcriptional activation of miR-34a contributes to p53-mediated apoptosis. Mol Cell, 2007. 26(5): p. 731-43. Due to the observance of additional conservation sites downstream of the previously published miR34a HG promoter, we also cloned a longer "promoter region", extending 251 additional base pairs into the miR34a HG into the same bi-directional reporter vector, hereafter refered to as p2.

<br></br>

####Methods

All cell lines were cultured at 5% CO2 and 37° C with HEK293T cells cultured in DMEM high glucose (Hyclone) and HCT116 cells in McCoy’s 5a (Life Technologies). All growth mediums were supplemented with 10% heat-inactivated FBS and 50 μg/ml of streptomycin and 50 μg/ml of penicillin. Cells were plated at 10,000 cells per well in a 96-well plate with a white bottom and cultured overnight. The following day cells were co-transfected with 10ng of empty, p1, or p2 plasmid and GFP using the standard lipofektamine 2000 (Life Technologies) protocol. The expression of GFP and luminescence was measured 24 h post transfection by using the Dual-Glo Luciferase Assay System (Promega) and detected by the GloMax-Multi+ Detection System (Promega). The expression of luminescence was normalized to GFP.

<br></br>

#####Constructs

```{r}
library(printr)
ps <- data.frame(
    name=c("p1", "p2"),
    sequence=c("GCG CCC TGC CTG GCC CCC ACC TGG TCC TCT TTC CTT TTC AGG TGG AGG AGA TGC CGC TGT
              CCC GTC GGT CTG GGG ACA GCC CAG CTC CCC GGA TCC CGG GCT GGA GAG ACG CGT CGC GGC   
              CCC GGG GCC TGG TGG CAC GAG CAG GAA GGA GGA CCC GGC GGC GGG CTC TGC CTG GGC TTG 
              CCT GGG CTT GTT CCG AGC CGG GCT GCT TCT CGG TGA CCA CGC AGA TCG GGG GCA TTT GGA
              GAT TTT GCG GGA GTC CTG CAG CCA AGC TCC GGG GCA GGA GAG GCC TGG AAG CCT GCA CTA
              CCT GCT C",
              "GCG CCC TGC CTG GCC CCC ACC TGG TCC TCT TTC CTT TTC AGG TGG AGG AGA TGC CGC TGT 
              CCC GTC GGT CTG GGG ACA GCC CAG CTC CCC GGA TCC CGG GCT GGA GAG ACG CGT CGC GGC
              CCC GGG GCC TGG TGG CAC GAG CAG GAA GGA GGA CCC GGC GGC GGG CTC TGC CTG GGC TTG 
              CCT GGG CTT GTT CCG AGC CGG GCT GCT TCT CGG TGA CCA CGC AGA TCG GGG GCA TTT GGA 
              GAT TTT GCG GGA GTC CTG CAG CCA AGC TCC GGG GCA GGA GAG GCC TGG AAG CCT GCA CTA 
              CCT GCT CGC CCC GTC CCA GCA TGC ACC CAG GTA AAC GCT TGT GTT TCT CAG TCC GTG CGA 
              AAG TTT GCA AAG AAG GAG GCG GGA ACT AGA CCA ACA ACT TTA ATA ATC ATA GTC ATA ATA 
              ATG AAA ACC CTG TTC AAT TGT GTA GCC TCC GTA AGG GGA AGA ATT TCC AAG AAG CAA CAA
              AAT GTT CTT TGA TTT CAT AAT CTC TGG GGA GAA AGG ATT CCT TTT TTG GGG TTT TGC TCG
              GCT AGT TCT TTC TTC CCT")
)
ps
```

<br></br>

#####Primers

```{r}
primers <- data.frame(
    name=c(
        "Luc_set_II_F",
        "Luc_set_II_R",
        "Renilla_pBiDir_F1",
        "Renilla_pBiDir_R1",
        "B-actin_F",
        "B-actin_R"
    ),
    sequence=c(
        "AAG ATT CAA AGT GCG CTG CTG",
        "TTG CCT GAT ACC TGG CAG ATG",
        "TAA CGC GGC CTC TTC TTA TTT",
        "GAT TTG CCT GAT TTG CCC ATA",
        "AGG TCA TCA CCA TTG GCA ATG AG",
        "CTT TGC GGA TGT CCA CGT CA"
    )
)
primers
```

<br></br>

####Results

```{r, out.width = "850px"}
url <- "P1-HCTandHEK/locusSimple.png"
knitr::include_graphics(file.path(projectUrl, url))
```

<br></br>

```{r, out.width = "850px"}
url <- "P1-HCTandHEK/constructSimple.png"
knitr::include_graphics(file.path(projectUrl, url))
```

<br></br>

```{r, message = FALSE}
url <- "P1-HCTandHEK.rda"
invisible(repmis::source_data(file.path(dataUrl, url)))
data <- . 

.calcP <- function(`Cell line`, gene, logValue, data) {
  bool1 <- pull(data, `Cell line`) == `Cell line`
  bool2 <- pull(data, gene) == gene
  bool3 <- pull(data, construct) == "Empty"
  bool <- bool1 & bool2 & bool3
  t.test(logValue, pull(data, logValue)[bool])$p.value
}

data <- data %>%
    mutate(logValue = log2(value))

stats <- data %>%
  group_by(`construct`, `Cell line`, gene) %>%
  summarize(
    n = n(),
    mean = mean(logValue),
    CI95l = t.test(logValue)$conf.int[1],
    CI95h = t.test(logValue)$conf.int[2],
    pValue = .calcP(`Cell line`, gene, logValue, data)
  ) %>%
  ungroup() %>%
  pFormat(.)

#vertical lines and labels in plot
ys <- stats %>%
  group_by(`Cell line`) %>%
  summarize(max = max(CI95h)) 

vl <- stats %>%
  group_by(`Cell line`, gene) %>%
  summarize(
    pValue = as.numeric(min(pValue))
  ) %>%
  ungroup() %>%
  pFormat(.) %>%
  mutate(
    x = case_when(
      gene ==    "Renilla" ~ 1 + 0.25,
      gene == "Luciferase" ~ 1 - 0.25
    ),
    xend = case_when(
      gene ==    "Renilla" ~ 2 + 0.25,
      gene == "Luciferase" ~ 2 - 0.25
    ),
    pFormat = pFormat,
    labelX = 1.5
  ) %>%
  left_join(ys) %>%
  mutate(max = max + c(0.25, 1, 0.25, 1))
```

```{r, fig.align='center', fig.height=8, fig.width=12}
data <- filter(data, construct %in% c("p1", "Empty"))
stats <- filter(stats, construct %in% c("p1", "Empty"))

#plot for Rmarkdown
ggplot(data = NULL) +
  geom_violin(
    data = data,
    aes(
      x = construct,
      y = logValue,
      group = interaction(construct, gene)
    ),
    trim = TRUE,
    alpha = 0.3,
    fill = "gray",
    color = NA,
    scale = "width",
    show.legend = FALSE
  ) +
  geom_dotplot(
    data = data, 
    aes(
      x = construct,
      y = logValue,
      fill = gene
    ),
    binwidth = 1/10,
    alpha = 0.5,
    colour = NA,
    position = position_dodge(width = 0.9), 
    binaxis = 'y', 
    stackdir = 'center',
    dotsize = 2,
    show.legend = TRUE
  ) +
  facet_grid(. ~ `Cell line`) +
  geom_linerange(
    data = stats,
    aes(
      x = construct,
      ymin = CI95l,
      ymax = CI95h,
      group = gene
    ),
    colour = "grey20",
    position = position_dodge(width = 0.9),
    show.legend = FALSE
  ) +
  geom_point(
    data = stats,
    aes(
      x = construct,
      y = mean,
      group = gene
    ),
    position = position_dodge(width = 0.9),
    show.legend = FALSE,
    colour = "gray20",
    size = 6,
    shape = 95
  ) +
  geom_label(
    data = vl,
    aes(
      x = labelX, 
      y = max + 0.25, 
      label = pFormat, 
      group = gene
    ),
    label.size = 0,
    label.padding = unit(0.01, "lines"),
    show.legend = FALSE,
    fill = "white",
    size = 5,
    family = "Arial Unicode MS",
    position = position_dodge(width = 0.9)
  ) +
  geom_segment(
    data = vl,
    aes(
      x = x,
      y = max,
      xend = xend,
      yend = max
    ),
    colour = "grey43",
    show.legend = FALSE
  ) +
  labs(
    x = "Construct",
    y = "log2 activity (a.u.)",
    title = "miR34a asRNA regulation by miR34a promoter",
    caption = 
            "miR34a asRNA and miR34a HG levels in HCT116 and HEK293t cells after transfection with P1 and P2 constructs.\nLuciferase represents miR34a HG and renilla represents miR34a asRNA.\nError bars (black vertical) indicate 95% CI and means are represented \nby black dots."
    ) +
  theme_few() +
  scale_fill_ptol() +
  scale_colour_ptol() +
  theme(
    plot.title = element_text(
      hjust = 0.5,
      face = "bold",
      size = 20
    ),
    plot.caption = element_text(
      hjust = 0, 
      margin = margin(t = 15),
      family = "Arial Unicode MS"
    ),
    legend.position = "top",
    legend.title = element_text(size = 17),
    legend.text = element_text(size = 15),
    axis.title = element_text(size = 17),
    axis.title.x = element_text(margin = margin(t = 10)),
    axis.text = element_text(size = 15),
    strip.text.x = element_text(size = 17)
  ) +
  guides(
    fill = guide_legend(title = "Gene", override.aes = list(size = 3)),
    colour = FALSE   
  )
```

<br></br>

####Conclusion

Upon transfection of this construct in two separate cell lines we observed increases in both luciferase and renilla indicating that miR34a HG and asRNA expression is regulated by a single promoter contained within the p1 construct. Transfection of the P2 construct showed to cause a reduction in luciferase (miR34a HG) levels but not renilla (miR34a AS) levels. This could be due to the presence of a STAT3 binding site in p2 which has been previously reported to negativley regulate miR34a HG.

<br></br><br></br>

###Figure 2d

####Introduction

Our previous results showed that miR34a HG and asRNA are simultaneously up-regulated in response to doxorubicin treatment leading us to consider the possibility that miR34a asRNA may be necessary for an appropriate miR34a response to apoptotic signaling. To explore this possibility we again utilized the p1 construct, co-transfecting it with an shRNA targeting renilla, and treating HCT116 cells with increasing levels of doxorubicin.

<br></br>

####Methods

We utilized the P1 construct where the overlapping region of miR34a HG and miR34a AS is cloned with luciferase downstream of miR34a HG and renilla downstream of miR34a AS. The p1 sequence was previously published in Raver-Shapira, N., et al., Transcriptional activation of miR-34a contributes to p53-mediated apoptosis. Mol Cell, 2007. 26(5): p. 731-43. All cell lines were cultured at 5% CO2 and 37° C with HCT116 cells cultured in McCoy’s 5a (Life Technologies). All growth mediums were supplemented with 10% heat-inactivated FBS and 50 μg/ml of streptomycin and 50 μg/ml of penicillin. 2.5x10^5 HCT116wt cells were seeded in a 12-well plate. After 24hrs these were co-transfected with the P1 construct (25ng) and shRenilla2.1 (250ng) using lipo2000 (Life Technologies) standard protocol. 24hrs post-transfection, doxorubicin treatment was initiated using 0, 300, or 500ng/ml. 24hrs post-treatment, RNA was extracted using the RNeasy mini kit (Qiagen) and subsequently treated with DNase (Ambion Turbo DNA-free, Life Technologies). 500ng RNA was used for cDNA synthesis using MuMLV (Life Technologies) and a 1:1 mix of oligo(dT) and random nanomers. QPCR was carried out using KAPA 2G SYBRGreen (Kapa Biosystems) using the Applied Biosystems 7900HT machine with the cycling conditions: 95 °C for 3 min, 95 °C for 3 s, 60 °C for 30 s.

<br></br>

#####Constructs

```{r}
library(printr)
p1 <- data.frame(
    name="p1",
    sequence="GCG CCC TGC CTG GCC CCC ACC TGG TCC TCT TTC CTT TTC AGG TGG AGG AGA TGC CGC TGT
              CCC GTC GGT CTG GGG ACA GCC CAG CTC CCC GGA TCC CGG GCT GGA GAG ACG CGT CGC GGC   
              CCC GGG GCC TGG TGG CAC GAG CAG GAA GGA GGA CCC GGC GGC GGG CTC TGC CTG GGC TTG 
              CCT GGG CTT GTT CCG AGC CGG GCT GCT TCT CGG TGA CCA CGC AGA TCG GGG GCA TTT GGA
              GAT TTT GCG GGA GTC CTG CAG CCA AGC TCC GGG GCA GGA GAG GCC TGG AAG CCT GCA CTA
              CCT GCT C"
)
p1
```

<br></br>

#####shRNAs

```{r}
shRNAs <- data.frame(
    name=c(
        "shRenilla2.1"
    ),
    sequence=c(
        "TAA CGG GAT TTC ACG AGG C"
    )
)
shRNAs
```

<br></br>

#####Primers

```{r}
primers <- data.frame(
    name=c(
        "Luc_set_II_F",
        "Luc_set_II_R",
        "Renilla_pBiDir_F1",
        "Renilla_pBiDir_R1",
        "B-actin_F",
        "B-actin_R"
    ),
    sequence=c(
        "AAG ATT CAA AGT GCG CTG CTG",
        "TTG CCT GAT ACC TGG CAG ATG",
        "TAA CGC GGC CTC TTC TTA TTT",
        "GAT TTG CCT GAT TTG CCC ATA",
        "AGG TCA TCA CCA TTG GCA ATG AG",
        "CTT TGC GGA TGT CCA CGT CA"
    )
)
primers
```

<br></br>

####Results

```{r, message = FALSE}
url <- "p1shRNAdox.rda"
invisible(repmis::source_data(file.path(dataUrl, url)))
data <- .

#Luciferase = miR34a HG
#Renilla = miR34a AS

#calculate dct
groups <- c("treatment", "Biological Replicate", "shRNA", "gene")
gois <- c("Luciferase", "Renilla")

data <- data %>%
  technicalMeans(., groups) %>%
  dct(., gois, Actin, groups)

#calculate ddct
groups[groups == "gene"] <- "GOI"
logical <- tibble(
  `shRNA` = pull(data, shRNA) == "shRNA Control",
  treatment = pull(data, treatment) == 0
)

data <- data %>%
  ddct(., logical, groups, med = FALSE) %>%
  folds(.)

#calculate stats
stats <- calcStats(data, shRNA, "shRNA Control", groups) %>%
  pFormat(.)

#prepare for plotting
data <- data %>%
  mutate(plotGOI = case_when(
    GOI == "Luciferase" ~ "  Luciferase\n(miR34a HG)",
    GOI == "Renilla" ~ "       Renilla\n(miR34a asRNA)"
  ))

stats <- stats %>%
  mutate(plotGOI = case_when(
    GOI == "Luciferase" ~ "  Luciferase\n(miR34a HG)",
    GOI == "Renilla" ~ "       Renilla\n(miR34a asRNA)"
  ))
```

```{r, fig.align='center', fig.height=8, fig.width=12, warning = FALSE}
#plot for rmarkdown
ggplot(data = NULL) +
  geom_violin(
    data = data,
    aes(
      x = treatment, 
      y = log2fold, 
      group = interaction(plotGOI, treatment)
    ),
    trim = TRUE,
    scale = "width",
    alpha = 0.3,
    colour = NA,
    fill = "gray",
    show.legend = FALSE
  ) +
  geom_dotplot(
    data = data, 
    aes(
      x = treatment,
      y = log2fold,
      fill = plotGOI
    ),
    binwidth = 1/10,
    alpha = 0.5,
    colour = NA,
    position = position_dodge(width = 0.9), 
    binaxis = 'y', 
    stackdir = 'center',
    dotsize = 2,
    show.legend = TRUE
  ) +
  facet_grid(. ~ shRNA, scale = "free") +
  geom_linerange(
    data = stats,
    aes(
      x = treatment, 
      ymin = CI95l, 
      ymax = CI95h, 
      group = plotGOI
    ),
    colour = "gray20",
    position = position_dodge(width = 0.9),
    show.legend = FALSE
  ) +
  geom_point(
    data = stats,
    aes(
      x = treatment, 
      y = mean, 
      group = plotGOI
    ),
    colour = "gray20",
    position = position_dodge(width = 0.9),
    shape = 95,
    size = 6,
    show.legend = FALSE
  ) +
  geom_label(
    data = stats,
    aes(
      x = treatment, 
      y = CI95h + 0.2,
      label = pFormat,
      group = plotGOI
    ),
    label.size = 0,
    label.padding = unit(0.01, "lines"),
    show.legend = FALSE,
    position = position_dodge(width = 0.9),
    fill = "white",
    size = 5,
    family = "Arial Unicode MS"
  ) +
  scale_y_continuous("\u0394\u0394Ct") +
  labs(
    x = "Doxorubicin (ng/ml)",
    title = "HCT116",
    caption =
            "miR34a (luficerase) and miR34a asRNA (renilla) in HCT116 cells which were co-transfected with the P1 construct and the specified shRNA after which they\nwere treated with the indicated dose of doxorubicin. Error bars (black vertical) indicate 95% CI and black points represent the mean."
  ) +
  theme_few() +
  scale_fill_ptol() +
  scale_color_ptol() +
  theme(
    plot.title = element_text(
      hjust = 0.5,
      face = "bold",
      size = 20
    ),
    plot.caption = element_text(
      hjust = 0, 
      margin = margin(t = 15),
      family = "Arial Unicode MS"
    ),
    legend.position = "top",
    legend.title = element_text(size = 17),
    legend.text = element_text(size = 15, hjust = 0.5),
    axis.title = element_text(size = 17),
    axis.title.x = element_text(margin = margin(t = 10)),
    axis.text = element_text(size = 15)
  ) + 
  guides(
    fill = guide_legend(title = "Gene", override.aes = list(size = 4)),
    colour = FALSE
  )
```

<br></br>

####Conclusion

Levels of luciferase were abrogated in the shRenilla-transfected cells compared to control shRNA-transfected cells. Collectively, these results indicate that miR34a asRNA positively regulates levels of miR34a HG and is crucial for a proper miR34a response to apoptotic stimuli.

<br></br><br></br>

##Figure 3 {.tabset}

###Figure 3a

####Introduction

We hypothesized that miR34a asRNA may function to mediate the levels of its sense gene, miR34a. To this end, we stably over-expressed miR34a asRNA, using lentiviral transduction, in 3 different p53-null cell lines and monitored the effects on miR34a.

<br></br>

####Methods

<br></br>

#####miR34a asRNA cloning

RNA was harvested from U2OS cells using the RNeasy mini kit (Qiagen) and subsequently treated with DNase (Ambion Turbo DNA-free, Life Technologies). 500ng RNA was used for cDNA synthesis using MuMLV (Life Technologies) and a 1:1 mix of oligo(dT) and random nanomers. PCR was run using the miR34aAS_cloning_F4 and miR34aAS_cloning_Ex3_1 primers for 40 cycles at annealing temperature 58˚C and run on an agarose gel. Bands were excised and extracted using the QIAquick gel extraction kit (Qiagen). PCR products were cloned using the Strata Clone kit (Aligent) and sequenced (results below). The insert for the lentiviral construct (p-HIV-I2AG-U6) was subsequently sub-cloned from the sequenced plasmid using primers with PacI and NheI cleavage sites. The cloned transcript largley mirrors the GENCODE annotated transcript.

<br></br>

######Cloning primers

```{r}
primers <- data.frame(
    name=c(
        "miR34aAS_cloning_F4",
        "miR34aAS_cloning_Ex3_1"
    ),
    sequence=c(
        "ACGCGTCTCTCCAGCCCGGGAT",
        "AATGATGGCCGCAACTAATGACGG"
    )
)
primers
```

<br></br>

######Sequencing results

```{r}
construct <- data.frame(
    name="miR34a_asRNA",
    sequence="ACG CGT CTC TCC AGC CCG GGA TCC GGG GAG CTG GGC TGT CCC CAG ACC GAC GGG ACA GCG GCA
                TCT CCT CCA CCT GAA AAG GAA AGA GGA CCA GTT TGC AGG ACT CCG AAC TGG GCC CGC GAG ATC 
                TCC ACC TGC GCA AAA CGA AAG GGC GGA TTC TCC TTG GAC TCA CGA GGC AAC CGC TCC CCG GGG 
                TGA GAA CGG GGG ACT CAT TCC TCC GGC ACT GGG AGA AGA CGA TTC TTC AGG AGG AGG ACA GGG 
                AAG CGA ATG CTA CCC AGA TGT CTC AGT ATA CTG GCT CGC GGC ACA TCG GGC AAA TGA ACC TAT 
                CAG ATA ACA ACG GCA GAT CAG ATG CCT GAG CAT TCA GAA GCA ACA GCT GTG GAG CCC CCG TGG 
                GTT CAG AAG GCC TGG TTC CCG TCT CCA GAA GCC TGG CTC TCC TCC CTC CTG GGC CCA CTA CTT 
                TGG CTT CTT GTT CCT ACG TAC AAG GAG TTG CGA AGA AGG CAA CTC TTC CCC TCC CTG AAG CCA
                AAG GAA TGA AAC AGA CTA GGG CGG GAG AGG TGG CCA TCC GTC ATT AGT TGC GGC CAT CA"
)
construct
```

<br></br>

#####Lentivirus production and infection

HEK293T cells were transfected with viral and expression constructs using Lipofectamine 2000 (Life Technologies), after which viral supernatants were harvested 48 and 72 hours post-transfection. Viral particles were concentrated using PEG-IT solution (Systems Biosciences) according to the manufacturer’s recommendations. HEK293T cells were used for virus titration and GFP expression was evaluated 72hrs post-infection via flow cytometry after which TU/ml was calculated. Cell lines were subsequently infected with the lentiviral particals using a multuplicity of infection = 1. After 48 hours, cells were placed under mycophenolic acid selection and the selection process was monitored using GFP expression via flow cytometry. When GFP expression was present in >90% of the cells, cells were frozen for downstream analysis.

<br></br>

#####QPCR

RNA was extracted using the RNeasy mini kit (Qiagen) and subsequently treated with DNase (Ambion Turbo DNA-free, Life Technologies). 500ng RNA was used for cDNA synthesis using MuMLV (Life Technologies) and a 1:1 mix of oligo(dT) and random nanomers. QPCR was carried out using KAPA 2G SYBRGreen (Kapa Biosystems) using the Applied Biosystems 7900HT machine with the cycling conditions: 95 °C for 3 min, 95 °C for 3 s, 60 °C for 30 s.
QPCR for miRNA expression analysis was performed according to the protocol for the TaqMan microRNA Assay kit (Life Technologies) with primer/probe sets purchased from the same company (TaqMan® MicroRNA Assay, hsa-miR-34a, human and Control miRNA Assay, RNU48, human) and the same cycling scheme as above.

<br></br>

######QPCR primers

```{r}
###Q-PCR primers
primers <- data.frame(
    name=c(
        "ß-actin Fwd",
        "ß-actin Rev",
        "miR34a_asF1",
        "miR34a_asR1"
        
    ),
    sequence=c(
        "AGGTCATCACCATTGGCAATGAG",
        "CTTTCGGGATGTCCACGTCA",
        "AGCGGCATCTCCTCCACCTGAAA",
        "TTGCCTCGTGAGTCCAAGGAGAAT"
        
    )
)
primers
```

<br></br>

####Results

```{r, message = FALSE}
url <- "stableLineExpression.rda"
invisible(repmis::source_data(file.path(dataUrl, url)))
data <- .

#remove Skov exp 1 due to variability in technical replicates
data <- filter(data, `Biological Replicate` != 1 | `Cell line` != "Skov3")

#Have to divide the analysis due to the presence of several house 
#keeping genes.
gois1 <- c("miR34a asRNA", "Actin")
gois2 <- c("miR34a", "RNU48")
data1 <- filter(data, gene %in% gois1)
data2 <- filter(data, gene %in% gois2)

#calculate dct
groups <- c("Genetic mod", "Biological Replicate", "Cell line", "gene")

data1 <- data1 %>%
  technicalMeans(., groups) %>%
  dct(., gois1, Actin, groups) %>%
  mutate(GOI = as.character(GOI))

data2 <- data2 %>%
  technicalMeans(., groups) %>%
  dct(., gois2, RNU48, groups) %>%
  mutate(GOI = as.character(GOI))

data <- bind_rows(data1, data2) %>%
  mutate(GOI = parse_factor(GOI, levels = c("miR34a asRNA", "miR34a")))

#calculate ddct
groups[groups == "gene"] <- "GOI"
logical <- tibble(
  `Biological Replicate` = pull(data, `Biological Replicate`) == 2,
  `Genetic mod` = pull(data, `Genetic mod`) == "mock"
)

data <- data %>%
  ddct(., logical, groups) %>%
  folds(.)

#calculate stats
stats <- calcStats(data, `Genetic mod`, "mock", groups) %>%
  pFormat(.)

#format for plotting
lev <- c("PC3\nmock", "PC3\nmiR34a asRNA", "Skov3\nmock", "Skov3\nmiR34a asRNA", "Saos2\nmock",   "Saos2\nmiR34a asRNA")

data <- data %>%
  mutate(Condition = paste(`Cell line`, `Genetic mod`, sep = "\n")) %>%
  mutate(Condition = parse_factor(Condition, levels = lev))

stats <- stats %>%
  mutate(Condition = paste(`Cell line`, `Genetic mod`, sep = "\n")) %>%
  mutate(Condition = parse_factor(Condition, levels = lev))
```

```{r, fig.align='center', fig.height=8, fig.width=12, eval=TRUE, message=FALSE, warning=FALSE}
#plot for rmarkdown
ggplot(data = NULL) +
  geom_violin(
    data = data,
    aes(
      x = Condition,
      y = log2fold,
      group = interaction(Condition, GOI)
    ),
    trim = TRUE,
    alpha = 0.5,
    fill = "grey",
    colour = NA,
    scale = "width",
    show.legend = FALSE
  ) +
  geom_dotplot(
    data = data, 
    aes(
      x = Condition,
      y = log2fold,
      fill = GOI,
      group = interaction(Condition, GOI)
    ),
    binwidth = 1/10,
    colour = NA,
    alpha = 0.5,
    position = position_dodge(width = 0.9), 
    binaxis = 'y', 
    stackdir = 'center',
    dotsize = 4,
    show.legend = TRUE
  ) +
  facet_grid(
    . ~ `Cell line`, 
    scales = "free", 
    space = "free"
  ) +
  geom_linerange(
    data = stats,
    aes(
      x = Condition,
      ymin = CI95l,
      ymax = CI95h,
      group = interaction(Condition, GOI)
    ),
    position = position_dodge(width = 0.9),
    colour = "gray20",
    show.legend = FALSE
  ) +
  geom_point(
    data = stats,
    aes(
      x = Condition,
      y = mean,
      group = interaction(Condition, GOI)
    ),
    colour = "gray20",
    shape = 95,
    size = 6,
    position = position_dodge(width = 0.9),
    show.legend = FALSE
  ) +
  geom_label(
    data = filter(stats, `Genetic mod` != "mock"),
    aes(
      x = Condition, 
      y = CI95h + 0.5, 
      label = pFormat, 
      group = interaction(Condition, GOI)
    ),
    fill = "white",
    label.size = 0,
    label.padding = unit(0.01, "lines"),
    position = position_dodge(width = 0.9),
    show.legend = FALSE,
    fontface = "bold",
    family = "Arial Unicode MS",
    size = 3
  ) +
  scale_y_continuous("\u0394\u0394Ct")+
  labs(
    x = "Cell line",
    title = "miR34a asRNA over-expressing cell lines",
    caption = 
            "miR34a asRNA and miR34a levels in PC3, Skov3, and Saos2 cell lines stably over-expressing miR34a asRNA compared to mock over-expressing control.\n\u0394\u0394ct = House keeping gene - Target gene - mock control. Error bars indicate 95% CI and black points represent the mean."
  )+
  theme_few() +
  scale_colour_ptol() +
  scale_fill_ptol() +
  theme(
    legend.position = "top",
    legend.title = element_text(size = 17),
    legend.text = element_text(size = 15),
    plot.title = element_text(
      hjust = 0.5,
      face = "bold",
      size = 20
    ),
    plot.caption = element_text(
      hjust = 0, 
      margin = margin(t = 15),
      family = "Arial Unicode MS"
    ),
    axis.title = element_text(size = 17),
    axis.title.x = element_text(margin = margin(t = 10)),
    axis.text = element_text(size = 12),
    strip.text.x = element_text(size = 17)
  ) +
  guides(
    fill = guide_legend(title = "Gene", override.aes = list(size = 4)),
    colour = FALSE
  )
```

<br></br>

####Conclusion

miR34a asRNA over-expression causes a concominant up-regulation of miR34a in all 3 cell lines. This indicates that miR34a asRNA concordantly regulates miR34a.

<br></br><br></br>


###Supplementarty Figure 2a (stable with HEK)

####Introduction
miR34a asRNA over-expressing cell lines, previously generated, showed a high overexpression level of miR34a asRNA despite a low multiplicity of infection. This is most likley due to the fact that all infected cell lines have practically no endogenous miR34a asRNA expression. To investigate the physiological relevance of the over-expression the miR34a asRNA levels were compared to the expression levels in HEK293t cells, which have a high endogenous miR34a expression level.

####Methods

#####miR34a asRNA cloning
RNA was harvested from U2OS cells using the RNeasy mini kit (Qiagen) and subsequently treated with DNase (Ambion Turbo DNA-free, Life Technologies). 500ng RNA was used for cDNA synthesis using MuMLV (Life Technologies) and a 1:1 mix of oligo(dT) and random nanomers. PCR was run using the miR34aAS_cloning_F4 and miR34aAS_cloning_Ex3_1 primers for 40 cycles at annealing temperature 58˚C and run on an agarose gel. Bands were excised and extracted using the QIAquick gel extraction kit (Qiagen). PCR products were cloned using the Strata Clone kit (Aligent) and sequenced (results below). The insert for the lentiviral construct (p-HIV-I2AG-U6) was subsequently sub-cloned from the sequenced plasmid using primers with PacI and NheI cleavage sites. The cloned transcript largley mirrors the GENCODE annotated transcript.

######Cloning primers
```{r}
library(printr)
primers <- data.frame(
    name=c(
        "miR34aAS_cloning_F4",
        "miR34aAS_cloning_Ex3_1"
    ),
    sequence=c(
        "ACGCGTCTCTCCAGCCCGGGAT",
        "AATGATGGCCGCAACTAATGACGG"
    )
)
primers
```

######Sequencing results
```{r}
construct <- data.frame(
    name="miR34a_asRNA",
    sequence="ACG CGT CTC TCC AGC CCG GGA TCC GGG GAG CTG GGC TGT CCC CAG ACC GAC GGG ACA GCG GCA
                TCT CCT CCA CCT GAA AAG GAA AGA GGA CCA GTT TGC AGG ACT CCG AAC TGG GCC CGC GAG ATC 
                TCC ACC TGC GCA AAA CGA AAG GGC GGA TTC TCC TTG GAC TCA CGA GGC AAC CGC TCC CCG GGG 
                TGA GAA CGG GGG ACT CAT TCC TCC GGC ACT GGG AGA AGA CGA TTC TTC AGG AGG AGG ACA GGG 
                AAG CGA ATG CTA CCC AGA TGT CTC AGT ATA CTG GCT CGC GGC ACA TCG GGC AAA TGA ACC TAT 
                CAG ATA ACA ACG GCA GAT CAG ATG CCT GAG CAT TCA GAA GCA ACA GCT GTG GAG CCC CCG TGG 
                GTT CAG AAG GCC TGG TTC CCG TCT CCA GAA GCC TGG CTC TCC TCC CTC CTG GGC CCA CTA CTT 
                TGG CTT CTT GTT CCT ACG TAC AAG GAG TTG CGA AGA AGG CAA CTC TTC CCC TCC CTG AAG CCA
                AAG GAA TGA AAC AGA CTA GGG CGG GAG AGG TGG CCA TCC GTC ATT AGT TGC GGC CAT CA"
)
construct
```

#####Lentivirus production and infection
HEK293T cells were transfected with viral and expression constructs using Lipofectamine 2000 (Life Technologies), after which viral supernatants were harvested 48 and 72 hours post-transfection. Viral particles were concentrated using PEG-IT solution (Systems Biosciences) according to the manufacturer’s recommendations. HEK293T cells were used for virus titration and GFP expression was evaluated 72hrs post-infection via flow cytometry after which TU/ml was calculated. Cell lines were subsequently infected with the lentiviral particals using a multuplicity of infection = 1. After 48 hours, cells were placed under mycophenolic acid selection and the selection process was monitored using GFP expression via flow cytometry. When GFP expression was present in >90% of the cells, cells were frozen for downstream analysis.

#####QPCR
RNA was extracted using the RNeasy mini kit (Qiagen) and subsequently treated with DNase (Ambion Turbo DNA-free, Life Technologies). 500ng RNA was used for cDNA synthesis using MuMLV (Life Technologies) and a 1:1 mix of oligo(dT) and random nanomers. QPCR was carried out using KAPA 2G SYBRGreen (Kapa Biosystems) using the Applied Biosystems 7900HT machine with the cycling conditions: 95 °C for 3 min, 95 °C for 3 s, 60 °C for 30 s.
QPCR for miRNA expression analysis was performed according to the protocol for the TaqMan microRNA Assay kit (Life Technologies) with primer/probe sets purchased from the same company (TaqMan® MicroRNA Assay, hsa-miR-34a, human and Control miRNA Assay, RNU48, human) and the same cycling scheme as above.

######Primers

```{r}
###Q-PCR primers
primers <- data.frame(
    name=c(
        "ß-actin Fwd",
        "ß-actin Rev",
        "miR34a_asF1",
        "miR34a_asR1"
        
    ),
    sequence=c(
        "AGGTCATCACCATTGGCAATGAG",
        "CTTTCGGGATGTCCACGTCA",
        "AGCGGCATCTCCTCCACCTGAAA",
        "TTGCCTCGTGAGTCCAAGGAGAAT"
        
    )
)
primers
```

####Results

```{r}
url <- "stableLineExpressionHEK.rda"
invisible(repmis::source_data(file.path(dataUrl, url)))
data <- .

#remove PC3 experiment 6, B-act mean Ct > 30, too high.
data <- filter(data, `Biological Replicate` != 6 | `Cell line` != "PC3")

#calculate dct
groups <- c("Condition", "Biological Replicate", "Cell line", "gene")
gois <- c("miR34a asRNA")

data <- technicalMeans(data, groups) %>%
  dct(., gois, Actin, groups)

#calculate ddct
groups[groups == "gene"] <- "GOI"
logical <- tibble(
  `Cell line` = data$`Cell line` == "HEK293t",
  Condition = data$Condition == "wt"
)

data <- data %>%
  ddct(., logical, groups) %>%
  folds(.)

#calculate stats
stats <- data %>%
  group_by(`Cell line`, Condition) %>%
  summarize(
    n = n(),
    mean = mean(log2fold),
    CI95l = t.test(log2fold)$conf.int[1],
    CI95h = t.test(log2fold)$conf.int[2]
  )
```

```{r, fig.align='center', fig.height=8, fig.width=12, eval=TRUE, message=FALSE, warning=FALSE}
ggplot(data = NULL)+
  geom_violin(
    data = data,
    aes(
      x = Condition,
      y = log2fold,
      fill = Condition
    ),
    scale = "width",
    trim = TRUE,
    alpha = 0.3,
    color = NA,
    show.legend = FALSE
  ) +
  geom_dotplot(
    data = data, 
    aes(
      x = Condition, 
      y = log2fold, 
      fill = Condition
    ),
    colour = NA,
    binwidth = 1/10,
    alpha = 0.5,
    position = position_dodge(width = 0.9), 
    binaxis='y', 
    stackdir='center',
    dotsize = 2,
    show.legend = TRUE
  ) +
  facet_grid(. ~ `Cell line`, scales = "free") +
  geom_point(
    data = stats,
    aes(
      x = Condition, 
      y = mean
    ),
    colour = "grey20",
    show.legend = FALSE
  ) +
  geom_linerange(
    data = stats,
    aes(
      x = Condition, 
      ymin = CI95l, 
      ymax = CI95h
    ),
    colour = "gray20",
    show.legend = FALSE
  ) +
  scale_y_continuous("\u0394\u0394Ct") +
  labs(
    x = "Cell line",
    title = "miR34a asRNA",
    caption = 
            "miR34a asRNA over-expression levels in PC3, Skov3, and Saos2 stable cell lines compared to HEK293t cells.\n\u0394\u0394ct = target gene - House keeping gene - HEK293t. Error bars indicate 95% CI and black points represent the mean."
  ) +
  theme_few() +
  scale_color_ptol() +
  scale_fill_ptol() +
  theme(
    plot.title = element_text(
      hjust = 0.5,
      face = "bold",
      size = 20
    ),
    plot.caption = element_text(
      hjust = 0, 
      margin = margin(t = 15)
    ),
    legend.position = "none",
    axis.title = element_text(size = 17),
    axis.title.x = element_text(margin = margin(t = 10)),
    axis.text = element_text(size = 15),
    strip.text.x = element_text(size = 15)
  )
```

####Conclusions
Results indicate that miR34a asRNA is over-expressed 13-22 fold greater that endogenous HEK293t levels. 

###Figure 3b

####Introduction

miR34a is known to regulate cell cycle and, specifically, induce G1-phase arrest. Due to the increase observed in miR34a expression levels in cell lines over-expressing miR34a asRNA, we investigated if the cell cycle was perturbed in these cells.

<br></br>

####Methods

All cell lines were cultured at 5% CO2 and 37° C with Skov3 and PC3 cells in RPMI (Hyclone) and 2 mM L-glutamine. All growth mediums were supplemented with 10% heat-inactivated FBS and 50 μg/ml of streptomycin and 50 μg/ml of penicillin. 1x10^5 cells per well in a 6-well plate and harvested after 24hrs. Cells were washed in PBS and fixed in 4% PFA at room temperature overnight. PFA was removed, and cells were resuspended in 95% EtOH. The samples were then rehydrated in distilled water, stained with DAPI and analyzed by flow cytometry on a LSRII (BD Biosciences) machine. Resulting data was further processed using Flowjo software.

<br></br>

####Results

```{r, message = FALSE}
url <- "stableLineCellCycle.rda"
invisible(repmis::source_data(file.path(dataUrl, url)))
data <- .

#convert the value (precent) to log2(fraction)
data <- mutate(data, value = log2(value / 100))

#qqplot shows that the data is reasonably normal. The shapiro test is also used below. Note that a p < 0.05 indicates non-normality.
#qqnorm(data$value)

#calculate stats
stats <- data %>%
  full_join(
    filter(data, condition == "mock"), 
    by = c("Biological Replicate", "Cell line", "phase"), 
    suffix = c("", ".y")
  ) %>%
  group_by(`Cell line`, condition, phase) %>%
  summarize(
    n = n(),
    mean = (2^mean(.data$value)),
    max = (2^max(.data$value)),
    CI95l = (2^t.test(.data$value)$conf.int[1]),
    CI95h = (2^t.test(.data$value)$conf.int[2]),
    pValue = t.test(.data$value, .data$value.y)$p.value,
    shapiroTest = shapiro.test(value)$p.value
  ) %>%
  ungroup() %>%
  pFormat(.)

#prepare for plotting
#change value back to percent
data <- data %>%
  mutate(value = 2^value)

#calculate start and end for horizontal plot lines
stats <- stats %>%
  group_by(phase, `Cell line`) %>%
  mutate(y = max(c(.data$CI95h, .data$max))) %>%
  ungroup() %>%
  arrange(`Cell line`, phase) %>%
  mutate(
    x = rep(sort(rep(1:3, 2)), 2) - 0.25,
    xend = rep(sort(rep(1:3, 2)), 2) + 0.25
  )
```

```{r, fig.align='center', fig.height=8, fig.width=12, eval=TRUE, message=FALSE, warning=FALSE}
#plot for rmarkdown
ggplot(data = NULL) +
  geom_violin(
    data = data,
    aes(
      x = phase, 
      y = value,
      group = interaction(phase, condition)
    ),
    scale = "width",
    alpha = 0.3,
    fill = "gray",
    weight = 10,
    trim = TRUE,
    color = NA,
    show.legend = FALSE
  ) +
  geom_dotplot(
    data = data, 
    aes(
      x = phase,
      y = value,
      fill = condition
    ),
    binwidth = 1/100,
    colour = NA,
    alpha = 0.5,
    position = position_dodge(width = 0.9), 
    binaxis = 'y', 
    stackdir = 'center',
    dotsize = 0.5,
    show.legend = TRUE
  ) +
  facet_grid(
    . ~ `Cell line`, 
    scale = "free"
  ) +
  geom_linerange(
    data = stats,
    aes(
      x = phase, 
      ymin = CI95l, 
      ymax = CI95h, 
      group = interaction(phase, condition)
    ),
    position = position_dodge(width = 0.9),
    colour = "grey25",
    show.legend = FALSE
  ) +
  geom_point(
    data = stats,
    aes(
      x = phase, 
      y = mean, 
      group = condition
    ), 
    colour = "black",
    position = position_dodge(width = 0.9),
    shape = 95,
    size = 6,
    show.legend = FALSE
  ) +
  geom_segment(
    data = stats,
    aes(
      x = x,
      y = y + 0.01, 
      xend = xend, 
      yend = y + 0.01
    ),
    colour = "grey43",
    show.legend = FALSE
  ) +
  geom_label(
    data = stats,
    aes(
      x = phase, 
      y = y + 0.02, 
      label = pFormat
    ),
    label.size = 0,
    label.padding = unit(0.01, "lines"),
    show.legend = FALSE,
    fill = "white",
    size = 3,
    family = "Arial Unicode MS"
  ) +
  labs(
    x = "Cell cycle phase",
    title = "Cell cycle",
    caption = 
            "Cell cycle analysis in miR34a asRNA over-expressing PC3, Skov3, and Saos2 stable cell lines compared to their respective mock control.\nError bars (black vertical) indicate 95% CI and horizontal lines (black) represent the mean."
  )+
  theme_few() +
  scale_y_continuous(labels = scales::percent_format())+
  scale_fill_ptol() +
  scale_colour_ptol(labels = c("PC3 mock", "PC3 miR34a asRNA")) +
  theme(
    plot.title = element_text(
      hjust = 0.5,
      face = "bold",
      size = 22
    ),
    plot.caption = element_text(
      hjust = 0, 
      margin = margin(t = 15),
      family = "Arial Unicode MS"
    ),
    legend.position = "top",
    axis.title = element_text(size = 17),
    axis.title.x = element_text(margin = margin(t = 10)),
    axis.text = element_text(size = 15),
    strip.text.x = element_text(size = 15),
    axis.title.y = element_blank(),
    legend.title = element_text(size = 17),
    legend.text = element_text(size = 15)
  ) + 
  guides(
    fill = guide_legend(title = "Cell line"),
    colour = FALSE
  ) 
```

<br></br>

####Conclusion

An increase in the percentage of cells in G1-phase can be seen in cell lines overexpressing miR34a asRNA. In addition, a concominant decrease of cells in G2-phase can be observed.

<br></br><br></br>

###Figure 3c

####Introduction

As p53 is a well-known regulator of senescence, growth, and apoptosis and has been shown to specifically important under conditions of cellular stress, such as starvation. Therefore, we investigated whether overexpression of miR34a AS affects growth rate under normal and starvation conditions by measuring changes in confluency over time.

<br></br>

####Methods

10^4 PC3 cells, either miR34a asRNA overexpressing or WT, were seeded in 96 well plates. After attachment (3-4h) media was replaced with either RPMI (Gibco, life technology) (supplemented with 2mM L-glutamine, 50ug/ml Penicillin-Streptomycin and 10% Fetal Calf Serum) or HBSS. Cells were incubated in Spark Multimode Microplate reader for 48 hours at 37°C with 5% CO2 in a humidity chamber. Confluency was measured every hour. 

<br></br>

####Results

```{r, message = FALSE}
url <- "growthStarvation.rda"
invisible(repmis::source_data(file.path(dataUrl, url)))
data <- .

#normalize to first time point for control (Mock)
.normFun <- function(
  data,
  Confluency,
  Treatment,
  Biological.Replicate,
  Technical.Replicate,
  Cell.line
){
  bool1 <- data$Treatment == Treatment
  bool2 <- data$`Biological Replicate` == Biological.Replicate
  bool3 <- data$`Technical Replicate` == Technical.Replicate
  bool4 <- data$Time == 0
  bool5 <- data$`Cell line` == Cell.line
  bool <- bool1 & bool2 & bool3 & bool4 & bool5
  Confluency / pull(data, Confluency)[bool]
}

data <- data %>%
  group_by(Treatment, `Biological Replicate`, `Technical Replicate`, Time, `Cell line`) %>%
  mutate(
    normConfluency = .normFun(
      data,
      Confluency,
      Treatment,
      `Biological Replicate`,
      `Technical Replicate`,
      `Cell line`
    )
  ) %>%
  ungroup()

#calculate mean for each technical replicate
data <- data %>%
  group_by(Treatment, `Biological Replicate`, Time, `Cell line`) %>%
  summarize(
    techMean = mean(normConfluency)
  ) %>%
  ungroup()

#calculate stats
stats <- data %>%
  filter(Time != 0) %>%
  group_by(`Cell line`, Treatment, Time) %>%
  summarize(
    min = min(techMean),
    max = max(techMean),
    mean = mean(techMean),
    median = median(techMean),
    CI95l = t.test(techMean)$conf.int[1],
    CI95h = t.test(techMean)$conf.int[2]
  )
```

####Results

```{r, fig.align='center', fig.height=8, fig.width=12, eval=TRUE, message=FALSE, warning=FALSE}
#plot for Rmarkdown
ggplot(data = NULL) + 
  geom_ribbon(
    data = stats,
    aes(
      x = Time,
      ymin = CI95l, 
      ymax = CI95h, 
      group = `Cell line`,
      fill = `Cell line`
    ), 
    alpha = 0.2,
    show.legend = FALSE
  ) +
  geom_linerange(
    data = stats,
    aes(
      x = Time,
      colour = `Cell line`,
      ymin = min,
      ymax = max
    ), 
    position = position_dodge(width = 1),
    show.legend = FALSE
  ) + 
  geom_point(
    data = stats,
    aes(
      x = Time, 
      y = median,
      group = `Cell line`,
      colour = `Cell line`
    ), 
    position = position_dodge(width = 1),
    show.legend = TRUE
  ) +
  facet_grid(. ~ Treatment) +
  scale_x_discrete(breaks = seq(0, 47, 5)) +
  theme_few() +
  scale_fill_ptol() +
  scale_colour_ptol() +
  theme(
    plot.title = element_text(
      hjust = 0.5,
      face = "bold",
      size = 20
    ),
    plot.caption = element_text(
      hjust = 0, 
      margin = margin(t = 15),
      family = "Arial Unicode MS"
    ),
    legend.position = "top",
    legend.title = element_text(size = 17),
    legend.text = element_text(size = 15),
    axis.title = element_text(size = 17),
    axis.title.x = element_text(margin = margin(t = 10)),
    axis.text = element_text(size = 15),
    strip.text.x = element_text(size = 17)
  ) +
  labs(
    x = "Time (hours)", 
    y = "Normalized % confluency",
    title = "miR34a asRNA's role in growth regulation.",
    caption = "The effects of miR34a asRNA overexpression on growth in normal and starvation conditions in the PC3 prostate cancer cell line."
  ) +
  guides(
    fill = guide_legend(title = "Cell line"), 
    colour = guide_legend(title = "Cell line")
  )
```

<br></br>

####Conclusions

miR34a asRNA overexpression causes a minor decrease in cell growth under normal conditions (RPMI) although under conditions of cellular stress through starvation, this effect is increased dramatically. 

<br></br><br></br>

###Figure 3d

####Introduction

miR34a asRNA over-expression was shown to increase the expression levels of miR34a. We investigated if this increase of miR34a was due to increased levels of transcription at the miR34a promoter.

<br></br>

####Methods

All cell lines were cultured at 5% CO2 and 37° C with Skov3 and PC3 cells in RPMI (Hyclone) and 2 mM L-glutamine and Saos2 cells cultured in DMEM high modified (Hyclone). All growth mediums were supplemented with 10% heat-inactivated FBS and 50 μg/ml of streptomycin and 50 μg/ml of penicillin. Cells were plated at 2-3x10^6 cells in a 15cm dish and incubated overnight. Cells were cross-linked in 1% formaldehyde, quenched in glycine (0.125M), resuspended in cell lysis buffer (5mM PIPES, 85mM KCL, 0.5% NP40, protease inhibitor), and sonicated (Bioruptor Sonicator Diagenode) in nucleus lysis buffer (50mM TRIS-HCL pH 8.0, 10mM EDTA, 1% SDS, protease inhibitor). The samples were incubated overnight at 4 °C with the appropriate antibody. Salmon sperm DNA/Protein A–agarose (Upstate/Millipore) was used to pull down the antibody. DNA was eluted in elution buffer (1% SDS, 100 mM NaHCO3), followed by reverse cross-linking, RNase-A and protease-K treatment. The DNA was eluted by using a Qiagen PCR purification kit. The following antibodies were used (4 μg/sample): phospho-polymerase II (abcam: ab5095). QPCR was carried out using KAPA 2G SYBRGreen (Kapa Biosystems) using the Applied Biosystems 7900HT machine with the cycling conditions: 95 °C for 3 min, 95 °C for 3 s, 60 °C for 30 s.

<br></br>

#####Primers

```{r}
primers <- data.frame(
    name=c(
        "miR34aChIP F1",
        "miR34aChIP R1"
    ),
    sequence=c(
       "AAA GTT TGC AAA GAA GGA GGC GGG",
       "AGG GAA GAA AGA ACT AGC CGA GCA"
    )
)
```

<br></br>

####Results

```{r, message = FALSE}
url <- "stableLinePolIIChIP.rda"
invisible(repmis::source_data(file.path(dataUrl, url)))
data <- .

#calculate means of technical replicates
data <- mutate(data, techMean = rowMeans(select(data, Quantity1, Quantity2)))

#calculate fraction of input
.fractionOfInput <- function(data, techMean, Biological.Replicate, condition) {
  bool1 <- data$`Biological Replicate` == Biological.Replicate
  bool2 <- data$condition == condition
  bool3 <- data$sample == "input"
  bool <- bool1 & bool2 & bool3
  techMean / pull(data, techMean)[bool]
}

data <- data %>%
  group_by(`Biological Replicate`, condition, sample) %>%
  mutate(fractionInput = .fractionOfInput(data, techMean, `Biological Replicate`, condition)) %>%
  ungroup() %>%
  filter(sample != "input")

#calculate fold over mock
.foldOverControl <- function(data, fractionInput, Biological.Replicate) {
  bool1 <- data$`Biological Replicate` == 3
  bool2 <- data$condition == "mock"
  bool <- bool1 & bool2
  fractionInput / pull(data, fractionInput)[bool]
}

data <- data %>%
  mutate(
    foldOverControl = .foldOverControl(data, fractionInput, `Biological Replicate`),
    logfoldOverControl= log2(foldOverControl)
  ) %>%
  ungroup()

#calculate stats 
stats <- data %>%
  group_by(condition) %>%
  summarize(
    n = n(),
    mean = mean(logfoldOverControl),
    CI95l =t.test(logfoldOverControl)$conf.int[1],
    CI95h =t.test(logfoldOverControl)$conf.int[2],
    pValue = t.test(logfoldOverControl, mu = 0)$p.value
  ) %>%
  pFormat(.) %>%
  mutate(pFormat = case_when(
    condition == "mock" ~ "", 
    TRUE ~ pFormat
  ))
```

```{r, fig.align='center', fig.height=8, fig.width=12, eval=TRUE, message=FALSE, warning=FALSE}
#plot for Rmarkdown
ggplot(data = NULL)+
  geom_violin(
    data = data,
    aes(
      x = condition,
      y = logfoldOverControl,
      group = condition
    ),
    trim = TRUE,
    alpha = 0.3,
    colour = NA,
    fill = "gray",
    show.legend = FALSE
  ) +
  geom_dotplot(
    data = data, 
    aes(
      x = condition, 
      y = logfoldOverControl, 
      fill = condition
    ),
    binwidth = 1/10,
    alpha = 0.5,
    colour = NA,
    position = position_dodge(width = 0.9), 
    binaxis='y', 
    stackdir='center',
    dotsize = 2,
    show.legend = TRUE
  ) +
  geom_linerange(
    data = stats,
    aes(x = condition, ymin = CI95l, ymax = CI95h),
    colour = "gray20",
    show.legend = FALSE
  ) +
  geom_point(
    data = stats,
    aes(x = condition, y = mean),
    colour = "gray20",
    shape = 95,
    size = 6,
    show.legend = FALSE
  ) +
  geom_segment(
    data = filter(stats, condition == "miR34a asRNA"),
    aes(x = 1, y = CI95h + 0.25, xend = 2, yend = CI95h + 0.25),
    colour = "grey43"
  ) +
  geom_label(
    data = stats,
    aes(x = 1.5, y = CI95h + 0.45, label = pFormat),
    label.size = 0,
    label.padding = unit(0.01, "lines"),
    show.legend = FALSE,
    fill = "white",
    size = 5,
    family = "Arial Unicode MS"
  ) +
  scale_y_continuous("log2(Fold)") +
  labs(
    x = "Cell line",
    y = "log2(Fold polII enrichment)",
    title = "Phosphorylated Poll II ChIP",
    caption = 
            "Phosphorylated Poll II ChIP in miR34a asRNA over-expressing PC3 stable cell lines compared to their respective mock control.\nError bars (black vertical) indicate 95% CI and black points represent the mean."
  )+
  theme_few() +
  scale_fill_ptol() +
  scale_color_ptol() +
  theme(
    plot.title = element_text(
      hjust = 0.5,
      face = "bold",
      size = 20
    ),
    plot.caption = element_text(
      hjust = 0, 
      margin = margin(t = 15),
      family = "Arial Unicode MS"
    ),
    legend.position = "none",
    axis.title = element_text(size = 17),
    axis.title.x = element_text(margin = margin(t = 10)),
    axis.text = element_text(size = 15)
  )
```

####Conclusion

Phosphorylated pol-II was increased at the miR34a promoter in miR34a asRNA over-expressing cell lines indicating that the regulation of miR34a by miR34a asRNA is transcriptional.

<br></br><br></br>


###Supplementarty Figure 2b (CCND1 exp)

<br></br><br></br>

Code

<br></br>

Description

<br></br><br></br>

###Supplementarty Figure 2c (CCND1 prot)

<br></br><br></br>

Code

<br></br>

Description

<br></br><br></br>

##Session info

```{r}
sessionInfo()
```
