---
title: "Vignette Title"
author: "Jason T. Serviss"
date: "`r Sys.Date()`"
output:
  html_document:
    theme: flatly
    toc: no
    code_folding: hide
vignette: >
  %\VignetteIndexEntry{Vignette Title}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, echo = FALSE}
library(printr)
library(RCurl)
```

### Figure 1a
####Introduction
An antisense transcript to the miR34a HG was detected using the UCSC genome browser.
<br></br><br></br>

####Methods
The USCS genome browser was utilized for the bioinformatic evaluation of antisense transcription utilizing the Human EST and ENCODE RNAseq tracks.
<br></br><br></br>

####Results
```{r, out.width = "850px"}
projectUrl <- "https://github.com/GranderLab/miR34a_asRNA_project/raw/master/inst/"
url <- "miR34aLocus/miR34aLocus_hg38.png"
knitr::include_graphics(file.path(projectUrl, url))
```
<br></br>
A schematic picture of the miR34a locus from UCSC genome browser (hg38) including miR34a HG and mature miR34a, and LINC01759. H3K4me3 ChIP-seq data and conservation over the locus is also shown.
<br></br><br></br>

####Conclusion
GENCODE annotation supports the presence of a transcript orinionating in the antisense orientation from the miR34a gene. 

###Figure 1b
####Introduction
We wished to examine expression in different tissue types and, therefore, utilized a panel of common cell lines to detect miR34a asRNA expression, as well as, miR34a HG. We included both p53+/+ and p53-/- cell lines in the cell line panel due to the fact that miR34a is a known downstream target of p53. 
<br></br><br></br>

####Methods
Cells were cultured with the appropriate growth medium and seeded at a desity between 50-75% confluency. The following day, RNA was extracted using the RNeasy mini kit (Qiagen) and subsequently treated with DNase (Ambion Turbo DNA-free, Life Technologies). 500ng RNA was used for cDNA synthesis using MuMLV (Life Technologies) and a 1:1 mix of oligo(dT) and random nanomers. PCR was run for 35 or 38 or 16 cycles for miR34a asRNA, miR34a HG, and beta-actin, respectivley, with 58˚C annealing temperature. PCR products were analyzed on a 2% agarose gel.
<br></br><br></br>

#####Primers

```{r}
primers <- data.frame(
    name=c(
        "miR34a asRNA F1",
        "miR34a asRNA R1",
        "miR34aHG_F",
        "miR34aHG_R",
        "B-actin_F",
        "B-actin_R"
    ),
    sequence=c(
        "AGC GGC ATC TCC TCC ACC TGA AA",
        "TTG CCT CGT GAG TCC AAG GAG AAT",
        "TCT GCT CCA GTG GCT GAT GAG AAA",
        "GTT CAC TGG CCT CAA AGT TGG CAT",
        "AGG TCA TCA CCA TTG GCA ATG AG",
        "CTT TGC GGA TGT CCA CGT CA"
    )
)
primers
```

####Results
```{r, out.width = "850px"}
projectUrl <- "https://github.com/GranderLab/miR34a_asRNA_project/raw/master/inst/"
url <- "miR34aASExpressionCellPanel/cellPanelLarge.png"
knitr::include_graphics(file.path(projectUrl, url))
```
<br></br>
Semi-quantitative PCR data from the screening of a panel of cell lines. 
* Indicates that, although the cell line is p53wt, other mechanisms are present which inhibit p53 function. 
<br></br><br></br>

####Conclusions
We were able to detect co-expression of both miR34a HG and miR34a AS in a variety of cell lines, including HCT116 human colon cancer cells and 293T human embryonic kidney cells. A correlation was observed between both miR34a HG and miR34a asRNA expression and p53 status, with p53+/+ cell lines tending to have higher expression of both transcripts.

###Figure 1c

#Insert code here from Erik and write Intro Methods etc

####Results
```{r, out.width = "850px"}
projectUrl <- "https://github.com/GranderLab/miR34a_asRNA_project/raw/master/inst/"
url <- "TCGAcorrelation/miR34a_RP3-510D11.2_diploid_only_edited.png"
knitr::include_graphics(file.path(projectUrl, url))
```
<br></br>

###Figure 1d
####Introduction
In order to define the 3' transcription stop site for the miR34a asRNA, 3'-RACE was performed.
<br></br><br></br>

####Methods
All cell lines were cultured at 5% CO2 and 37° C with U2OS cells grown in McCoy’s 5a (Life Technologies). All growth mediums were supplemented with 10% heat-inactivated FBS and 50 μg/ml of streptomycin and 50 μg/ml of penicillin. U2OS cell RNA was polyA-tailed using yeast polyA polymerase after which cDNA was synthesized using oligo(dT) primers. Nested-PCR was performed first using a forward primer in miR34a asRNA exon 1 (miR-34a asRNA F10 forward primer - ACGCGTCTCTCCAGCCCGGGAT) and a tailed oligo(dT) primer (poly T T7-FAM) followed by a second PCR using an alternate miR34a asRNA exon 1 primer (miR-34a asRNA F1 forward primer - AGCGGCATCTCCTCCACCTGAAA) and a reverse primer binding to the tail of the previously used oligo(dT) primer (FAM primer). PCR products were gel purified and cloned the Strata Clone Kit (Agilent Technologies), and sequenced.
<br></br><br></br>

####Results
```{r, out.width = "850px"}
projectUrl <- "https://github.com/GranderLab/miR34a_asRNA_project/raw/master/inst/"
url <- "3primeRACE/results_hg38.png"
knitr::include_graphics(file.path(projectUrl, url))
```
<br></br>
3’-RACE sequencing results displayed in the UCSC genome browser together with the annotated miR34a asRNA transcript from GENCODE.
<br></br><br></br>

####Conclusions
Sequencing of the resulting cloned cDNA indicated the transcripts 3’ transcription termination site to be 125bp upstream of the RP3-510D11.2 transcript’s annotated termination site.

###Figure 1e
####Introduction
We desired to determine the 5'-start site for the miR34a asRNA.
<br></br><br></br>

####Methods
To accomplish this we performed the primer walk assay. All cell lines were cultured at 5% CO2 and 37° C with HEK293T cells cultured in DMEM high glucose (Hyclone). All growth mediums were supplemented with 10% heat-inactivated FBS and 50 μg/ml of streptomycin and 50 μg/ml of penicillin. RNA was extracted using the RNeasy mini kit (Qiagen) and subsequently treated with DNase (Ambion Turbo DNA-free, Life Technologies). 500ng RNA was used for cDNA synthesis using MuMLV (Life Technologies) and a 1:1 mix of oligo(dT) and random nanomers. PCR was performed for 35 cycles at 62°C annealing temperature and the resulting products were analyzed on a 2% agarose gel.
<br></br><br></br>

#####Primers
```{r}
primers <- data.frame(
    name=c(
        "miR34aAS_F10",
        "miR34aAS_F11",
        "miR34aAS_F12",
        "miR34aAS_F13",
        "miR34aAS_F14",
        "miR34aAS_F15",
        "miR34aAS_R1"
    ),
    sequence=c(
        "ACG CGT CTC TCC AGC CCG GGA T",
        "ATC TGC GTG GTC ACC GAG AAG CA",
        "CGC ACG GAC TGA GAA ACA CAA G",
        "ACG GAG GCT ACA CAA TTG AAC AGG",
        "AGG GAA GAA AGA ACT AGC CGA GCA",
        "CAT TTG CTG CAA TAT CAC CGT GGC",
        "TTG CCT CGT GAG TCC AAG GAG AAT"
    )
)
primers
```

<br></br><br></br>

####Results

```{r, out.width = "850px"}
projectUrl <- "https://github.com/GranderLab/miR34a_asRNA_project/raw/master/inst/"
url <- "primerWalk/schematic_hg38.png"
knitr::include_graphics(file.path(projectUrl, url))
```
<br></br>

A schematic picture of the miR34a asRNA locus including the location of the primers used in the primer walk assay.
<br></br>

```{r, out.width = "500px"}
projectUrl <- "https://github.com/GranderLab/miR34a_asRNA_project/raw/master/inst/"
url <- "primerWalk/gel.png"
knitr::include_graphics(file.path(projectUrl, url))
```
<br></br>
The semi-quantitative PCR results from the primer walk assay performed using HEK293T cells.
<br></br><br></br>

####Conclusion
It appears as though the miR34a asRNA transcript start site is somewhat upstream of the annotated start site. Although there is some product present for the F12 primer, the majority of transcription seems to be originating from the F11 primer. This indicates that the start site is between 71-224bp upstream of the annotated start site.

###Figure 1f and supplementary figure ??????????????
####Introduction
We utilized a bioinformatic approach to evaluate the coding potential of the miR34a asRNA transcript. The Coding-potential assessment tool uses a linear regression model to evaluate coding-potential by examining ORF length, ORF coverage, Fickett score and hexamer score. We further confirmed these results using the Coding-potential Calculator which utilizes a support based machine-based classifier and accesses an alternate set of discriminatory features.
<br></br><br></br>

####Methods
Protein-coding capacity was evaluated using the [Coding-potential Assessment Tool](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3616698/) and [Coding-potential Calculator](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1933232/) with default settings. Transcript sequences for use with Coding-potential Assessment Tool were downloaded from the UCSC genome browser using the following IDs: HOTAIR (ENST00000455246.1), XIST (ENST00000429829.1), β-actin (ENST00000331789.5), Tubulin (ENST00000427480.1), and MYC (ENST00000377970). Transcript sequences for use with Coding-potential Calculator were downloaded from the UCSC genome browser using the following IDs: HOTAIR (uc031qho.1), β-actin (uc003soq.4).
<br></br><br></br>

####Results
#####Coding Potential Assesment Tool


```{r}
projectUrl <- "https://github.com/GranderLab/miR34a_asRNA_project/raw/master/inst/"
url <- "codingPotential/CodingPotentialAssessmentTool/results.txt"
cpat <-read.table(url(file.path(projectUrl, url)), header = TRUE, sep = "\t")
cpat
```

<br></br>

Coding potential analysis results from the Coding-potential Assessment Tool including miR34a asRNA and two characterized non-coding transcripts (HOTAIR and XIST) and 3 known coding transcripts (β-actin, tubulin, and MYC). 

#####Coding Potential Calculator
```{r}
projectUrl <- "https://github.com/GranderLab/miR34a_asRNA_project/raw/master/inst/"
url <- "codingPotential/CodingPotentialCalculator/results.txt"
cpc <-read.table(url(file.path(projectUrl, url)), header = TRUE, sep = "\t")
cpc
```

<br></br>

Coding potential analysis results using the Coding-potential calculator for miR34a asRNA, HOTAIR, and β-actin.

<br></br><br></br>

####Conclusions
Results indicated that miR34a asRNA has a similar lack of coding capacity to the known non-coding transcripts HOTAIR and XIST and differs greatly when examining these parameters to the known coding transcripts beta-actin, tubulin, and MYC. However, to fully evaluate coding potential methods such as mass spectrometry or ribosome profiling must be used. 

```{r}
sessionInfo()
```
