---
title: "Vignette Title"
author: "Jason T. Serviss"
date: "`r Sys.Date()`"
output:
  html_document:
    theme: flatly
    toc: yes
    code_folding: hide
vignette: >
  %\VignetteIndexEntry{Vignette Title}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, echo = FALSE, message = FALSE}
library(printr)
library(RCurl)
library(miR34AasRNAproject)
library(tidyverse)
library(ggthemes)
```

```{r}
projectUrl <- "https://github.com/GranderLab/miR34a_asRNA_project/raw/master/inst/"
dataUrl <- "https://github.com/GranderLab/miR34a_asRNA_project/raw/master/data"
```

### Figure 1a
####Introduction
An antisense transcript to the miR34a HG was detected using the UCSC genome browser.
<br></br><br></br>

####Methods
The USCS genome browser was utilized for the bioinformatic evaluation of antisense transcription utilizing the Human EST and ENCODE RNAseq tracks.
<br></br><br></br>

####Results
```{r, out.width = "850px"}
url <- "miR34aLocus/miR34aLocus_hg38.png"
knitr::include_graphics(file.path(projectUrl, url))
```
<br></br>
A schematic picture of the miR34a locus from UCSC genome browser (hg38) including miR34a HG and mature miR34a, and LINC01759. H3K4me3 ChIP-seq data and conservation over the locus is also shown.
<br></br><br></br>

####Conclusion
GENCODE annotation supports the presence of a transcript orinionating in the antisense orientation from the miR34a gene. 

###Figure 1b
####Introduction
We wished to examine expression in different tissue types and, therefore, utilized a panel of common cell lines to detect miR34a asRNA expression, as well as, miR34a HG. We included both p53+/+ and p53-/- cell lines in the cell line panel due to the fact that miR34a is a known downstream target of p53. 
<br></br><br></br>

####Methods
Cells were cultured with the appropriate growth medium and seeded at a desity between 50-75% confluency. The following day, RNA was extracted using the RNeasy mini kit (Qiagen) and subsequently treated with DNase (Ambion Turbo DNA-free, Life Technologies). 500ng RNA was used for cDNA synthesis using MuMLV (Life Technologies) and a 1:1 mix of oligo(dT) and random nanomers. PCR was run for 35 or 38 or 16 cycles for miR34a asRNA, miR34a HG, and beta-actin, respectivley, with 58˚C annealing temperature. PCR products were analyzed on a 2% agarose gel.
<br></br><br></br>

#####Primers

```{r}
primers <- data.frame(
    name=c(
        "miR34a asRNA F1",
        "miR34a asRNA R1",
        "miR34aHG_F",
        "miR34aHG_R",
        "B-actin_F",
        "B-actin_R"
    ),
    sequence=c(
        "AGC GGC ATC TCC TCC ACC TGA AA",
        "TTG CCT CGT GAG TCC AAG GAG AAT",
        "TCT GCT CCA GTG GCT GAT GAG AAA",
        "GTT CAC TGG CCT CAA AGT TGG CAT",
        "AGG TCA TCA CCA TTG GCA ATG AG",
        "CTT TGC GGA TGT CCA CGT CA"
    )
)
primers
```

####Results
```{r, out.width = "850px"}
url <- "miR34aASExpressionCellPanel/cellPanelLarge.png"
knitr::include_graphics(file.path(projectUrl, url))
```
<br></br>
Semi-quantitative PCR data from the screening of a panel of cell lines. 
* Indicates that, although the cell line is p53wt, other mechanisms are present which inhibit p53 function. 
<br></br><br></br>

####Conclusions
We were able to detect co-expression of both miR34a HG and miR34a AS in a variety of cell lines, including HCT116 human colon cancer cells and 293T human embryonic kidney cells. A correlation was observed between both miR34a HG and miR34a asRNA expression and p53 status, with p53+/+ cell lines tending to have higher expression of both transcripts.

###Figure 1c

####Insert code here from Erik and write Intro Methods etc

####Results
```{r, out.width = "850px"}
url <- "TCGAcorrelation/miR34a_RP3-510D11.2_diploid_only_edited.png"
knitr::include_graphics(file.path(projectUrl, url))
```
<br></br>

###Figure 1d
####Introduction
In order to define the 3' transcription stop site for the miR34a asRNA, 3'-RACE was performed.
<br></br><br></br>

####Methods
All cell lines were cultured at 5% CO2 and 37° C with U2OS cells grown in McCoy’s 5a (Life Technologies). All growth mediums were supplemented with 10% heat-inactivated FBS and 50 μg/ml of streptomycin and 50 μg/ml of penicillin. U2OS cell RNA was polyA-tailed using yeast polyA polymerase after which cDNA was synthesized using oligo(dT) primers. Nested-PCR was performed first using a forward primer in miR34a asRNA exon 1 (miR-34a asRNA F10 forward primer - ACGCGTCTCTCCAGCCCGGGAT) and a tailed oligo(dT) primer (poly T T7-FAM) followed by a second PCR using an alternate miR34a asRNA exon 1 primer (miR-34a asRNA F1 forward primer - AGCGGCATCTCCTCCACCTGAAA) and a reverse primer binding to the tail of the previously used oligo(dT) primer (FAM primer). PCR products were gel purified and cloned the Strata Clone Kit (Agilent Technologies), and sequenced.
<br></br><br></br>

####Results
```{r, out.width = "850px"}
url <- "3primeRACE/results_hg38.png"
knitr::include_graphics(file.path(projectUrl, url))
```
<br></br>
3’-RACE sequencing results displayed in the UCSC genome browser together with the annotated miR34a asRNA transcript from GENCODE.
<br></br><br></br>

####Conclusions
Sequencing of the resulting cloned cDNA indicated the transcripts 3’ transcription termination site to be 125bp upstream of the RP3-510D11.2 transcript’s annotated termination site.

###Figure 1e
####Introduction
We desired to determine the 5'-start site for the miR34a asRNA.
<br></br><br></br>

####Methods
To accomplish this we performed the primer walk assay. All cell lines were cultured at 5% CO2 and 37° C with HEK293T cells cultured in DMEM high glucose (Hyclone). All growth mediums were supplemented with 10% heat-inactivated FBS and 50 μg/ml of streptomycin and 50 μg/ml of penicillin. RNA was extracted using the RNeasy mini kit (Qiagen) and subsequently treated with DNase (Ambion Turbo DNA-free, Life Technologies). 500ng RNA was used for cDNA synthesis using MuMLV (Life Technologies) and a 1:1 mix of oligo(dT) and random nanomers. PCR was performed for 35 cycles at 62°C annealing temperature and the resulting products were analyzed on a 2% agarose gel.
<br></br><br></br>

#####Primers
```{r}
primers <- data.frame(
    name=c(
        "miR34aAS_F10",
        "miR34aAS_F11",
        "miR34aAS_F12",
        "miR34aAS_F13",
        "miR34aAS_F14",
        "miR34aAS_F15",
        "miR34aAS_R1"
    ),
    sequence=c(
        "ACG CGT CTC TCC AGC CCG GGA T",
        "ATC TGC GTG GTC ACC GAG AAG CA",
        "CGC ACG GAC TGA GAA ACA CAA G",
        "ACG GAG GCT ACA CAA TTG AAC AGG",
        "AGG GAA GAA AGA ACT AGC CGA GCA",
        "CAT TTG CTG CAA TAT CAC CGT GGC",
        "TTG CCT CGT GAG TCC AAG GAG AAT"
    )
)
primers
```

<br></br><br></br>

####Results

```{r, out.width = "850px"}
url <- "primerWalk/schematic_hg38.png"
knitr::include_graphics(file.path(projectUrl, url))
```
<br></br>

A schematic picture of the miR34a asRNA locus including the location of the primers used in the primer walk assay.
<br></br>

```{r, out.width = "500px"}
url <- "primerWalk/gel.png"
knitr::include_graphics(file.path(projectUrl, url))
```
<br></br>
The semi-quantitative PCR results from the primer walk assay performed using HEK293T cells.
<br></br><br></br>

####Conclusion
It appears as though the miR34a asRNA transcript start site is somewhat upstream of the annotated start site. Although there is some product present for the F12 primer, the majority of transcription seems to be originating from the F11 primer. This indicates that the start site is between 71-224bp upstream of the annotated start site.

###Figure 1f and supplementary figure ??????????????
####Introduction
We utilized a bioinformatic approach to evaluate the coding potential of the miR34a asRNA transcript. The Coding-potential assessment tool uses a linear regression model to evaluate coding-potential by examining ORF length, ORF coverage, Fickett score and hexamer score. We further confirmed these results using the Coding-potential Calculator which utilizes a support based machine-based classifier and accesses an alternate set of discriminatory features.
<br></br><br></br>

####Methods
Protein-coding capacity was evaluated using the [Coding-potential Assessment Tool](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3616698/) and [Coding-potential Calculator](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1933232/) with default settings. Transcript sequences for use with Coding-potential Assessment Tool were downloaded from the UCSC genome browser using the following IDs: HOTAIR (ENST00000455246.1), XIST (ENST00000429829.1), β-actin (ENST00000331789.5), Tubulin (ENST00000427480.1), and MYC (ENST00000377970). Transcript sequences for use with Coding-potential Calculator were downloaded from the UCSC genome browser using the following IDs: HOTAIR (uc031qho.1), β-actin (uc003soq.4).
<br></br><br></br>

####Results
#####Coding Potential Assesment Tool


```{r}
url <- "codingPotential/CodingPotentialAssessmentTool/results.txt"
cpat <-read.table(url(file.path(projectUrl, url)), header = TRUE, sep = "\t")
cpat
```

<br></br>

Coding potential analysis results from the Coding-potential Assessment Tool including miR34a asRNA and two characterized non-coding transcripts (HOTAIR and XIST) and 3 known coding transcripts (β-actin, tubulin, and MYC). 

#####Coding Potential Calculator
```{r}
url <- "codingPotential/CodingPotentialCalculator/results.txt"
cpc <- read.table(url(file.path(projectUrl, url)), header = TRUE, sep = "\t")
cpc
```

<br></br>

Coding potential analysis results using the Coding-potential calculator for miR34a asRNA, HOTAIR, and β-actin.

<br></br><br></br>

####Conclusions
Results indicated that miR34a asRNA has a similar lack of coding capacity to the known non-coding transcripts HOTAIR and XIST and differs greatly when examining these parameters to the known coding transcripts beta-actin, tubulin, and MYC. However, to fully evaluate coding potential methods such as mass spectrometry or ribosome profiling must be used. 

###Figure 2a

####Introduction
miR34a is a known downstream target of p53 and has been previously shown to exhibit increased expression with cellular apoptotic signaling. We hypothesized that miR34a asRNA may be regulated in a similar fashion whereby transcription is stimulated by activation of p53. To test this we treated p53+/+ and p53-/- HCT116 cells with increasing concentrations of the DNA damaging agent doxorubicin and monitored miR34a asRNA expression.
<br></br><br></br>

####Methods
HCT116 cells were cultured in DMEM high modified (Hyclone, GE healthcare) supplemented with supplemented with 2mM L-glutamine, 50ug/ml Penicillin-Streptomycin and 10% Fetal Calf Serum.  
25 x 10^5 HCT116 cells were plated in 6 well plates. 24 hours later media was exchanged and doxorubicin was added to a final concentration of 100, 200 or 500 ng/ml. Cells were harvested for RNA extraction 24 hours later using trypsin. RNA was extracted using Nucleospin RNA kit (Machery-Nagel Ref. 740955) according to manufacturer‟s protocol and DNase treated using Ambion Turbo DNA-free according to manufacturer‟s protocol (Life Technologies Ref. AM1907). cDNA was synthesized using ~500 ng RNA with M-MLV (Life Technologies Ref 28025013) and a mixture of oligo(dT) with nanomers in accordance with the manufacturer's protocol.
qPCR quantification was carried out using the PowerUp SYBR Green Master Mix (Thermo Fisher Scientific, Ref. A25777) on the CFX96 Touch Real-Time PCR Detection System: 50°C for 2 min, 95°C for 2min, and 95°C for 1 sec followed by 60°C for 30 sec repeated for 40 cycles. The following primers were used:
<br></br><br></br>

#####Primers
```{r}
primers <- data.frame(
    name=c(
        "ß-actin Fwd",
        "ß-actin Rev",
        "miR34a HG_F",
        "miR34a HG_R",
        "miR34a_asF1",
        "miR34a_asR1"
        
    ),
    sequence=c(
        "AGGTCATCACCATTGGCAATGAG",
        "CTTTCGGGATGTCCACGTCA",
        "TCTGCTCCAGTGGCTGATGAGAAA",
        "GTTCACTGGCCTCAAAGTTGGCAT",
        "AGCGGCATCTCCTCCACCTGAAA",
        "TTGCCTCGTGAGTCCAAGGAGAAT"
        
    )
)
primers
```
<br></br><br></br>

####Results

Perform ddct analysis and statistical testing.
```{r}
#download and format data
url <- "HCT116p53null.rda"
repmis::source_data(file.path(dataUrl, url))
data <- .

#calculate dct
groups <- c("Condition", "Treatment", "Biological Replicate", "Cell line", "gene")
gois <- c("miR34a asRNA", "miR34a HG")

data <- technicalMeans(data, groups) %>%
  dct(., gois, Actin, groups)

#calculate ddct
groups[groups == "gene"] <- "GOI"
logical <- tibble(
  Treatment = data$Treatment == 0
)

data <- data %>%
  ddct(., logical, groups) %>%
  folds(.)

#calculate stats
stats <- calcStats(data, Condition, "p53-wt", groups) %>%
  pFormat(.)

#setup plotting variables
stats <- stats %>%
  group_by(GOI, Treatment) %>%
  mutate(
    max = max(CI95h),
    x = case_when(
      Treatment ==   0 ~ 1 + 0.25,
      Treatment == 100 ~ 2 + 0.25,
      Treatment == 200 ~ 3 + 0.25,
      Treatment == 500 ~ 4 + 0.25
    ),
    xend = case_when(
      Treatment ==   0 ~ 1 - 0.25,
      Treatment == 100 ~ 2 - 0.25,
      Treatment == 200 ~ 3 - 0.25,
      Treatment == 500 ~ 4 - 0.25
    )
  ) %>%
  ungroup()
```

Plot the results.
```{r, fig.align='center', fig.height=8, fig.width=12, eval=TRUE, message=FALSE, warning=FALSE}
#plot for Rmarkdown
ggplot(data = NULL) +
    geom_violin(
        data = data,
        aes(x = Treatment, y = log2fold, fill = Condition),
        trim = TRUE,
        alpha = 0.3,
        color = NA,
        scale = "width",
        show.legend = TRUE
    ) +
    geom_point(
        data = stats,
        aes(x = Treatment, y = mean, group = Condition), 
        position = position_dodge(width = 0.9), 
        size = 4
    ) +
    facet_grid(. ~ GOI) +
    geom_linerange(
        data = stats,
        aes(x = Treatment, ymin = CI95l, ymax = CI95h, group = Condition),
        colour = "black",
        position = position_dodge(width = 0.9),
        show.legend = FALSE
    ) +
    geom_label(
        data = stats,
        aes(x = Treatment, y = max + 0.22, label = pFormat),
        label.size = 0,
        label.padding = unit(0.01, "lines"),
        show.legend = FALSE,
        fill = "white",
        size = 5,
        family = "Arial Unicode MS",
        position = position_dodge(width = 0.9)
    ) +
    geom_segment(
        data = stats,
        aes(x = x, y = max + 0.1, xend = xend, yend = max + 0.1),
        colour="grey43",
        show.legend = FALSE
    ) +
    scale_y_continuous(
        "\u0394\u0394Ct",
        breaks=c(-1,0,1,2,3,4,5),
        sec.axis = sec_axis(
            ~2^., 
            name = expression(2^{- ~ Delta ~ Delta ~ "Ct"}), 
            breaks=c(2^-1,2^0,2^1,2^2,2^3,2^4,2^5)
        )
    )+
    labs(
      x="Treatment",
      title="miR34a asRNA regulation by p53 induction",
      caption=
            "Monitoring the effects of 24 hours doxorubicin treatment on miR34a asRNA and HG in P53+/+ and p53-/- HCT116 cells."
    )+
    theme_few()+
    scale_fill_ptol()+
    scale_colour_ptol()+
    #values=c("#014d64", "#6794a7")
    theme(
        plot.title = element_text(
            hjust = 0.5,
            face="bold",
            size=20
        ),
        plot.caption = element_text(
            hjust=0, 
            margin=margin(t=15),
            family="Arial Unicode MS"
        ),
        legend.position = "top",
        legend.title=element_text(size=17),
        legend.text=element_text(size=15),
        axis.title=element_text(size=17),
        axis.title.x=element_text(margin=margin(t=10)),
        axis.text=element_text(size=15),
        strip.text.x=element_text(size=17)
    )+
    guides(fill = guide_legend(title = "Cell line", override.aes = list(size=3)))
```
<br></br><br></br>

####Conclusions
We observed increasing miR34a HG and asRNA expression with increasing doses of doxorubicin indicating that these two transcripts are co-regulated by p53. Although p53+/+ cells showed a 25-fold increase over untreated cells in miR34a AS expression at the highest tested doxorubicin concentration, this effect was largely abrogated, 11 fold over untreated, in p53-/- cells, indicating that p53 is a major regulator of miR34a asRNA expression. 

###Figure 2b
####Introduction
miR34a is a known downstream target of p53 and has been previously shown to exhibit increased expression with cellular apoptotic signaling. We hypothesized that miR34a asRNA may be regulated in a similar fashion whereby transcription is stimulated by activation of p53. To test this we treated HCT116 and HEK293t cells with 200 ng/ml of the DNA damaging agent doxorubicin for 24 hours and monitored miR34a asRNA expression. 

####Methods
All cell lines were cultured at 5% CO2 and 37° C with HEK293T cells cultured in DMEM high glucose (Hyclone) and HCT116 cells in McCoy’s 5a (Life Technologies). All growth mediums were supplemented with 10% heat-inactivated FBS and 50 μg/ml of streptomycin and 50 μg/ml of penicillin. Cells were plated at 300,000 cells per well in a 6-well plate and cultured overnight. The following day cells were treated with 0, 100, 200, or 500 ng/ml doxorubicin for 24hrs. RNA was extracted using the RNeasy mini kit (Qiagen) and subsequently treated with DNase (Ambion Turbo DNA-free, Life Technologies). 500ng RNA was used for cDNA synthesis using MuMLV (Life Technologies) and a 1:1 mix of oligo(dT) and random nanomers. QPCR was carried out using KAPA 2G SYBRGreen (Kapa Biosystems) using the Applied Biosystems 7900HT machine with the cycling conditions: 95 °C for 3 min, 95 °C for 3 s, 60 °C for 30 s.

```{r}
primers <- data.frame(
    name=c(
        "ß-actin Fwd",
        "ß-actin Rev",
        "miR34a HG_F",
        "miR34a HG_R",
        "miR34a_asF1",
        "miR34a_asR1"
        
    ),
    sequence=c(
        "AGGTCATCACCATTGGCAATGAG",
        "CTTTCGGGATGTCCACGTCA",
        "TCTGCTCCAGTGGCTGATGAGAAA",
        "GTTCACTGGCCTCAAAGTTGGCAT",
        "AGCGGCATCTCCTCCACCTGAAA",
        "TTGCCTCGTGAGTCCAAGGAGAAT"
        
    )
)
primers
```

####Results
```{r}
url <- "HctHekDox.rda"
repmis::source_data(file.path(dataUrl, url))
data <- .

#calculate dct
groups <- c("Treatment", "Biological Replicate", "Cell line", "gene")
gois <- c("miR34a asRNA", "miR34a HG")

data <- technicalMeans(data, groups) %>%
  dct(., gois, Actin, groups)

#calculate ddct
groups[groups == "gene"] <- "GOI"
logical <- tibble(
  `Biological Replicate` = pull(data, `Biological Replicate`) == 1,
  Treatment = pull(data, Treatment) == "untreated"
)

data <- data %>%
  ddct(., logical, groups) %>%
  folds(.)

#calculate stats
stats <- calcStats(data, Treatment, "untreated", groups) %>%
  pFormat(.)
```

```{r, fig.align='center', fig.height=8, fig.width=12, message=FALSE, warning=FALSE}
#plot for Rmarkdown
ggplot(data = NULL) +
  geom_violin(
    data = data,
    aes(x = Treatment, y = log2fold, fill = GOI),
    trim = TRUE,
    alpha = 0.3,
    color = NA,
    scale = "width",
    show.legend = FALSE
  ) +
  geom_point(
    data = stats,
    aes(x = Treatment, y = mean, group = GOI),
    size = 4,
    position = position_dodge(width = 0.9)
  ) +
  facet_grid(. ~ `Cell line`) +
  geom_linerange(
    data = stats,
    aes(x = Treatment, ymin = CI95l, ymax = CI95h, group = GOI),
    colour = "black",
    position = position_dodge(width = 0.9),
    show.legend = FALSE
  ) +
  geom_label(
    data = stats,
    aes(x = Treatment, y = CI95h + 0.15, label = pFormat, group = GOI),
    label.size = 0,
    label.padding = unit(0.01, "lines"),
    show.legend = FALSE,
    fill = "white",
    size = 5,
    family = "Arial Unicode MS",
    position = position_dodge(width = 0.9)
  )+
  scale_y_continuous(
    "\u0394\u0394Ct",
    breaks=c(-1,0,1,2,3,4,5)
  )+
  labs(
    x = "Treatment",
    title = "miR34a asRNA response to p53 induction",
    caption = 
            "Q-PCR showing miR34a asRNA and miR34a HG levels in HCT116 and HEK293t cells after treatment with 200 ng/ml doxorubicin for 24hrs.\nError bars (black vertical) indicate 95% CI and means are represented \nby black dots."
  )+
  theme_few()+
  scale_fill_ptol()+
  scale_colour_ptol()+
  theme(
    plot.title = element_text(
      hjust = 0.5,
      face = "bold",
      size = 20
    ),
    plot.caption = element_text(
      hjust = 0, 
      margin = margin(t = 15),
      family = "Arial Unicode MS"
    ),
    legend.position = "top",
    legend.title = element_text(size = 17),
    legend.text = element_text(size = 15),
    axis.title = element_text(size = 17),
    axis.title.x = element_text(margin = margin(t = 10)),
    axis.text = element_text(size = 15),
    strip.text.x = element_text(size = 17)
  )+
  guides(color = guide_legend(title = "Gene", override.aes = list(size = 3)))
```

####Conclusions
We observed increases in miR34a HG and asRNA expression upon doxorubicin treatment indicating that these two transcripts are co-regulated. 

###Figure 2c

####Introduction
It is likely, due to the head-to head orientation of miR34a HG and asRNA, that transcription initiation may be activated from a single promoter in a bi-directional manner. To investigate whether miR34a HG and asRNA are transcribed from the same promoter as convergent transcripts, we cloned the miR34a HG promoter, including the p53 binding site, into a luciferase/renilla dual reporter vector which we hereafter refer to as p1. The p1 sequence was previously published in Raver-Shapira, N., et al., Transcriptional activation of miR-34a contributes to p53-mediated apoptosis. Mol Cell, 2007. 26(5): p. 731-43. Due to the observance of additional conservation sites downstream of the previously published miR34a HG promoter, we also cloned a longer "promoter region", extending 251 additional base pairs into the miR34a HG into the same bi-directional reporter vector, hereafter refered to as p2.

####Methods
All cell lines were cultured at 5% CO2 and 37° C with HEK293T cells cultured in DMEM high glucose (Hyclone) and HCT116 cells in McCoy’s 5a (Life Technologies). All growth mediums were supplemented with 10% heat-inactivated FBS and 50 μg/ml of streptomycin and 50 μg/ml of penicillin. Cells were plated at 10,000 cells per well in a 96-well plate with a white bottom and cultured overnight. The following day cells were co-transfected with 10ng of empty, p1, or p2 plasmid and GFP using the standard lipofektamine 2000 (Life Technologies) protocol. The expression of GFP and luminescence was measured 24 h post transfection by using the Dual-Glo Luciferase Assay System (Promega) and detected by the GloMax-Multi+ Detection System (Promega). The expression of luminescence was normalized to GFP.

#####Constructs
```{r}
library(printr)
ps <- data.frame(
    name=c("p1", "p2"),
    sequence=c("GCG CCC TGC CTG GCC CCC ACC TGG TCC TCT TTC CTT TTC AGG TGG AGG AGA TGC CGC TGT
              CCC GTC GGT CTG GGG ACA GCC CAG CTC CCC GGA TCC CGG GCT GGA GAG ACG CGT CGC GGC   
              CCC GGG GCC TGG TGG CAC GAG CAG GAA GGA GGA CCC GGC GGC GGG CTC TGC CTG GGC TTG 
              CCT GGG CTT GTT CCG AGC CGG GCT GCT TCT CGG TGA CCA CGC AGA TCG GGG GCA TTT GGA
              GAT TTT GCG GGA GTC CTG CAG CCA AGC TCC GGG GCA GGA GAG GCC TGG AAG CCT GCA CTA
              CCT GCT C",
              "GCG CCC TGC CTG GCC CCC ACC TGG TCC TCT TTC CTT TTC AGG TGG AGG AGA TGC CGC TGT 
              CCC GTC GGT CTG GGG ACA GCC CAG CTC CCC GGA TCC CGG GCT GGA GAG ACG CGT CGC GGC
              CCC GGG GCC TGG TGG CAC GAG CAG GAA GGA GGA CCC GGC GGC GGG CTC TGC CTG GGC TTG 
              CCT GGG CTT GTT CCG AGC CGG GCT GCT TCT CGG TGA CCA CGC AGA TCG GGG GCA TTT GGA 
              GAT TTT GCG GGA GTC CTG CAG CCA AGC TCC GGG GCA GGA GAG GCC TGG AAG CCT GCA CTA 
              CCT GCT CGC CCC GTC CCA GCA TGC ACC CAG GTA AAC GCT TGT GTT TCT CAG TCC GTG CGA 
              AAG TTT GCA AAG AAG GAG GCG GGA ACT AGA CCA ACA ACT TTA ATA ATC ATA GTC ATA ATA 
              ATG AAA ACC CTG TTC AAT TGT GTA GCC TCC GTA AGG GGA AGA ATT TCC AAG AAG CAA CAA
              AAT GTT CTT TGA TTT CAT AAT CTC TGG GGA GAA AGG ATT CCT TTT TTG GGG TTT TGC TCG
              GCT AGT TCT TTC TTC CCT")
)
ps
```

#####Primers
```{r}
primers <- data.frame(
    name=c(
        "Luc_set_II_F",
        "Luc_set_II_R",
        "Renilla_pBiDir_F1",
        "Renilla_pBiDir_R1",
        "B-actin_F",
        "B-actin_R"
    ),
    sequence=c(
        "AAG ATT CAA AGT GCG CTG CTG",
        "TTG CCT GAT ACC TGG CAG ATG",
        "TAA CGC GGC CTC TTC TTA TTT",
        "GAT TTG CCT GAT TTG CCC ATA",
        "AGG TCA TCA CCA TTG GCA ATG AG",
        "CTT TGC GGA TGT CCA CGT CA"
    )
)
primers
```

####Results
```{r, out.width = "850px"}
url <- "P1-HCTandHEK/locusSimple.png"
knitr::include_graphics(file.path(projectUrl, url))
```

<br></br><br></br>

```{r, out.width = "850px"}
url <- "P1-HCTandHEK/constructSimple.png"
knitr::include_graphics(file.path(projectUrl, url))
```

<br></br><br></br>

```{r}
data <- read.table('~/GitHub/miR34aFigures/inst/P1-HCTandHEK/luminesenseData.txt', header=TRUE, sep="\t")

summary <- ddply(
    data, 
    c("cellLine", "gene", "construct"),
    summarize,
    mean = mean(value),
    pValue = t.test(value, data[data$cellLine == cellLine & data$gene == gene & data$construct == "empty", "value"])$p.value,
    CI95l = t.test(value)$conf.int[1],
    CI95h = t.test(value)$conf.int[2]
)

data <- merge(data, summary, by=c("cellLine", "gene", "construct"))

data$pValue <- ifelse(data$pValue == 1, NA, data$pValue)

data$pStars <- ifelse(
    data$pValue < 0.05 & data$pValue > 0.0049, 
    "\u2605", 
    ifelse(
        data$pValue < 0.005 & data$pValue > 0.00049, 
        "\u2605\u2605",
        ifelse(
            data$pValue < 0.0005, 
            "\u2605\u2605\u2605",
            ""
        )
    )
)

#prepare for plotting
data$gene <- ifelse(data$gene == "renilla", "Renilla", "Luciferase")
data$construct <-gsub("empty", "Empty", data$construct)
```

```{r, fig.align='center', fig.height=8, fig.width=12, eval=TRUE, message=FALSE, warning=FALSE}
#plot for Rmarkdown
ggplot(data, aes(x=construct, y=value))+
    geom_violin(
        aes(fill=gene),
        trim=TRUE,
        alpha=0.3,
        color=NA,
        scale="width",
        show.legend = FALSE
    )+
    geom_point( 
        aes(colour=gene),
        size=4,
        position=position_dodge(width=0.9),
        stroke=0
    )+
    geom_linerange(
        aes(ymin=CI95l, ymax=CI95h, group=gene),
        colour="black",
        position=position_dodge(width=0.9),
        show.legend = FALSE
    )+
    geom_point(
        aes(y=mean, group=gene),
        colour="black",
        position=position_dodge(width=0.9),
        show.legend = FALSE
    )+
    facet_grid(.~cellLine)+
    geom_label(
        aes(y=CI95h+0.15, label=pStars, group=gene),
        label.size=0,
        label.padding=unit(0.01, "lines"),
        show.legend = FALSE,
        fill="white",
        size=5,
        family="Arial Unicode MS",
        position=position_dodge(width=0.9)
    )+
    labs(
        x="Construct",
        y="Activity (a.u.)",
        title="miR34a asRNA regulation by miR34a promoter",
        caption=
            "miR34a asRNA and miR34a HG levels in HCT116 and HEK293t cells after transfection with P1 and P2 constructs.\nLuciferase represents miR34a HG and renilla represents miR34a asRNA.\nError bars (black vertical) indicate 95% CI and means are represented \nby black dots. \u2605 = p < 0.05; \u2605\u2605 = p < 0.005, \u2605\u2605\u2605 = p < 0.0005."
    )+
    theme_few()+
    scale_fill_ptol()+
    scale_colour_ptol()+
    #values=c("#014d64", "#6794a7")
    theme(
        plot.title = element_text(
            hjust = 0.5,
            face="bold",
            size=20
        ),
        plot.caption = element_text(
            hjust=0, 
            margin=margin(t=15),
            family="Arial Unicode MS"
        ),
        legend.position="top",
        legend.title=element_text(size=17),
        legend.text=element_text(size=15),
        axis.title=element_text(size=17),
        axis.title.x=element_text(margin=margin(t=10)),
        axis.text=element_text(size=15),
        strip.text.x=element_text(size=17)
    )+
    guides(color = guide_legend(title = "Gene", override.aes = list(size=3)))

#plot for pdf
p <- ggplot(data, aes(x=construct, y=value))+
    geom_violin(
        aes(fill=gene),
        trim=TRUE,
        alpha=0.3,
        color=NA,
        scale="width",
        lwd=0.1
    )+
    #geom_point( 
    #    aes(colour=gene),
    #    size=0.6,
    #    position=position_dodge(width=0.9)
    #)+
    geom_linerange(
        aes(ymin=CI95l, ymax=CI95h, group=gene),
        colour="black",
        position=position_dodge(width=0.9),
        show.legend = FALSE,
        lwd=0.15
    )+
    geom_point(
        aes(y=mean, group=gene),
        colour="black",
        position=position_dodge(width=0.9),
        show.legend = FALSE,
        size=0.5
    )+
    facet_grid(.~cellLine)+
    geom_label(
        aes(y=CI95h+1200, label=pStars, group=gene),
        label.size=0,
        label.padding=unit(0.01, "lines"),
        show.legend = FALSE,
        fill="white",
        size=1,
        family="Arial Unicode MS",
        position=position_dodge(width=0.9)
    )+
    labs(
        x="Construct",
        y="Activity (a.u.)",
        title="miR34a asRNA promoter region"
    )+
    theme_few()+
    scale_fill_ptol()+
    scale_colour_ptol()+
    #values=c("#014d64", "#6794a7")
    theme(
        plot.title = element_text(
            hjust = 0.5,
            face="bold",
            size=10,
            margin=margin(b=2, unit="mm")
        ),
        plot.margin = unit(c(0.1,0.1,0.1,0.1), "lines"),
        legend.position="top",
        legend.margin=unit(1,"mm"),
        legend.key.size = unit(1/5, "cm"),
        legend.title=element_text(size=9),
        legend.text=element_text(size=8),
        axis.title=element_text(size=9),
        axis.title.x=element_text(margin=margin(t=5)),
        axis.text.y=element_text(size=8),
        axis.text.x=element_text(size=7),
        strip.text.x=element_text(size=9, margin = margin(0.01, 0, 0.1, 0, "cm")),
        panel.border = element_rect(fill=NA, size=0.25),
        axis.ticks = element_line(size=0.25),
        axis.ticks.length = unit(1/15, "cm")
    )+
    guides(fill = guide_legend(title = "Gene", override.aes = list(size=2)))

ggsave(plot=p, filename="~/GitHub/miR34aFigures/inst/P1-HCTandHEK/P1-HCTandHEK_P1andP2.pdf", device=cairo_pdf, height=60, width=83, units="mm")

#only P1
data <- subset(data, construct %in% c("Empty", "p1"))
p1 <- ggplot(data, aes(x=construct, y=value))+
    geom_violin(
        aes(fill=gene),
        trim=TRUE,
        alpha=0.3,
        color=NA,
        scale="width",
        lwd=0.1
    )+
    #geom_point( 
    #    aes(colour=gene),
    #    size=0.6,
    #    position=position_dodge(width=0.9)
    #)+
    geom_linerange(
        aes(ymin=CI95l, ymax=CI95h, group=gene),
        colour="black",
        position=position_dodge(width=0.9),
        show.legend = FALSE,
        lwd=0.15
    )+
    geom_point(
        aes(y=mean, group=gene),
        colour="black",
        position=position_dodge(width=0.9),
        show.legend = FALSE,
        size=0.5,
    )+
    facet_grid(.~cellLine)+
    geom_label(
        aes(y=CI95h+1200, label=pStars, group=gene),
        label.size=0,
        label.padding=unit(0.01, "lines"),
        show.legend = FALSE,
        fill="white",
        size=1,
        family="Arial Unicode MS",
        position=position_dodge(width=0.9)
    )+
    labs(
        x="Construct",
        y="Activity (a.u.)"
        #title="miR34a asRNA promoter region"
    )+
    theme_few()+
    scale_fill_ptol()+
    scale_colour_ptol()+
    #values=c("#014d64", "#6794a7")
    theme(
        plot.title = element_text(
            hjust = 0.5,
            face="bold",
            size=10,
            margin=margin(b=2, unit="mm")
        ),
        plot.margin = unit(c(0.1,0.1,0.1,0.1), "lines"),
        legend.position="top",
        legend.margin=unit(1,"mm"),
        legend.key.size = unit(1/5, "cm"),
        legend.title=element_text(size=9),
        legend.text=element_text(size=8),
        axis.title=element_text(size=9),
        axis.title.x=element_text(margin=margin(t=5)),
        axis.text.y=element_text(size=8),
        axis.text.x=element_text(size=7),
        strip.text.x=element_text(size=9, margin = margin(0.01, 0, 0.1, 0, "cm")),
        panel.border = element_rect(fill=NA, size=0.25),
        axis.ticks = element_line(size=0.25),
        axis.ticks.length = unit(1/15, "cm")
    )+
    guides(fill = guide_legend(title = "Gene", override.aes = list(size=2)))

ggsave(plot=p1, filename="~/GitHub/miR34aFigures/inst/P1-HCTandHEK/P1-HCTandHEK_P1.pdf", device=cairo_pdf, height=60, width=83, units="mm")

```

#Conclusion
Upon transfection of this construct in two separate cell lines we observed increases in both luciferase and renilla indicating that miR34a HG and asRNA expression is regulated by a single promoter contained within the p1 construct. Transfection of the P2 construct showed to cause a reduction in luciferase (miR34a HG) levels but not renilla (miR34a AS) levels. This could be due to the presence of a STAT3 binding site in p2 which has been previously reported to negativley regulate miR34a HG.

###Session info
```{r}
sessionInfo()
```
