---
title: "Coding Potenital of the miR34a asRNA"
author: "Jason T. Serviss"
date: "31/01/2017"
output:
  html_document:
    theme: flatly
    toc: no
    code_folding: hide
---

#Introduction
We utilized a bioinformatic approach to evaluate the coding potential of the miR34a asRNA transcript. The Coding-potential assessment tool uses a linear regression model to evaluate coding-potential by examining ORF length, ORF coverage, Fickett score and hexamer score. We further confirmed these results using the Coding-potential Calculator which utilizes a support based machine-based classifier and accesses an alternate set of discriminatory features.

#Methods
Protein-coding capacity was evaluated using the Coding-potential Assessment Tool [43] and Coding-potential Calculator [44] with default settings. Transcript sequences for use with Coding-potential Assessment Tool were downloaded from the UCSC genome browser using the following IDs: HOTAIR (ENST00000455246.1), XIST (ENST00000429829.1), β-actin (ENST00000331789.5), Tubulin (ENST00000427480.1), and MYC (ENST00000377970). Transcript sequences for use with Coding-potential Calculator were downloaded from the UCSC genome browser using the following IDs: HOTAIR (uc031qho.1), β-actin (uc003soq.4).

#Results
```{r}
library(printr)
```

##Coding Potential
###Coding Potential Assesment Tool
```{r}
cpat <- read.table('CodingPotentialAssessmentTool/results.txt', header=TRUE, sep="\t")
cpat
```

Coding potential analysis results from the Coding-potential Assessment Tool including miR34a asRNA and 
2 characterized non-coding transcripts (HOTAIR and XIST) and 3 known coding transcripts (β-actin, tubulin, and MYC). 

###Coding Potential Calculator
```{r}
cpc <- read.table('CodingPotentialCalculator/results.txt', header=TRUE, sep="\t")
colnames(cpc) <- gsub("\\.", " ", colnames(cpc))
cpc
```

Coding potential analysis results using the Coding-potential calculator for miR34a asRNA, HOTAIR, and β-actin.

#Conclusions
Results indicated that miR34a asRNA has a similar lack of coding capacity to the known non-coding transcripts HOTAIR and XIST and differs greatly when examining these parameters to the known coding transcripts beta-actin, tubulin, and MYC. However, to fully evaluate coding potential methods such as mass spectrometry or ribosome profiling must be used. 