---
title: "Cellular localisation"
author: "Jason T. Serviss"
date: "31/01/2017"
output:
  html_document:
    theme: flatly
    toc: no
    code_folding: hide
---

##Introduction
AsRNAs are known to be capable of exerting regulatory functions on both a transcriptional and post-transcriptional level and therefore, the cellular localization of the asRNA can be informative concerning the method of regulation. 

##Methods
All cell lines were cultured at 5% CO2 and 37° C with HEK293T cells cultured in DMEM high glucose (Hyclone). All growth mediums were supplemented with 10% heat-inactivated FBS and 50 μg/ml of streptomycin and 50 μg/ml of penicillin. Cells were plated at 300,000 cells per well in a 6-well plate and cultured overnight. Fractionation was performed using the PARIS kit according to the manufacturer's recommendations (Life Technologies). 0.5ul RNA was used for cDNA synthesis using MuMLV (Life Technologies) and a 1:1 mix of oligo(dT) and random nanomers. QPCR was carried out using KAPA 2G SYBRGreen (Kapa Biosystems) using the Applied Biosystems 7900HT machine with the cycling conditions: 95 °C for 3 min, 95 °C for 3 s, 60 °C for 30 s.

```{r}
library(printr)
primers <- data.frame(
    name=c(
        "ß-actin Fwd",
        "ß-actin Rev",
        "miR34a HG_F",
        "miR34a HG_R",
        "miR34a_asF1",
        "miR34a_asR1",
        "U48_F",
        "U48_R"
        
    ),
    sequence=c(
        "AGG TCA TCA CCA TTG GCA ATG AG",
        "CTT TCG GGA TGT CCA CGT CA",
        "TCT GCT CCA GTG GCT GAT GAG AAA",
        "GTT CAC TGG CCT CAA AGT TGG CAT",
        "AGC GGC ATC TCC TCC ACC TGA AA",
        "TTG CCT CGT GAG TCC AAG GAG AAT",
        "AGT GAT GAT GAC CCC AGG TA",
        "GGT CAG AGC GCT GCG GTG AT"
        
    )
)
primers
```

##Results
```{r}
library(reshape2)
library(ggplot2)
library(ggthemes)
library(plyr)
library(scales)

data <- read.table('~/GitHub/miR34aFigures/inst/cellularLocalization/QPCRdata.txt', header=TRUE, sep="\t")
data$meanCt <- rowMeans(data[,c("Ct1", "Ct2")])
data$meanAntilog <- 2^-data$meanCt

sums <- ddply(
    data,
    c("gene"),
    summarize,
    total = sum(meanAntilog)
)

data <- merge(data, sums, by="gene")
data$percentage <- 100 * (data$meanAntilog / data$total)

#prepare for plotting
data$gene <- gsub("miR34a asRNA", "miR34a\nasRNA", data$gene)
data$gene <- gsub("miR34a HG", "miR34a\nHG", data$gene)
data$gene <- factor(data$gene, levels=c("miR34a\nasRNA", "miR34a\nHG", "B-actin", "RNU48"))
```

```{r, fig.align='center', fig.height=8, fig.width=12, eval=TRUE, message=FALSE, warning=FALSE}
#plot for Rmarkdown
ggplot(data, aes(x=gene, y=percentage))+
    geom_bar(aes(fill=fraction), position="fill", stat="identity", alpha=0.6)+
    scale_y_continuous(labels = percent_format())+
    labs(
      x="Gene",
      y="%",
      title="miR34a asRNA cellular localization",
      caption=
            "Q-PCR results from fractionated HEK293T cells investigating the cellular localization of the miR34a asRNA transcript."
    )+
    theme_few()+
    scale_fill_ptol()+
    scale_colour_ptol()+
    #values=c("#014d64", "#6794a7")
    theme(
        plot.title = element_text(
            hjust = 0.5,
            face="bold",
            size=20
        ),
        plot.caption = element_text(
            hjust=0, 
            margin=margin(t=15),
            family="Arial Unicode MS"
        ),
        legend.position="top",
        legend.title=element_text(size=17),
        legend.text=element_text(size=15),
        axis.title=element_text(size=17),
        axis.title.x=element_text(margin=margin(t=10)),
        axis.text=element_text(size=15),
        strip.text.x=element_text(size=17)
    )+
    guides(fill = guide_legend(title = "Fraction", override.aes = list(size=3)))

#plot for pdf
p <- ggplot(data, aes(x=gene, y=percentage))+
    geom_bar(aes(fill=fraction), position="fill", stat="identity", alpha=0.6)+
    scale_y_continuous(labels = percent_format())+
    labs(
      x="Gene",
      y="%"
    )+
    theme_few()+
    scale_fill_ptol()+
    scale_colour_ptol()+
    theme(
        plot.title = element_text(
            hjust = 0.5,
            face="bold",
            size=10,
            margin=margin(b=2, unit="mm")
        ),
        plot.margin = unit(c(0.1,0.1,0.1,0.1), "lines"),
        legend.position="top",
        legend.margin=unit(1,"mm"),
        legend.key.size = unit(1/5, "cm"),
        legend.title=element_text(size=9),
        legend.text=element_text(size=8),
        axis.title=element_text(size=9),
        axis.title.x=element_text(margin=margin(t=5)),
        axis.text.y=element_text(size=8),
        axis.text.x=element_text(size=7),
        strip.text.x=element_text(size=9, margin = margin(0.01, 0, 0.1, 0, "cm")),
        panel.border = element_rect(fill=NA, size=0.25),
        axis.ticks = element_line(size=0.25),
        axis.ticks.length = unit(1/15, "cm")
    )+
    guides(fill = guide_legend(title = "Fraction", override.aes = list(size=2)))

ggsave(plot=p, filename="~/GitHub/miR34aFigures/inst/cellularLocalization/cellularLocalization.pdf", device=cairo_pdf, height=60, width=83, units="mm")

```

##Conclusion
Semi-quantitative PCR revealed that the miR34a asRNA transcript localizes to both the nucleus and cytoplasm although the majority is located in the cytoplasmic fraction.