---
title: "P1 shRenilla HEK293t transfections"
author: "Jason T. Serviss"
date: "12/01/2017"
output:
  html_document:
    theme: flatly
    toc: no
    code_folding: hide
---

#Introduction
Functional characterization of individual antisense transcripts has previously shown their capability to regulate their sense gene. We therefore investigated the possibility that miR34a asRNA may regulate miR34a HG levels. We hypothesized that the overlapping regions of the sense and antisense transcripts may have a crucial role in miR34a asRNAs ability to regulate miR34a HG via RNA:DNA or RNA:RNA interaction. Accordingly, we first co-transfected the p1 construct, containing the overlapping region of the two transcripts, and an short hairpin (sh) RNA targeting renilla into HEK293T cells .

#Methods
We utilized the P1 construct where the overlapping region of miR34a HG and miR34a AS is cloned with luciferase downstream of miR34a HG and renilla downstream of miR34a AS. The p1 sequence was previously published in Raver-Shapira, N., et al., Transcriptional activation of miR-34a contributes to p53-mediated apoptosis. Mol Cell, 2007. 26(5): p. 731-43. All cell lines were cultured at 5% CO2 and 37° C with HEK293T cells cultured in DMEM high glucose (Hyclone). All growth mediums were supplemented with 10% heat-inactivated FBS and 50 μg/ml of streptomycin and 50 μg/ml of penicillin. 2.5x10^5 293T cells were seeded in a 12-well plate. After 24hrs these were co-transfected with 50 ng of the P1 construct and 250 ng shRenilla using lipo2k standard protocol. After 48 hours, RNA was extracted using the RNeasy mini kit (Qiagen) and subsequently treated with DNase (Ambion Turbo DNA-free, Life Technologies). 500ng RNA was used for cDNA synthesis using MuMLV (Life Technologies) and a 1:1 mix of oligo(dT) and random nanomers. QPCR was carried out using KAPA 2G SYBRGreen (Kapa Biosystems) using the Applied Biosystems 7900HT machine with the cycling conditions: 95 °C for 3 min, 95 °C for 3 s, 60 °C for 30 s.

```{r}
library(printr)
p1 <- data.frame(
    name="p1",
    sequence="GCG CCC TGC CTG GCC CCC ACC TGG TCC TCT TTC CTT TTC AGG TGG AGG AGA TGC CGC TGT
              CCC GTC GGT CTG GGG ACA GCC CAG CTC CCC GGA TCC CGG GCT GGA GAG ACG CGT CGC GGC   
              CCC GGG GCC TGG TGG CAC GAG CAG GAA GGA GGA CCC GGC GGC GGG CTC TGC CTG GGC TTG 
              CCT GGG CTT GTT CCG AGC CGG GCT GCT TCT CGG TGA CCA CGC AGA TCG GGG GCA TTT GGA
              GAT TTT GCG GGA GTC CTG CAG CCA AGC TCC GGG GCA GGA GAG GCC TGG AAG CCT GCA CTA
              CCT GCT C"
)
p1
```

```{r}
shRNAs <- data.frame(
    name=c(
        "shRenilla1.1",
        "shRenilla2.1"
    ),
    sequence=c(
        "AAT ACA CCG CGC TAC TGG C",
        "TAA CGG GAT TTC ACG AGG C"
    )
)
shRNAs
```

```{r}
primers <- data.frame(
    name=c(
        "Luc_set_II_F",
        "Luc_set_II_R",
        "Renilla_pBiDir_F1",
        "Renilla_pBiDir_R1",
        "B-actin_F",
        "B-actin_R"
    ),
    sequence=c(
        "AAG ATT CAA AGT GCG CTG CTG",
        "TTG CCT GAT ACC TGG CAG ATG",
        "TAA CGC GGC CTC TTC TTA TTT",
        "GAT TTG CCT GAT TTG CCC ATA",
        "AGG TCA TCA CCA TTG GCA ATG AG",
        "CTT TGC GGA TGT CCA CGT CA"
    )
)
primers
```

#Results
```{r}
library(reshape2)
library(ggplot2)
library(ggthemes)
library(plyr)
data <- read.table('~/GitHub/miR34aFigures/inst/HEK293tP1/QPCRdata.txt', header=TRUE, sep="\t")

goi <- subset(data, gene %in% c("Luciferase", "Renilla"))
hk <- subset(data, gene == "B-actin")
hk <- rbind(hk,hk)
hk <- hk[order(hk$belongsTo), ]

colnames(goi)[5:6] <- c("Ct1GOI", "Ct2GOI")
colnames(hk)[5:6] <- c("Ct1HK", "Ct2HK")

data <- cbind(goi, hk[,c("Ct1HK", "Ct2HK")])
data$meanGOI <- rowMeans(data[,c("Ct1GOI", "Ct2GOI")])
data$meanHK <- rowMeans(data[,c("Ct1HK", "Ct2HK")])

#ddct
data$dct <- data$meanGOI - data$meanHK
ddct <- ddply(
    data,
    c("experiment", "shRNA", "belongsTo", "gene"),
    summarize,
    ddct=dct-data[data$experiment == experiment & data$shRNA == "shCrtl" & data$belongsTo == belongsTo & data$gene == gene, "dct"]
)
data <- merge(data, ddct, by=c("experiment", "shRNA", "belongsTo", "gene"))

data$fold <- 2^-data$ddct
data$log2fold <- log2(data$fold)

#calculate stats
stats <- ddply(
    data,
    c("shRNA", "belongsTo", "gene"),
    summarize,
    mean=mean(log2fold),
    pValue=t.test(log2fold, data[data$shRNA == "shCrtl" & data$belongsTo == belongsTo & data$gene == gene ,"log2fold"], alternative="less")$p.value,
    CI95l=t.test(log2fold)$conf.int[1],
    CI95h=t.test(log2fold)$conf.int[2]
)

stats$pStars <- ifelse(
    stats$pValue < 0.05 & stats$pValue > 0.0049, 
    "\u2605", 
    ifelse(
        stats$pValue < 0.005 & stats$pValue > 0.00049, 
        "\u2605\u2605",
        ifelse(
            stats$pValue < 0.0005, 
            "\u2605\u2605\u2605",
            ""
        )
    )
)

data <- merge(data, stats, by=c("shRNA", "belongsTo", "gene"))
```

```{r, fig.align='center', fig.height=8, fig.width=12, eval=TRUE, message=FALSE, warning=FALSE}
#plot for rmarkdown
ggplot(data, aes(x=shRNA, y=log2fold))+
    geom_violin(
        aes(fill=gene),
        trim=TRUE,
        alpha=0.3,
        color=NA,
        scale="width",
        show.legend = FALSE
    )+
    geom_point( 
        aes(colour=gene),
        size=4,
        position=position_dodge(width=0.9)
    )+
#    geom_linerange(
#        aes(ymin=CI95l, ymax=CI95h, group=gene),
#        colour="black",
#        position=position_dodge(width=0.9),
#        show.legend = FALSE
#    )+
    geom_point(
        aes(y=mean, group=gene),
        colour="black",
        position=position_dodge(width=0.9),
        show.legend = FALSE
    )+
#    geom_label(
#        aes(y=CI95h+0.15, label=pStars, group=gene),
#        label.size=0,
#        label.padding=unit(0.01, "lines"),
#        show.legend = FALSE,
#        fill="white",
#        size=5,
#        family="Arial Unicode MS",
#        position=position_dodge(width=0.9)
#    )+
    scale_y_continuous(
        "\u0394\u0394Ct",
        sec.axis = sec_axis(
            ~2^., 
            name = expression(2^{- ~ Delta ~ Delta ~ "Ct"}),
            label=round(c(2^-6,2^-5,2^-4,2^-3,2^-2,2^-1,2^0), digits=3),
            breaks=c(2^-6,2^-5,2^-4,2^-3,2^-2,2^-1,2^0)
        )
    )+
    labs(
      x="shRNA",
      title="HEK293t",
      caption=
            "miR34a (luficerase) and miR34a asRNA (renilla) in HEK293t cells which were co-transfected with the P1 construct and the specified shRNA.\nError bars (black vertical) indicate 95% CI and means are represented by black dots."
    )+
    theme_few()+
    scale_fill_ptol()+
    scale_colour_ptol()+
    #values=c("#014d64", "#6794a7")
    theme(
        plot.title = element_text(
            hjust = 0.5,
            face="bold",
            size=20
        ),
        plot.caption = element_text(
            hjust=0, 
            margin=margin(t=15),
            family="Arial Unicode MS"
        ),
        legend.position="top",
        legend.title=element_text(size=17),
        legend.text=element_text(size=15),
        axis.title=element_text(size=17),
        axis.title.x=element_text(margin=margin(t=10)),
        axis.text=element_text(size=15),
        strip.text.x=element_text(size=17)
    )+
    guides(color = guide_legend(title = "Gene", override.aes = list(size=3)))

#plot for pdf
```

#Conclusions
Analysis of luciferase and renilla expression revealed that by decreasing levels of the renilla transcript (corresponding to miR34a asRNA) luciferase (corresponding to miR34a HG) levels were concomitantly decreased. These results indicate that miR34a asRNA is capable of regulating the miR34a HG in a concordant manner and that the overlapping region of the miR34a asRNA may be sufficient to mediate this regulation.
