---
title: "miR34a asRNA transcript stability"
author: "Jason T. Serviss"
date: "02/02/2017"
output:
  html_document:
    theme: flatly
    toc: no
    code_folding: hide
---

#Introduction
We desired to deduce the relative stability of the miR34a asRNA and HG transcripts.

#Methods
All cell lines were cultured at 5% CO2 and 37° C with U2OS cells grown in McCoy’s 5a (Life Technologies). All growth mediums were supplemented with 10% heat-inactivated FBS and 50 μg/ml of streptomycin and 50 μg/ml of penicillin. Cells were treated with 10nM actinomycin D (Sigma) for the indicated times. RNA was extracted using the RNeasy mini kit (Qiagen) and subsequently treated with DNase (Ambion Turbo DNA-free, Life Technologies). 500ng RNA was used for cDNA synthesis using MuMLV (Life Technologies) and a 1:1 mix of oligo(dT) and random nanomers. QPCR was carried out using KAPA 2G SYBRGreen (Kapa Biosystems) using the Applied Biosystems 7900HT machine with the cycling conditions: 95 °C for 3 min, 95 °C for 3 s, 60 °C for 30 s.

```{r}
library(printr)
primers <- data.frame(
    name=c(
        "ß-actin Fwd",
        "ß-actin Rev",
        "miR34a HG_F",
        "miR34a HG_R",
        "miR34a_asF1",
        "miR34a_asR1"
        
    ),
    sequence=c(
        "AGGTCATCACCATTGGCAATGAG",
        "CTTTCGGGATGTCCACGTCA",
        "TCTGCTCCAGTGGCTGATGAGAAA",
        "GTTCACTGGCCTCAAAGTTGGCAT",
        "AGCGGCATCTCCTCCACCTGAAA",
        "TTGCCTCGTGAGTCCAAGGAGAAT"
        
    )
)
primers
```

#Results

```{r}
library(reshape2)
library(ggplot2)
library(ggthemes)
library(plyr)
data <- read.table('~/GitHub/miR34aFigures/inst/transcriptStability/QPCRdata.txt', header=TRUE, sep="\t")

goi <- subset(data, gene %in% c("miR34a asRNA", "miR34a HG"))
hk <- subset(data, gene == "B-actin")

colnames(goi)[c(4:5)] <- c("Ct1GOI", "Ct2GOI")
colnames(hk)[c(4:5)] <- c("Ct1HK", "Ct2HK")

data <- merge(goi, hk[,!colnames(hk) %in% "gene"], by=c("experiment", "treatment"))

#ddct
data$meanGOI <- rowMeans(data[,c("Ct1GOI", "Ct2GOI")])
data$meanHK <- rowMeans(data[,c("Ct1HK", "Ct2HK")])
data$dct <- data$meanGOI - data$meanHK
ddct <- ddply(
    data,
    c("gene", "treatment", "experiment"),
    summarize,
    ddct=dct-data[data$gene == gene & data$experiment == experiment & data$treatment == 0, "dct"]
)

data <- merge(data, ddct, by=c("gene", "treatment", "experiment"))
data$fold <- 2^-data$ddct
data$log2fold <- log2(data$fold)

#calculate stats
stats <- ddply(
    data, 
    c("gene", "treatment"),
    summarize,
    mean=mean(log2fold),
    sd=sd(log2fold)
)

data <- merge(data, stats, by=c("gene", "treatment"))

#prepare for plotting
data$treatment <- factor(data$treatment, levels=c("0", "1", "2", "4"))
data$gene <- factor(data$gene, levels=c("miR34a asRNA", "miR34a HG"))
```

```{r, fig.align='center', fig.height=8, fig.width=12, eval=TRUE, message=FALSE, warning=FALSE}
#plot for Rmarkdown
ggplot(data, aes(x=treatment, y=log2fold))+
    geom_point( 
        aes(colour=gene),
        fill="white",
        size=4,
        alpha=0.6,
        position=position_dodge(width=0.9)
    )+
    geom_linerange(
        aes(ymin=mean-sd, ymax=mean+sd, group=gene),
        colour="black",
        position=position_dodge(width=0.9),
        show.legend = FALSE
    )+
    geom_point(
        aes(y=mean, group=gene),
        colour="black",
        position=position_dodge(width=0.9),
        show.legend = FALSE
    )+
    scale_y_continuous(
        "\u0394\u0394Ct",
        #breaks=c(-1,0,1,2,3,4,5),
        sec.axis = sec_axis(
            ~2^., 
            name = expression(2^{- ~ Delta ~ Delta ~ "Ct"}), 
            breaks=c(2^-4,2^-3,2^-2,2^-1,2^0)
        )
    )+
    labs(
      x="Treatment (hours)",
      title="miR34a asRNA and HG stability",
      caption=
            "Q-PCR showing miR34a asRNA and miR34a HG levels in U2OS cells after treatment with 10nM actinomycin D.\nError bars (black vertical) indicate standard deviation and means are represented by black dots."
    )+
    theme_few()+
    scale_fill_ptol()+
    scale_colour_ptol()+
    #values=c("#014d64", "#6794a7")
    theme(
        plot.title = element_text(
            hjust = 0.5,
            face="bold",
            size=20
        ),
        plot.caption = element_text(
            hjust=0, 
            margin=margin(t=15),
            family="Arial Unicode MS"
        ),
        legend.position="top",
        legend.title=element_text(size=17),
        legend.text=element_text(size=15),
        axis.title=element_text(size=17),
        axis.title.x=element_text(margin=margin(t=10)),
        axis.text=element_text(size=15),
        strip.text.x=element_text(size=17)
    )+
    guides(color = guide_legend(title = "Gene", override.aes = list(size=3)))

#plot for pdf
p <- ggplot(data, aes(x=treatment, y=log2fold))+
    geom_point( 
        aes(colour=gene),
        stroke=0,
        size=2,
        alpha=0.6,
        position=position_dodge(width=0.9)
    )+
    geom_linerange(
        aes(ymin=mean-sd, ymax=mean+sd, group=gene),
        colour="black",
        position=position_dodge(width=0.9),
        show.legend = FALSE,
        lwd=0.2
    )+
    geom_point(
        aes(y=mean, group=gene),
        colour="black",
        stroke=0.75,
        position=position_dodge(width=0.9),
        show.legend = FALSE,
        size=0
    )+
    scale_y_continuous(
        "\u0394\u0394Ct",
        #sec.axis = sec_axis(
        #    ~2^., 
        #    name = expression(2^{- ~ Delta ~ Delta ~ "Ct"}), 
        #    breaks=c(2^-4,2^-3,2^-2,2^-1,2^0)
        #)
    )+
    labs(
      x="Treatment (hours)"
      #title="miR34a asRNA and HG stability"
    )+
    theme_few()+
    scale_fill_ptol()+
    scale_colour_ptol()+
    theme(
        plot.title = element_text(
            hjust = 0.5,
            face="bold",
            size=10,
            margin=margin(b=2, unit="mm")
        ),
        plot.margin = unit(c(0.1,0.1,0.1,0.1), "lines"),
        legend.position="top",
        legend.margin=unit(1,"mm"),
        legend.key.size = unit(1/5, "cm"),
        legend.title=element_text(size=9),
        legend.text=element_text(size=8),
        axis.title=element_text(size=9),
        axis.title.x=element_text(margin=margin(t=5)),
        axis.text.y=element_text(size=8),
        axis.text.x=element_text(size=7),
        strip.text.x=element_text(size=9, margin = margin(0.01, 0, 0.1, 0, "cm")),
        panel.border = element_rect(fill=NA, size=0.25),
        axis.ticks = element_line(size=0.25),
        axis.ticks.length = unit(1/15, "cm")
    )+
    guides(color = guide_legend(title = "Gene", override.aes = list(size=2)))

ggsave(plot=p, filename="~/GitHub/miR34aFigures/inst/transcriptStability/transcriptStability.pdf", device=cairo_pdf, height=60, width=83, units="mm")

```

#Conclusions
While the levels of miR34a HG fall quickly over the 4 hour treatment period, levels of miR34a asRNA remain relativley stable.
