## Referee #1
We kindly thank the referee for his insightful review of our manuscript. We have addressed all of the referee's concerns and questions with new experiments or textual clarifications, and believe it has increased the overall value of the manuscript. Furthermore, we have taken the opportunity to further clarify several points both in our reply and in the manuscript itself.#### Referee 1 Comment:The manuscript is well written, it is logically arranged and easy to follow. Data are clearly presented, analyzed and interpreted. This reviewer has an only concern regarding the biological impact of the observed mechanism. The analysis of cell proliferation and cell cycle. Cell growth and cell proliferation are two different phenotypes. Authors used indirect method for analysis of cell growth - using analysis of confluence. I would recommend to perform direct analysis of cell numbers and determine proliferation rate (growth curve) using ordinary methods - count the cells using cell counter (Coulter, Casy) or using hemocytometer or any other image-based method which is able to identify and count single cells. Confluency of the cells may be affected by morphology, adhesion capabilities etc. Moreover, authors present the values as normalized data. It seems to be important to present the number of cells in time as real numbers normalized per area. Authors need to explain the discrepancy between results of cell cycle analysis and analysis of cell proliferation (cell numbers), growth curve should be presented for both cell lines for time interval 0 – 72 (or 96h) to cover at least 2-3 divisions. Data from the cell cycle analysis (examples of models of histograms from Modfit sw), should be part of supplementary data). HBSS culture condition seems to me not appropriate. I would recommend performing the experiment in low serum or serum free condition but in standard media which is necessary for cell maintenance. Modulation of cell cycle (F3b) seems to be very weak and biologically not significant. In this reviewer opinion, the authors should modify their interpretation and statements in to context of realtively weak phenotypes observed in cell-based experiments.#### Referee 1 Response:
**In response to the statement:** _“Cell growth and cell proliferation are two different phenotypes. Authors used indirect method for analysis of cell growth - using analysis of confluence. I would recommend to perform direct analysis of cell numbers and determine proliferation rate (growth curve) using ordinary methods - count the cells using cell counter (Coulter, Casy) or using hemocytometer or any other image-based method which is able to identify and count single cells.”_The authors agree that confluency measurements may incorporate multiple phenotypes that could be affected by the overexpression of lncTAM34a. To address this issue, an additional experiment was performed where proliferation rate was measured using the CellTrace violet assay. CellTrace violet is a fluorescent dye that covalently binds to all free amines on the surface and inside of cells. As cells proliferate, the dye becomes diluted and thus measuring fluorescence intensity via flow cytometry over time allows quantification of proliferation at a single cell level. Importantly, the assay is not affected by cell morphology or cell adhesion capabilities. We believe this method to be more accurate and allow the assaying of a comparatively larger numbers of cells and have, therefore, chosen to use it quantify proliferation rate as per the reviewers suggestion. This data has now been added to the manuscript as Supplementary Figure 5B and is also included below.

<br></br>
<p align="center">
<img src="/Users/jasonserviss/Github/miR34a_asRNA_project/inst/supp_figure5b/supp_figure5b.png" alt="Drawing" style="width: 600px;"/>
</p>

_The effects of miR34a asRNA overexpression on proliferation in normal and starvation conditions in the PC3 prostate cancer cell line. The y-axis illustrates the mean difference (∆; Time - Time 0) in flouresence intensity for each biological replicate and condition where decreased flouresence intensity indicates increased proliferation. Boxplots show the distribution of the mean differences from each of the biological replicates (n = 3). Grey lines indicate the polynomial regression model with the shadows indicating the 95% confidence intervals. Estimates, standard error (std.error), and P values for the cell line covariate for each model are indicated in the upper left hand corner._

The results confirm the role of increased lncTAM34a expression resulting in a decrease in proliferation. The results differ somewhat to the confluency analysis in regards to the degree to which these changes occur in the context of cellular stress. Several change have been made in the text to reflect the insights gained from this experiment and are outlined below:

This: "Under normal growth conditions there is a small but significant reduction (P = 3.0e-8; linear regression, Fig. 3c) in confluency in the lncTAM34a over-expressing cell lines compared to mock control. However, these effects on cell growth are drastically increased in starvation conditions (P = 9.5e-67; linear regression; Fig. 3c). This is in agreement with our previous results, and suggests thatlncTAM34a-mediated increases in miR34a expression are crucial under conditions of stress and necessary for the initiation of an appropriate cellular response. In summary, we find that over-expression of lncTAM34a is sufficient to increase miR34a expression and gives rise to known phenotypes observed upon induction of miR34a."

Has been changed to this: "Under normal growth conditions there is a small but significant reduction (P = 3.1e-6; polynomial regression, Fig. 3c, Supplementary Figure 5A) in confluency in the lncTAM34a over-expressing cell lines compared to mock control. However, these effects on cell growth are clearly increased in starvation conditions (P = 4.3e-85; polynomial regression; Fig. 3c, Supplementary Figure 5A). Specifically measuring proliferation rate, as opposed to confluency, largely confirms the observation that increases in lncTAM34a expression mediate decreases in cell cycle, although the degree to which cellular stress effects these changes differs when measuring confluency or proliferation (Supplementary Figure 5B). This suggests that additional cellular phenotypes, such as morphology or cell adhesion capability, may also be affected by increased lncTAM34a expression in contexts of cellular stress. In summary, we find that over-expression of lncTAM34a is sufficient to increase miR34a expression and gives rise to known phenotypes observed upon induction of miR34a."We would also like to note that the method for modeling the data for Figure 3C was changed from linear regression to polynomial regression as it was found to provide a better fit for the non-linear trends observed in the data. The appropriate changes to reflect this have been made in the figure and throughout the text.
**In response to the statement:** _“It seems to be important to present the number of cells in time as real numbers normalized per area.”_
Upon rereading the materials and methods for the experiment where confluency was measured, we realized that the description of the normalization procedure was incorrect. It has therefore been changed as outlined below. 
This: “Percentage of confluency was normalized to the control sample in each condition...”. 
Has now been changed to this: “Fold confluency was then calculated as % confluency / % confluency time 0 for each condition and cell line...”
Effectively this calculation causes each data point to represent the fold difference compared to time point 0 within each condition and cell line. This fact was potentially not previously clear to the reader and has now been altered via a) clarification in the materials and methods (outlined above), b) the y-axis of figure 3c is now labeled "Fold confluency", and c) the calculation is present in the figure legend.
We choose to calculate fold confluency instead of showing the percentage confluency values due to the fact that, in our opinion, it better illustrates the differences between the two cell lines, as opposed to the culture conditions, which is in line with the conclusions we draw. To provide the reader with a representative example of the differences in percentage of confluency, photos taken by the plate reader machine and used to quantify confluency during the experiment have now been added as Supplementary Figure 5A and is also shown below.

<br></br>
<p align="center">
<img src="/Users/jasonserviss/Github/miR34a_asRNA_project/inst/supp_figure5a/supp_figure5a.png" alt="Drawing" style="width: 400px;"/>
</p>

_Representative pictures from the growth analysis shown in Figure 3C. The green color indicates the surface covered by cells as detected by the analysis software._

We believe this to provide a good visual proxy for the real number of cells within the culture area. Considering the conclusions drawn from Figure 3c, we feel that expressing the results in fold, and labeled as “Fold confluency” in the figure, instead of "Normalized % Confluency", is the most straightforward and clear way to convey the observed results..**In response to the statement:** _“Authors need to explain the discrepancy between results of cell cycle analysis and analysis of cell proliferation (cell numbers), growth curve should be presented for both cell lines for time interval 0 – 72 (or 96h) to cover at least 2-3 divisions.”_The authors are unsure which analyses the reviewer is referring to with the statement "cell cycle analysis and analysis of cell proliferation (cell numbers)". It is clear that the "cell cycle analysis" referes to Figure 3B and we assume that "analysis of cell proliferation (cell numbers)" refers to Figure 3C. Although, if this assumption is correct, we cannot see the discrepancy the reviewer is referring to. The conclusions drawn from Figure 3B are as follows:

"Cell cycle analysis via determination of DNA content showed a significant increase in G1 phase cells and a concomitant decrease in G2 phase cells in the PC3 and Skov3 lncTAM34a over-expressing cell lines, indicating G1 arrest (Fig. 3b, Supplementary Figure 4B)."

The conclusions drawn from Figure 3C under normal (RPMI) culture conditions are as follows:

"Under normal growth conditions there is a small but significant reduction (P = 3.1e-6; polynomial regression, Fig. 3c, Supplementary Figure 5A-B) in confluency in the lncTAM34a over-expressing cell lines compared to mock control."

The conclusions drawn from Figure 3B indicate that we would expect to see a reduction in confluency when lncTAM34a is overexpressed in the experiment performed and shown in Figure 3C and, in fact, that is what is shown. Concordantly, the conclusions drawn from Figure 3C indicate that we would expect a decrease in the proportion of cycling cells in Figure 3B and, again, this is what the data indicates.

Confluency is an indirect measurement of multiple specific cellular phenotypes, as the reviewer has pointed out - including not only proliferation, but also growth, morphology, and adhesion capabilities. The additional experiment performed to specifically measure proliferation (Supplementary Figure 5B), also indicates that there is a reduction of proliferation upon lncTAM34a overexpression in normal (RPMI) culture conditions. Despite this, there is some discrepancy concerning the observed degree of change in confluency and proliferation upon encountering starvation conditions that is now highlighted in the main text. Overall, the results indicate that lncTAM34a overexpression leads to a decrease in proliferation in line with results from previous studies examining the effects of miR34a upregulation on the cell cycle (Raver-Shapira et al., 2007, _Mol. Cell_, Figure 6F). This is specifically stated in the main text in the following manner:

"In summary, we find that over-expression of lncTAM34a is sufficient to increase miR34a expression and gives rise to known phenotypes observed upon induction of miR34a."

Also please note, in addition, that the experiment shown in Supplementary Figure 5B was performed over a 72 hour time frame to cover 2-3 cell divisions as requested by the reviewer.
**In response to the statement:** _“Data from the cell cycle analysis (examples of models of histograms from Modfit sw), should be part of supplementary data).”_The requested data has now been added as Supplementary Figure 4B.**In response to the statement:** _“HBSS culture condition seems to me not appropriate. I would recommend performing the experiment in low serum or serum free condition but in standard media which is necessary for cell maintenance.”_The additional experiment analyzing cellular proliferation rate (Supplementary Figure 5B) was performed with the starvation condition, 0.1% FBS, to address this comment. Supplementary Figure 5A also shows that proliferation is ongoing, especially in the mock cell line, when cells are cultured in HBSS. 
**In response to the statement:** _“Modulation of cell cycle (F3b) seems to be very weak and biologically not significant. In this reviewer opinion, the authors should modify their interpretation and statements in to context of realtively weak phenotypes observed in cell-based experiments.”_Although the effect size specifically in Figure 3B may be relatively small, the authors feel that the conclusions drawn from the data are justified. Specifically, the conclusions are:

"Collectively, these results indicate that lncTAM34a-mediated induction of miR34a is sufficient to result in the corresponding miR34a-directed effects on cell cycle."The effect sizes in Figure 3B (8% and 9% increase in G1 phase cells in PC3 and Skov3 cells, respectively, upon lncTAM34a overexpression) are > 2 fold larger than what has been observed in a previous study upon direct transfection of miR34a in 293 cells (Raver-Shapira et al., 2007, _Mol. Cell_, Figure 6F). Importantly, miR34a has subsequently been shown to mediate tumor size in _in vivo_ models in multiple studies (examples include: Park et al., 2014, _Cancer res._, Figure 5A-B. Daige et al., 2014, Mol. Cancer Ther., Figure 2D, 3D). The strong effect on tumor size mediated by small changes in cell cycle is expected since they will, over time, result in exponential changes in absolute cell numbers. Effective evaluation of whether this would also be the case upon increased lncTAM34a expression would necessitate appropriate _in vivo_ models which are outside the scope of this study.## Referee #2
We would like to kindly thank the reviewer for their thorough reading of the manuscript and the useful and constructive comments that were provided. We believe that we have now addressed all of the reviewer comments the replies to which can be found below.**Referee 2 Comment #1:**
Figure 1B: as normal PCR provides only semiquantitative results, it would be best to provide this data qualitatively, through qPCR as well as providing data on p53 mRNA and protein expression in your panel of cell lines so correlation analysis can be performed. Caution should be taken when using the term ‘correlate’ as this suggests statistical analysis has been performed, of which I could find no evidence for Fig1B. Only two cell lines were p53-, and some cell lines which were p53+ had no lncRNA or mirHG expression. Either use more p53- cell lines to support your statement that the p53 negative cell lines had lower expression of these transcripts, or rephrase your sentence.**Referee 2 Comment #1 Response:**We agree that our usage of the term “correlate” in context with the data presented in the Figure 1B was unclear and has been altered in the text (see below). Data has been previously produced with additional TP53 null cell lines although this information is not currently included in the figure. We are hesitant to add these additional cell lines due to the fact that they were not run in the same conditions or experiment as the cell lines currently in the figure. We include the data here for the referees review and have, in addition, altered the text to correspond with the reviewer’s suggestion based on the information presented in the manuscript as follows:
   This text: “These results indicate that miR34a HG and lncTAM34a are co-expressed and that their expression levels correlate with TP53 status, with TP53-/- cells tending to have decreased or undetectable expression of both transcripts.”Has been changed to: “These results indicate that miR34a HG and lncTAM34a are co-expressed and that their expression levels are related to TP53 status, with TP53-/- cells tending to have decreased or undetectable expression of both transcripts.”

![](/Users/jasonserviss/Desktop/miR34aOldDontDelete/reviewersComments1/TP53null.png)

In addition we note that, although it would be feasible perform correlation analysis in cell lines that are either TP53 wild-type or TP53-null, many of the cell lines have TP53 mutations or other alterations known to effect TP53 function (indicated with a +<sup>*</sup> in Figure 1B). Due to the fact that these mutations may or may not influence TP53’s function, and additional alterations (e.g. SV40 large T antigen expression in HEK293T cells or human papillomavirus 18 E6 expression in Hela cells) may affect its function to differing degrees, we would be hesitant to consider these cell lines as a homogeneous group in a correlation analysis. Furthermore, for the reasons outlined above, it would seem as though TP53 expression, on an mRNA or protein level, is not an appropriate proxy for its ability to function within the cell. Finally, the primary, and potentially most interesting, evidence for the relationship between TP53 status, miR34a expression, and lncTAM34a expression lies in the analysis performed in patient samples in Figure 1C and Supplementary Figure 1A-B.**Referee 2 Comment #2:**
Some of the gel figures (Figure 1B, S2) had no labels on the molecular weight markers, please include those so it is easy to cross reference expected product sizes.

**Referee 2 Comment #2 Response:**
Molecular weight labels have now been added to all semi-quantitative PCR figures. These include: Figure 1B, Figure 1E, and Supplementary Figure 2B.**Referee 2 Comment #3:**
Supplementary figure S3: as P1 is only exon 1 of both the lncRNA and the HG, I have doubts whether the expression of renilla and luciferase indicates transcription of both these genes, rather bidirectional transcription of exon 1 which may not extend to the full transcript. I am not convinced that knocking down renilla (lncTAM34) and the concomitant decrease of luciferase is representative of an actual mir34 HG response. Based on these experiments, I don’t think you can definitely say that lncRNATAM affects mirHG.**Referee 2 Comment #3 Response:**The authors agree with the reviewer that this experiment does not, in itself, provide the strongest evidence to support the conclusions drawn in the manuscript. We have now included additional experiments that show a concomitant decrease in miR34a HG levels upon siRNA-mediated targeting of the endogenous lncTAM34a transcript (Fig. 2d, shown below). Taking into account these new results, together with the additional evidence presented in the manuscript, it is of the authors opinion that the conclusion that lncTAM34a positively regulates miR34a is well supported in the data. 

<br></br>
<p align="center">
<img src="/Users/jasonserviss/Github/miR34a_asRNA_project/inst/figure2d/figure2d.pdf" alt="Drawing" style="width: 400px;"/>
</p>

_lncTAM34a (n = 4) and miR34a HG (n = 3) levels after 48 hour siRNA-mediated knock down of lncTAM34a in U2OS cells. Individual points represent results from independent experiments, error bars show the 95% CI, black horizontal lines represent the mean, and P values are shown over long horizontal lines indicating the comparison tested._  
**Referee 2 Comment #4:**
In figure 2, the P1 construct is employed because of the various isoforms of the lncRNA, but it is then later overexpressed using a lentiviral system without mention of how they tackled this multiple isoform issue.
**Referee 2 Comment #4 Response:**
We agree that this section should be clarified. In these experiments, we chose to overexpress the most abundant isoform in the stable cell lines. Evidence for the fact that there is an isoform that is dominantly expressed is currently in the manuscript (Supplementary Figure 7) although it is not directly pointed out to the reader that a) this is the case or b) that we made this decision. Therefore following text has been altered in the manuscript:

This: "Utilizing a lentiviral system, we stably over-expressed the lncTAM34a transcript in three TP53 -null cell lines, PC3 (prostate cancer), Saos2 (osteogenic sarcoma), and Skov3 (ovarian adenocarcinoma)."

Has been changed to this: "Utilizing a lentiviral system, we stably over-expressed the most abundant isoform (Supplementary Figure 7) of lncTAM34a in three TP53 -null cell lines, PC3 (prostate cancer), Saos2 (osteogenic sarcoma), and Skov3 (ovarian adenocarcinoma)."

In addition we would like to add that, although we were able to successfully clone the dominantly expressed isoform, we are unable to specifically target this isoform due to the structure of the locus. This is due to the fact that many of the more lowly expressed alternatively spliced transcripts include extra exons (Supplementary Figure 2C). This has also been more explicitly stated in the text in the following manner (lines 226-228):

"Knockdown of endogenous lncTAM34a is complicated by its various isoforms (Supplementary Figure 2C) and targeting individual isoforms is not possible due to the structure of the locus."**Referee 2 Comment #5:**For supplementary figure 4C, it would be helpful to include a representative western blot figure instead of just the analysis. **Referee 2 Comment #5 Response:**
We agree that this would be helpful to the reader. A representative figure has now been added to Supplementary Figure 4C.**Referee 2 Comment #6:**
For supplementary figure 3D, since lncRNA and mirHG share the same promoter which you showed in Figure 2C, how can you be sure that the subsequent increase in pol binding after lncRNA expression is just not because you have introduced overexpression vectors for lncRNA? Is the effect you observe just because in the process of activating the lncRNA, polymerases are recruited to this region?**Referee 2 Comment #6 Response:**
We assume that the reviewer is referring to Figure 3D in the main text (Supplementary Figure 3D does not exist) that shows increased polII binding at the miR34a locus upon lncTAM34a overexpression. The increase in polII binding is detected via primers located outside of the cloned and overexpressed region and, for this reason, the multiple copies of the over-expressed transcript in the genome have no effect on the detected enrichment of polII binding at the endogenous locus. Although we demonstrate, in Figure 2C, that the promoter region in the P1 construct is sufficient to drive expression of both miR34a HG and lncTAM34a, the full regulatory region that controls transcription may be larger than the region cloned in the P1 construct. In fact, the miR34a HG and lncTAM34a promoter region shows broad enrichment for both activating histone marks (Figure 1A) and phosphorylated polII binding in cell lines where the sense and antisense transcripts are expressed. In the figure below, from the UCSC genome browser, we can see evidence for phosphorylated polII binding, detected via ChIP-seq, in two cell lines that express miR34a HG and lncTAM34a (Hela and HepG2) and one that does not (K562) (please note that all of these cell lines are included in the cell panel screening in Figure 1B). The polII ChIP primers (chipF and chipR) clearly show that they are within the polII-binding region, in cell lines where miR34a and lncTAM34a are expressed, but upstream of the 5’ end (cloningF) of the cloned and overexpressed transcript. We have made the following modification in the main text to indicate that we have no evidence to believe that P1 constitutes the entire transcriptional regulatory sequence but, instead, have evidence that it is sufficient to drive expression of both transcripts.
This text: “Upon transfection of p1 into HCT116 and HEK293T cell lines we observed increases in both luciferase and renilla indicating that miR34a HG and lncTAM34a expression can be regulated by a single promoter contained within the p1 construct (Fig. 2c).”
Has been changed to this: “Upon transfection of p1 into HCT116 and HEK293T cell lines we observed increases in both luciferase and renilla indicating that the sequence in the P1 construct is sufficient to drive both miR34a HG and lncTAM34a expression (Fig. 2c).”

![](/Users/jasonserviss/Desktop/miR34aOldDontDelete/reviewersComments1/ChipPrimers.png)**Referee 2 Comment #7:**
For figure 4A, there are only 5 p53 mutated samples in comparison to p53 wild type. Although this is statistically significant, without comparable sample size the relevance of this is questionable.**Referee 2 Comment #7 Response:**We wish to stress that the main message of the figure and the related supplementary figures is to show 1) the correlation between low lncTAM34a/miR34a expression and worse survival and 2) how this effect is similar but also independent on the effect of the TP53 mutation status. We are aware that, given the small sample size, the log-rank p-value is relative for TP53 in Figure 4A, although TP53 mutations effect on outcome is well known in the literature (Poeta et al., 2007, _New Eng. J. Med._, 357;25. Zenz et al., 2010, _J. Clin. Oncol._, 28;29.). Additionally, the data represented in Figure 4A is only a subset of the total analysis, the results of which are shown in full in Supplementary Figure 6.
**Referee 2 Comment #8:**
While the characterization of the lncRNA was very thoroughly done, it is difficult to ascertain from these results whether the lncRNA is functional or whether it is because you change expression of the lncRNA, you are inadvertently targeting the mirHG, and it is not the effects of the lncRNA per se. **Referee 2 Comment #8 Response:**
We believe this comment is mostly based on the issues surrounding Referee 2 comment #6 that, hopefully, were sufficiently addressed in the response. To recapitulate, transcription of miR34a was evaluated via phosphorylated pollII enrichment at the endegenous locus upon overexpression of lncTAM34a. lncTAM34a overexpression caused an increase in phosphorylated polII binding at the miR34a locus and is detected via primers located outside of the cloned and overexpressed region. For this reason, the multiple copies of the over-expressed transcript in the genome have no effect on the detected enrichment of polII binding at the endogenous locus.