---
title: "Figure 2d"
subtitle: "miR34a asRNA knock down in U2OS cells"
author: "Jason T. Serviss"
date: "31/01/2017"
output:
  html_document:
    theme: flatly
    code_folding: hide
vignette: >
  %\VignetteIndexEntry{Vignette Title}
  %\VignetteEncoding{UTF-8}
  %\VignetteEngine{knitr::rmarkdown}
editor_options: 
  chunk_output_type: console
---

```{r, echo = FALSE, message = FALSE}

packages <- c(
    "tidyverse",
    "printr",
    "ggthemes",
    "readr",
    "miR34AasRNAproject",
    "grid",
    "gtable"
)
purrr::walk(packages, library, character.only = TRUE)
rm(packages)

```

#### _Introduction_

We hypothesized that miR34a asRNA may function to mediate the levels of its sense gene, miR34a. To this end, we utilized
siRNA targeting exon 2 of the miR34a asRNA transcript to lower its expression levels and observe possible effects on 
miR34a HG in U2OS cells.

<br></br>

#### _Methods_

<br></br>

_Cell culture, transfection and luminescence quantification_

All cell lines were cultured at 5% CO2 and 37° C with U2OS cells cultured in McCoy’s 5a (Life Technologies). Growth mediums were supplemented with 10% heat-inactivated FBS and 50 μg/ml of streptomycin and 50 μg/ml of penicillin. Cells were plated at 25,000 cells per well in a 12-well plate and cultured overnight. The following day cells were transfected with 10nM si-Ctl or si-lncTAM34a using the standard lipofektamine 2000 (Life Technologies) protocol. si-Ctl was purchased from Qiagen (cat. nr. 1027310) and si-lncTAM34a was purchased from Eurofins. The sequence for si-Ctl is not available from the manufacturer although the sequence for si-lncTAM34a is shown below. Cells were harvested for RNA extraction 48 hours after transfection.

<br></br>

_siRNA_

```{r}

siRNA <- data.frame(
  name = c("si-lncTAM34a"),
  sequence = c("GGG AGA AGA CGA UUC UUU")
)
siRNA

```

<br></br>

_QPCR_

RNA was extracted using the RNeasy mini kit (Qiagen) and subsequently treated with DNase (Ambion Turbo DNA-free, Life Technologies). 500ng RNA was used for cDNA synthesis using MuMLV (Life Technologies) and a 1:1 mix of oligo(dT) and random nanomers. QPCR was carried out using KAPA 2G SYBRGreen (Kapa Biosystems) using the Applied Biosystems 7900HT machine with the cycling conditions: 95 °C for 3 min, 95 °C for 3 s, 60 °C for 30 s. QPCR primers are shown below.

<br></br>

_QPCR Primers_

```{r}

primers <- data.frame(
    name=c(
        "miR34a asRNA F1",
        "miR34a asRNA R1",
        "miR34aHG_F",
        "miR34aHG_R",
        "B-actin_F",
        "B-actin_R"
    ),
    sequence=c(
        "AGC GGC ATC TCC TCC ACC TGA AA",
        "TTG CCT CGT GAG TCC AAG GAG AAT",
        "TCT GCT CCA GTG GCT GAT GAG AAA",
        "GTT CAC TGG CCT CAA AGT TGG CAT",
        "AGG TCA TCA CCA TTG GCA ATG AG",
        "CTT TGC GGA TGT CCA CGT CA"
    )
)
primers

```

<br></br>

_Analysis_

Three independant experiments (biological replicates) were performed in total measuring miR34a HG whereas miR34a asRNA was measured in 4 biological replicates. The Student's t-test was used to compare the si-Ctl and si-lncTAM34a conditions log2 fold change for both genes.

<br></br>

#### _Results_

```{r, message = FALSE}

data <- getData("Figure 2d")

#calculate dct
groups <- c("siRNA", "Biological Replicate", "gene")
gois <- c("miR34a HG", "lncTAM34a")

data <- data %>%
  gather(Ct, ctMean, -(gene:`Biological Replicate`)) %>%
  group_by_at(groups) %>%
  mutate(`Technical Replicate` = 1:n()) %>%
  dct(., gois, Actin, c(groups, 'Technical Replicate'), F)

#calculate ddct
groups[groups == "gene"] <- "GOI"
logical <- tibble(
  siRNA = pull(data, siRNA) == "si-Ctl"
)

data <- data %>%
  mutate(br = `Biological Replicate`) %>%
  ddct(., logical, c(groups, "br")) %>%
  filter(`Technical Replicate` == 1) %>%
  select(-`Technical Replicate`) %>%
  distinct() %>%
  na.omit() %>%
  folds(.)

#calculate stats
stats <- calcStats(data, siRNA, "si-Ctl", groups) %>%
  pFormat(.)

#calculate 95% CI and mean on linear scale
stats <- data %>%
  #filter(siRNA == "si-lncTAM34a") %>%
  group_by(GOI, siRNA) %>%
  summarize(
    CI95l.linear = t.test(fold)$conf.int[1], 
    CI95h.linear = t.test(fold)$conf.int[2]
  ) %>%
  ungroup() %>%
  full_join(stats, by = c("siRNA", "GOI"))

stats <- data %>%
  group_by(GOI, siRNA) %>%
  summarize(mean.linear = mean(fold)) %>%
  ungroup() %>%
  full_join(stats, by = c("siRNA", "GOI"))

#vertical lines and labels in plot
vl <- stats %>%
  mutate(
    x = case_when(
      GOI == "miR34a HG" ~ 2 - 0.25,
      GOI == "lncTAM34a" ~ 1 - 0.25
    ),
    xend = case_when(
      GOI ==  "miR34a HG" ~ 2 + 0.25,
      GOI == "lncTAM34a" ~ 1 + 0.25
    ),
    label = if_else(GOI == "miR34a HG", 2, 1), 
    max = max(CI95h.linear)
  )
```

```{r, fig.align='center', fig.height=8, fig.width=12, warning = FALSE}

p <- ggplot(data = NULL) +
  geom_dotplot(
    data = data, 
    aes(
      x = `GOI`,
      y = fold,
      fill = siRNA
    ),
    binwidth = 0.01,
    colour = NA,
    #alpha = 0.5,
    position = position_dodge(width = 0.9), 
    binaxis = 'y', 
    stackdir = 'center',
    dotsize = 4,
    show.legend = TRUE
  ) +
  scale_y_continuous("Fold")+
  labs(x = "Gene")

markdown <- p +
  geom_linerange(
    data = stats,
    aes(
      x = GOI,
      ymin = CI95l.linear,
      ymax = CI95h.linear,
      group = interaction(siRNA, GOI)
    ),
    position = position_dodge(width = 0.9),
    colour = "gray20",
    show.legend = FALSE
  ) +
  geom_point(
    data = stats,
    aes(
      x = GOI,
      y = mean.linear,
      group = interaction(siRNA, GOI)
    ),
    colour = "gray20",
    shape = 95,
    size = 6,
    position = position_dodge(width = 0.9),
    show.legend = FALSE
  ) +
  geom_label(
    data = vl,
    aes(
      x = label, 
      y = max + 0.075, 
      label = pFormat, 
      group = interaction(siRNA, GOI)
    ),
    fill = "white",
    label.size = 0,
    label.padding = unit(0.01, "lines"),
    show.legend = FALSE,
    fontface = "bold",
    family = "Arial Unicode MS",
    size = 3
  ) +
  geom_segment(
    data = vl,
    aes(
      x = x,
      y = max + 0.05,
      xend = xend,
      yend = max + 0.05
    ),
    colour = "grey43",
    show.legend = FALSE
  ) +
  guides(
    fill = guide_legend(
      title = "Gene", 
      override.aes = list(size = 6)
    )
  )

plotRmarkdown(markdown)

pdf <- p +
  geom_linerange(
    data = stats,
    aes(
      x = GOI,
      ymin = CI95l.linear,
      ymax = CI95h.linear,
      group = interaction(siRNA, GOI)
    ),
    position = position_dodge(width = 0.9),
    colour = "grey30",
    size = 0.3,
    show.legend = FALSE
  ) +
  geom_point(
    data = stats,
    aes(
      x = GOI,
      y = mean.linear,
      group = interaction(siRNA, GOI)
    ),
    colour = "grey30",
    shape = 95,
    size = 4,
    position = position_dodge(width = 0.9),
    show.legend = FALSE
  ) +
  geom_label(
    data = vl,
    aes(
      x = label, 
      y = max + 0.1, 
      label = pFormat, 
      group = interaction(siRNA, GOI)
    ),
    fill = "white",
    colour = "grey30",
    label.size = 0,
    label.padding = unit(0.01, "lines"),
    show.legend = FALSE,
    fontface = "bold",
    family = "Arial Unicode MS",
    size = 2.5
  ) +
  geom_segment(
    data = vl,
    aes(
      x = x,
      y = max + 0.05,
      xend = xend,
      yend = max + 0.05
    ),
    colour = "grey30",
    size = 0.3,
    show.legend = FALSE
  ) +
  guides(
    fill = guide_legend(
      title = "Gene", 
      override.aes = list(size = 4)
    )
  )

figure <- plotPDF(pdf)

#path <- file.path("~/GitHub/miR34a_asRNA_project/inst", fileMap(type = "pdf")["Figure 2d"][[1]])
#ggsave(
#  plot = figure,
#  filename = path,
#  device = cairo_pdf,
#  height = 60,
#  width = 68,
#  units = "mm"
#)
```

<div class="caption">
lncTAM34a (n = 4) and miR34a HG (n = 3) levels after 48 hour siRNA-mediated knock down of lncTAM34a in U2OS cells.
</div>

<br></br>

#### _Conclusion_

miR34a asRNA KD gives rise to a ~20% decrease in miR34a HG expression levels.

<br></br><br></br>

```{r}
sessionInfo()
```