---
title: "Figure 1d"
subtitle: "miR34a asRNA 3-prime RACE"
author: "Jason T. Serviss"
date: "31/01/2017"
output:
  html_document:
    theme: flatly
    code_folding: hide
vignette: >
  %\VignetteIndexEntry{Vignette Title}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

<style> 
.caption { 
  color: Black; 
  font-family: "Times New Roman", Times, serif;
  font-size: 1.0em; 
}
</style>

```{r, echo = FALSE, message = FALSE}

packages <- c(
    "tidyverse",
    "printr",
    "ggthemes",
    "readr",
    "miR34AasRNAproject",
    "grid",
    "gtable"
)
purrr::walk(packages, library, character.only = TRUE)
rm(packages)

```

```{r}

projectUrl <- "https://github.com/GranderLab/miR34a_asRNA_project/raw/master/inst"
dataUrl <- "https://github.com/GranderLab/miR34a_asRNA_project/raw/master/"

```

#### _Introduction_

In order to define the 3' transcription termination site for the miR34a asRNA, 3'-RACE was performed.

<br></br>

#### _Methods_

<br></br>

_Cell culture and 3-prime RACE_

All cell lines were cultured at 5% CO2 and 37° C with U2OS cells grown in McCoy’s 5a (Life Technologies). All growth mediums were supplemented with 10% heat-inactivated FBS and 50 μg/ml of streptomycin and 50 μg/ml of penicillin. U2OS cell RNA was polyA-tailed using yeast polyA polymerase after which cDNA was synthesized using oligo(dT) primers. Nested-PCR was performed first using a forward primer in miR34a asRNA exon 1 (miR34a_asRNA_F10) and a tailed oligo(dT) primer (polyT_T7FAM) followed by a second PCR using an alternate miR34a asRNA exon 1 primer (miR34a_asRNA_F1) and a reverse primer binding to the tail of the previously used oligo(dT) primer (FAMprimer). PCR products were gel purified and cloned the Strata Clone Kit (Agilent Technologies), and sequenced.

<br></br>

_Primers_

```{r}

primers <- data.frame(
  name = c("miR34a_asRNA_F10", "polyT_T7FAM", "miR34a_asRNA_F1", "FAMprimer"),
  sequence = c(
    "ACG CGT CTC TCC AGC CCG GGA T",
    "CAG TGA ATT GTA ATA CGA CTC ACT ATA GGG ACA TCC GTA GCT CGT CCA GGA CCC TTT TTT TTT TTT TTT TTT VN",
    "AGC GGC ATC TCC TCC ACC TGA AA",
    "CCG TAG CTC GTC CAG GAC CC"
  )
)
primers

```

<br></br>

```{r, fig.align='center', fig.height=2, fig.width=12, warning = FALSE, message = FALSE}
data <- tibble(
  gene = readr::parse_factor(
    c(rep("miR34a asRNA", 5), rep("lnc34a", 3), "miR34a asRNA F1", "miR34a asRNA F10"),
    levels = c("miR34a asRNA", "lnc34a", "miR34a asRNA F1", "miR34a asRNA F10")
  ),
  feature = c(
    "exon", "intron", "exon", "intron", "exon",
    "exon", "intron", "exon",
    "primer", "primer"
  ),
  start = c(
    9181991, 9182316, 9183633, 9183807, 9191393,
    9181737, 9182295, 9196912,
    9182279, 9182223
    ),
  stop = c(
    9182316, 9183633, 9183807, 9191393, 9191686,
    9182295, 9196912, 9197043,
    9182301, 9182244
  )
)

p <- ggplot(data = NULL) +
  gggenes::geom_gene_arrow(
    data = filter(data, feature %in% c("exon") & gene == "miR34a asRNA"),
    aes(xmin = start, xmax = stop, y = gene, fill = gene),
    arrowhead_width = unit(0, "mm"),
    arrowhead_height = unit(0, "mm"),
    colour = "gray"
  )  +
  geom_segment(
    data = filter(data, feature == "intron" & gene == "miR34a asRNA"),
    aes(x = start, xend = stop, y = gene, yend = gene),
    linetype = "dotted",
    colour = "gray",
    size = 1
  ) +
  geom_segment(
    data = filter(data, feature == "primer"),
    aes(x = start, xend = stop, y = 1.1, yend = 1.1, colour = gene),
    size = 5 
  ) +
  labs(x = "Chromosome 1") +
  guides(fill = FALSE, colour = guide_legend(title = "Primer"))

plotRmarkdown(p) +
  theme_void() +
  scale_fill_brewer()
```

<div class = "caption">
3’-RACE schematic illustrating primer placement on the miR34a asRNA transcript for 3'-RACE.
</div>

<br></br>

#### _Results_

```{r, out.width = "850px"}

url <- fileMap(type = "png")["figure1d"][[1]]
knitr::include_graphics(file.path(projectUrl, url))

```

<div class = "caption">
3’-RACE sequencing results displayed in the UCSC genome browser together with the annotated miR34a asRNA transcript (LINC01759) from GENCODE.
</div>

<br></br>

#### _Conclusions_

Sequencing of the resulting cloned cDNA indicated the transcripts 3’ transcription termination site to be 125bp upstream of the RP3-510D11.2 transcript’s annotated termination site.

<br></br><br></br>

```{r}
sessionInfo()
```
