---
title: "Figure 1c"
subtitle: "miR34a asRNA and miR34a TCGA correlation"
author: "Jimmy Van den Eynden and Jason T. Serviss"
date: "31/01/2017"
output:
  html_document:
    theme: flatly
    code_folding: hide
vignette: >
  %\VignetteIndexEntry{Vignette Title}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

<style> 
.caption { 
  color: Black; 
  font-family: "Times New Roman", Times, serif;
  font-size: 1.0em; 
}
</style>

```{r, echo = FALSE, message = FALSE}

packages <- c(
    "tidyverse",
    "printr",
    "ggthemes",
    "readr",
    "miR34AasRNAproject",
    "grid",
    "gtable"
)
purrr::walk(packages, library, character.only = TRUE)
rm(packages)

```

```{r}

projectUrl <- "https://github.com/GranderLab/miR34a_asRNA_project/raw/master/inst"
dataUrl <- "https://github.com/GranderLab/miR34a_asRNA_project/raw/master/"

```

#### _Introduction_
We interogated RNA sequencing data from TCGA to determine in miR34a and miR34a asRNA expression was correlated in both TP53 wild type and mutated samples.

<br></br>

#### _Methods_

RNA-Seq data and copy number data were downloaded from TCGA and processed as described previously (Ashouri et al. 2016). Briefly, RNA-Seq data were aligned to the human hg19 assembly and quantified using GENCODE (v19) annotated HTSeq-counts and FPKM normalizations. Expression data from miR34a and miR34 asRNA (identified as RP3-510D11.2) were used for further analysis. Copy number amplitudes for GENCODE genes were determined from segmented copy-number data. Samples that were diploid for miR34 asRNA were identified as those samples that had copy number amplitudes between -0.1 and 0.1.

Somatic mutation data were downloaded from the Genomics Data Commons data portal (GDC) as mutation annotation format (maf) files, called using Mutect2 on 30/10/2017 (v7) (Grossman et al. 2016). 

<br></br>

#### _Results_
```{r}
url <- fileMap(type = "rds")["Figure 1c"][[1]]
data <- read_rds(gzcon(url(file.path(dataUrl, url))))

#normalize expression values
data <- mutate(
    data,
    RP3 = log2(RP3 / max(RP3)),
    miR34a = log2(miR34a / max(miR34a))  
  )

#add BRCA subtypes
data <- data %>%
  mutate(PAM50 = if_else(is.na(PAM50), "", PAM50)) %>%
  mutate(cancer_PAM50 = gsub("BRCA", "BRCA ", paste(cancer, PAM50, sep = "")))

#sort 
data <- arrange(data, cancer_PAM50, TP53, RP3)

#remove infinite data
data <- filter(data, !is.infinite(RP3) & !is.infinite(miR34a))

#diploid only
data <- filter(data, abs(RP3_cna) < 0.1)
```

```{r, fig.align='center', fig.height=8, fig.width=12, message = FALSE}
#plot
plotTCGAcorrelation(data, type = "diploid only")

##save to pdf
# path <- file.path("~/GitHub/miR34a_asRNA_project/inst", fileMap(type = "pdf")["Figure 1c"][[1]])
# pdf(path, width = 7.7, height = 4)
# plotTCGAcorrelation(data, type = "diploid only")
# invisible(dev.off())
```

<div class="caption">
A graphical depiction of the TCGA correlation analysis. Bladder Urothelial Carcinoma (BLCA), Breast invasive carcinoma (BRCA), Head and Neck squamous cell carcinoma (HNSC), Lower Grade Glioma (LGG), Liver hepatocellular carcinoma (LIHC), Lung adenocarcinoma (LUAD), Lung squamous cell carcinoma (LUSC), Ovarian serous cystadenocarcinoma (OV), Prostate adenocarcinoma (PRAD), Skin Cutaneous Melanoma (SKCM), Stomach adenocarcinoma (STAD).
</div>

See Supplementary Figure 1a for R-squared and p-values corersponding with the analysis.

<br></br>

#### _Conclusions_
miR34a asRNA and miR34a tend to be highly correlated in the cancer types examined in both TP53 wild type and mutated samples. In addition, the results indicate that both miR34a asRNA and miR34a expression levels are decreased in the presence of TP53 mutations.

<br></br><br></br>

```{r}
sessionInfo()
```
