---
title: "Supplementary Figure 1f"
subtitle: "miR34a asRNA transcript stability"
author: "Jason T. Serviss"
date: "31/01/2017"
output:
  html_document:
    theme: flatly
    code_folding: hide
vignette: >
  %\VignetteIndexEntry{Vignette Title}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

<style> 
.caption { 
  color: Black; 
  font-family: "Times New Roman", Times, serif;
  font-size: 1.0em; 
}
</style>

```{r, echo = FALSE, message = FALSE}

packages <- c(
    "tidyverse",
    "printr",
    "ggthemes",
    "readr",
    "miR34AasRNAproject",
    "grid",
    "gtable"
)
purrr::walk(packages, library, character.only = TRUE)
rm(packages)

```

```{r}

projectUrl <- "https://github.com/GranderLab/miR34a_asRNA_project/raw/master/inst/"
dataUrl <- "https://github.com/GranderLab/miR34a_asRNA_project/raw/master/data"

```

####_Introduction_

We desired to deduce the relative stability of the miR34a asRNA and HG transcripts.

<br></br>

####_Methods_

<br></br>

_Cell culture and QPCR_

All cell lines were cultured at 5% CO2 and 37° C with U2OS cells grown in McCoy’s 5a (Life Technologies). All growth mediums were supplemented with 10% heat-inactivated FBS and 50 μg/ml of streptomycin and 50 μg/ml of penicillin. Cells were treated with 10nM actinomycin D (Sigma) for the indicated times. RNA was extracted using the RNeasy mini kit (Qiagen) and subsequently treated with DNase (Ambion Turbo DNA-free, Life Technologies). 500ng RNA was used for cDNA synthesis using MuMLV (Life Technologies) and a 1:1 mix of oligo(dT) and random nanomers. QPCR was carried out using KAPA 2G SYBRGreen (Kapa Biosystems) using the Applied Biosystems 7900HT machine with the cycling conditions: 95 °C for 3 min, 95 °C for 3 s, 60 °C for 30 s.

<br></br>

_Analysis_

QPCR was performed with technical replicates for each sample. Delta ct was calculated by subtracting the mean of the technical replicates for each sample from the mean of the technical replicates for the house keeping gene for the corresponding sample. Delta-delta ct was subsequently calculated by subtracting the delta ct values for the no treatment sample from the delat ct values for each individual sample. The experiment was performed in two independant experiments which are shown individually in the plot below. 95% confidence interval was calculated for each gene and treatment time point using the values obtained from each independant experiment.

<br></br>

_Primers_

```{r}

primers <- data.frame(
    name=c(
        "ß-actin Fwd",
        "ß-actin Rev",
        "miR34a HG_F",
        "miR34a HG_R",
        "miR34a_asF1",
        "miR34a_asR1"
        
    ),
    sequence=c(
        "AGGTCATCACCATTGGCAATGAG",
        "CTTTCGGGATGTCCACGTCA",
        "TCTGCTCCAGTGGCTGATGAGAAA",
        "GTTCACTGGCCTCAAAGTTGGCAT",
        "AGCGGCATCTCCTCCACCTGAAA",
        "TTGCCTCGTGAGTCCAAGGAGAAT"
        
    )
)
primers

```

<br></br>

####_Results_

```{r, message = FALSE}

url <- "suppFigure1f_transcriptStability.rds"
data <- read_rds(gzcon(url(file.path(dataUrl, url))))

#calculate dct
groups <- c("treatment", "Biological Replicate", "gene")
gois <- c("miR34a asRNA", "miR34a HG")

data <- technicalMeans(data, groups) %>%
  dct(., gois, Actin, groups, F)

#calculate ddct
groups[groups == "gene"] <- "GOI"
logical <- tibble(
  treatment = data$treatment == 0
)

data <- data %>%
  ddct(., logical, groups, F) %>%
  folds(.)

#calculate stats
stats <- data %>%
  group_by(GOI, treatment) %>%
  summarize(mean = mean(log2fold), sd = sd(log2fold)) %>%
  ungroup()

```

```{r, fig.align='center', fig.height=8, fig.width=12}

p <- ggplot(data = NULL)+
  scale_y_continuous("\u0394\u0394Ct") +
  labs(
    x = "Treatment (hours)",
    title = "miR34a asRNA and HG stability"
  ) +
  guides(color = guide_legend(title = "Gene", override.aes = list(size = 3)))

markdown <- p +
  geom_point(
    data = data,
    aes(
      x = treatment,
      y = log2fold,
      colour = GOI
    ),
    fill = "white",
    size = 4,
    alpha = 0.6,
    position = position_dodge(width = 0.9)
  ) +
  geom_linerange(
    data = stats,
    aes(
      x = treatment,
      ymin = mean - sd,
      ymax = mean + sd,
      group = GOI
    ),
    colour = "black",
    position = position_dodge(width = 0.9),
    show.legend = FALSE
  ) +
  geom_point(
    data = stats,
    aes(
      x = treatment,
      y = mean,
      group = GOI
    ),
    colour = "black",
    position = position_dodge(width = 0.9),
    show.legend = FALSE,
    shape = 95,
    size = 6
  )

plotRmarkdown(markdown)

pdf <- p +
  geom_point(
    data = data,
    aes(
      x = treatment,
      y = log2fold,
      colour = GOI
    ),
    fill = "white",
    alpha = 0.6,
    stroke = 0,
    position = position_dodge(width = 0.9)
  ) +
  geom_linerange(
    data = stats,
    aes(
      x = treatment,
      ymin = mean - sd,
      ymax = mean + sd,
      group = GOI
    ),
    colour = "black",
    position = position_dodge(width = 0.9),
    size = 0.1,
    show.legend = FALSE
  ) +
  geom_point(
    data = stats,
    aes(
      x = treatment,
      y = mean,
      group = GOI
    ),
    colour = "black",
    position = position_dodge(width = 0.9),
    show.legend = FALSE,
    shape = 95
  )

figure <- plotPDF(pdf)
path <- "~/GitHub/miR34a_asRNA_project/inst/suppFigure1f_transcriptStability/suppFigure1f_transcriptStability.pdf"

#ggsave(
#  plot = figure, 
#  filename = path, 
#  device = cairo_pdf, 
#  height = 60,
#  width = 83,
#  units = "mm"
#)
```

<div class="caption">
Q-PCR showing miR34a asRNA and miR34a HG levels in U2OS cells after treatment with 10nM actinomycin D. The points represent the values obtained from each independant experiment (n = 2). 95% confidence interval (vertical black lines) and mean (horizontal black lines) are also shown.
</div>

<br></br>

####_Conclusions_

While the levels of miR34a HG fall quickly over the 4 hour treatment period, levels of miR34a asRNA remain relativley stable.

<br></br><br></br>
