---
title: "Figure 3a"
subtitle: "miR34a asRNA stable over-expression cell lines"
author: "Jason T. Serviss"
date: "31/01/2017"
output:
  html_document:
    theme: flatly
    code_folding: hide
vignette: >
  %\VignetteIndexEntry{Vignette Title}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

<style> 
.caption { 
  color: Black; 
  font-family: "Times New Roman", Times, serif;
  font-size: 1.0em; 
}
</style>

```{r, echo = FALSE, message = FALSE}

packages <- c(
    "tidyverse",
    "printr",
    "ggthemes",
    "readr",
    "miR34AasRNAproject",
    "grid",
    "gtable"
)
purrr::walk(packages, library, character.only = TRUE)
rm(packages)

```

```{r}

projectUrl <- "https://github.com/GranderLab/miR34a_asRNA_project/raw/master/inst"
dataUrl <- "https://github.com/GranderLab/miR34a_asRNA_project/raw/master/"

```

#### _Introduction_

We hypothesized that miR34a asRNA may function to mediate the levels of its sense gene, miR34a. To this end, we stably over-expressed miR34a asRNA, using lentiviral transduction, in 3 different p53-null cell lines and monitored the effects on miR34a.

<br></br>

#### _Methods_

<br></br>

_miR34a asRNA cloning_

RNA was harvested from U2OS cells using the RNeasy mini kit (Qiagen) and subsequently treated with DNase (Ambion Turbo DNA-free, Life Technologies). 500ng RNA was used for cDNA synthesis using MuMLV (Life Technologies) and a 1:1 mix of oligo(dT) and random nanomers. PCR was run using the miR34aAS_cloning_F4 and miR34aAS_cloning_Ex3_1 primers for 40 cycles at annealing temperature 58˚C and run on an agarose gel. Bands were excised and extracted using the QIAquick gel extraction kit (Qiagen). PCR products were cloned using the Strata Clone kit (Aligent) and sequenced (results below). The insert for the lentiviral construct (p-HIV-I2AG-U6) was subsequently sub-cloned from the sequenced plasmid using primers with PacI and NheI cleavage sites. The cloned transcript largley mirrors the GENCODE annotated transcript.

<br></br>

_Cloning primers_

```{r}

primers <- data.frame(
    name=c(
        "miR34aAS_cloning_F4",
        "miR34aAS_cloning_Ex3_1"
    ),
    sequence=c(
        "ACGCGTCTCTCCAGCCCGGGAT",
        "AATGATGGCCGCAACTAATGACGG"
    )
)
primers

```

<br></br>

_Sequencing results_

```{r}

construct <- data.frame(
    name="miR34a_asRNA",
    sequence="ACG CGT CTC TCC AGC CCG GGA TCC GGG GAG CTG GGC TGT CCC CAG ACC GAC GGG ACA GCG GCA
                TCT CCT CCA CCT GAA AAG GAA AGA GGA CCA GTT TGC AGG ACT CCG AAC TGG GCC CGC GAG ATC 
                TCC ACC TGC GCA AAA CGA AAG GGC GGA TTC TCC TTG GAC TCA CGA GGC AAC CGC TCC CCG GGG 
                TGA GAA CGG GGG ACT CAT TCC TCC GGC ACT GGG AGA AGA CGA TTC TTC AGG AGG AGG ACA GGG 
                AAG CGA ATG CTA CCC AGA TGT CTC AGT ATA CTG GCT CGC GGC ACA TCG GGC AAA TGA ACC TAT 
                CAG ATA ACA ACG GCA GAT CAG ATG CCT GAG CAT TCA GAA GCA ACA GCT GTG GAG CCC CCG TGG 
                GTT CAG AAG GCC TGG TTC CCG TCT CCA GAA GCC TGG CTC TCC TCC CTC CTG GGC CCA CTA CTT 
                TGG CTT CTT GTT CCT ACG TAC AAG GAG TTG CGA AGA AGG CAA CTC TTC CCC TCC CTG AAG CCA
                AAG GAA TGA AAC AGA CTA GGG CGG GAG AGG TGG CCA TCC GTC ATT AGT TGC GGC CAT CA"
)
construct

```

<br></br>

_Lentivirus production and infection_

HEK293T cells were transfected with viral and expression constructs using Lipofectamine 2000 (Life Technologies), after which viral supernatants were harvested 48 and 72 hours post-transfection. Viral particles were concentrated using PEG-IT solution (Systems Biosciences) according to the manufacturer’s recommendations. HEK293T cells were used for virus titration and GFP expression was evaluated 72hrs post-infection via flow cytometry after which TU/ml was calculated. Cell lines were subsequently infected with the lentiviral particals using a multuplicity of infection = 1. After 48 hours, cells were placed under mycophenolic acid selection and the selection process was monitored using GFP expression via flow cytometry. When GFP expression was present in >90% of the cells, cells were frozen for downstream analysis.

<br></br>

_QPCR_

RNA was extracted using the RNeasy mini kit (Qiagen) and subsequently treated with DNase (Ambion Turbo DNA-free, Life Technologies). 500ng RNA was used for cDNA synthesis using MuMLV (Life Technologies) and a 1:1 mix of oligo(dT) and random nanomers. QPCR was carried out using KAPA 2G SYBRGreen (Kapa Biosystems) using the Applied Biosystems 7900HT machine with the cycling conditions: 95 °C for 3 min, 95 °C for 3 s, 60 °C for 30 s. QPCR for miRNA expression analysis was performed according to the protocol for the TaqMan microRNA Assay kit (Life Technologies) with primer/probe sets purchased from the same company (TaqMan® MicroRNA Assay, hsa-miR-34a, human and Control miRNA Assay, RNU48, human) and the same cycling scheme as above. QPCR primers are shown below.

<br></br>

_Analysis_

Two experimental (technical) replicates were included in each QPCR run and delta ct was calculated for each sample using the mean of the gene of interest's technical replicates and the house keeping gene's technical replicates. delta-delta ct was calculated for each sample by subtracting the median dct value for the corresponding mock samples. At least three independant experiments (biological replicates) were performed in total (see more below). The Student's t-test was used to compare the mock vs miR34a asRNA overexpressing group's delta-delta ct values for both genes.  

<br></br>

The sample numbers for each group are as follows:
```{r}

n <- data.frame(
  name = c(
    "PC3 mock",
    "PC3 miR34a asRNA",
    "Skov3 mock",
    "Skov3 miR34a asRNA",
    "Saos2 mock",
    "Saos2 miR34a asRNA"
  ),
  n = c(3, 5, 4, 4, 5, 5)
)
n

```

<br></br>

_QPCR primers_

```{r}

primers <- data.frame(
    name=c(
        "ß-actin Fwd",
        "ß-actin Rev",
        "miR34a_asF1",
        "miR34a_asR1"
        
    ),
    sequence=c(
        "AGGTCATCACCATTGGCAATGAG",
        "CTTTCGGGATGTCCACGTCA",
        "AGCGGCATCTCCTCCACCTGAAA",
        "TTGCCTCGTGAGTCCAAGGAGAAT"
        
    )
)
primers

```

<br></br>

#### _Results_

```{r, message = FALSE}

url <- fileMap(type = "rds")["figure3a"][[1]]
data <- read_rds(gzcon(url(file.path(dataUrl, url))))

#remove Skov exp 1 due to variability in technical replicates
#data <- filter(data, `Biological Replicate` != 1 | `Cell line` != "Skov3")

#Have to divide the analysis due to the presence of several house 
#keeping genes.
gois1 <- c("miR34a asRNA", "Actin")
gois2 <- c("miR34a", "RNU48")
data1 <- filter(data, gene %in% gois1)
data2 <- filter(data, gene %in% gois2)

#calculate dct
groups <- c("Genetic mod", "Biological Replicate", "Cell line", "gene")

data1 <- data1 %>%
  technicalMeans(., groups) %>%
  dct(., gois1, Actin, groups, F) %>%
  mutate(GOI = as.character(GOI))

data2 <- data2 %>%
  technicalMeans(., groups) %>%
  dct(., gois2, RNU48, groups, F) %>%
  mutate(GOI = as.character(GOI))

data <- bind_rows(data1, data2) %>%
  mutate(GOI = parse_factor(GOI, levels = c("miR34a asRNA", "miR34a")))

#calculate ddct
groups[groups == "gene"] <- "GOI"
logical <- tibble(
  `Biological Replicate` = pull(data, `Biological Replicate`) == 2,
  `Genetic mod` = pull(data, `Genetic mod`) == "mock"
)

data <- data %>%
  ddct(., logical, groups) %>%
  folds(.)

#calculate stats
stats <- calcStats(data, `Genetic mod`, "mock", groups) %>%
  pFormat(.)

#format for plotting
lev <- c("mock", "miR34a asRNA", "Skov3\nmock", "Skov3\nmiR34a asRNA", "Saos2\nmock",   "Saos2\nmiR34a asRNA")

data <- data %>%
  mutate(`Genetic mod` = gsub("miR34a asRNA", "miR34a\nasRNA", `Genetic mod`)) %>%
  mutate(`Genetic mod` = parse_factor(`Genetic mod`, levels = c("mock", "miR34a\nasRNA")))

stats <- stats %>%
  mutate(`Genetic mod` = gsub("miR34a asRNA", "miR34a\nasRNA", `Genetic mod`)) %>%
  mutate(`Genetic mod` = parse_factor(`Genetic mod`, levels = c("mock", "miR34a\nasRNA")))

#vertical lines and labels in plot
ys <- stats %>%
  group_by(`Cell line`) %>%
  summarize(max = max(CI95h)) 

vl <- stats %>%
  group_by(`Cell line`, GOI) %>%
  summarize(
    pValue = as.numeric(min(pValue))
  ) %>%
  ungroup() %>%
  pFormat(.) %>%
  mutate(
    x = case_when(
      GOI ==    "miR34a" ~ 1 + 0.25,
      GOI == "miR34a asRNA" ~ 1 - 0.25
    ),
    xend = case_when(
      GOI ==    "miR34a" ~ 2 + 0.25,
      GOI == "miR34a asRNA" ~ 2 - 0.25
    )
  ) %>%
  inner_join(ys) %>%
  mutate(
    max = max + c(0.55, 1.95, 0.25, 1.65, 0.25, 1.65)
  )
```

```{r, fig.align='center', fig.height=8, fig.width=12}

p <- ggplot(data = NULL) +
  geom_violin(
    data = data,
    aes(
      x = `Genetic mod`,
      y = log2fold,
      group = interaction(`Genetic mod`, GOI)
    ),
    trim = TRUE,
    alpha = 0.5,
    fill = "grey",
    colour = NA,
    scale = "width",
    show.legend = FALSE
  ) +
  geom_dotplot(
    data = data, 
    aes(
      x = `Genetic mod`,
      y = log2fold,
      fill = GOI,
      group = interaction(`Genetic mod`, GOI)
    ),
    binwidth = 1/10,
    colour = NA,
    #alpha = 0.5,
    position = position_dodge(width = 0.9), 
    binaxis = 'y', 
    stackdir = 'center',
    dotsize = 4,
    show.legend = TRUE
  ) +
  facet_grid(
    . ~ `Cell line`, 
    scales = "free", 
    space = "free"
  ) +
  scale_y_continuous("\u0394\u0394Ct")+
  labs(x = "Cell line")

markdown <- p +
  geom_linerange(
    data = stats,
    aes(
      x = `Genetic mod`,
      ymin = CI95l,
      ymax = CI95h,
      group = interaction(`Genetic mod`, GOI)
    ),
    position = position_dodge(width = 0.9),
    colour = "gray20",
    show.legend = FALSE
  ) +
  geom_point(
    data = stats,
    aes(
      x = `Genetic mod`,
      y = mean,
      group = interaction(`Genetic mod`, GOI)
    ),
    colour = "gray20",
    shape = 95,
    size = 6,
    position = position_dodge(width = 0.9),
    show.legend = FALSE
  ) +
  geom_label(
    data = vl,
    aes(
      x = 1.5, 
      y = max + 0.4, 
      label = pFormat, 
      group = interaction(`Cell line`, GOI)
    ),
    fill = "white",
    label.size = 0,
    label.padding = unit(0.01, "lines"),
    position = position_dodge(width = 0.9),
    show.legend = FALSE,
    fontface = "bold",
    family = "Arial Unicode MS",
    size = 3
  ) +
  geom_segment(
    data = vl,
    aes(
      x = x,
      y = max,
      xend = xend,
      yend = max
    ),
    colour = "grey43",
    show.legend = FALSE
  ) +
  guides(
    fill = guide_legend(
      title = "Gene", 
      override.aes = list(size = 6)
    )
  )

plotRmarkdown(markdown)

pdf <- p +
  geom_linerange(
    data = stats,
    aes(
      x = `Genetic mod`,
      ymin = CI95l,
      ymax = CI95h,
      group = interaction(`Genetic mod`, GOI)
    ),
    position = position_dodge(width = 0.9),
    colour = "gray20",
    size = 0.1,
    show.legend = FALSE
  ) +
  geom_point(
    data = stats,
    aes(
      x = `Genetic mod`,
      y = mean,
      group = interaction(`Genetic mod`, GOI)
    ),
    colour = "gray20",
    shape = 95,
    position = position_dodge(width = 0.9),
    show.legend = FALSE
  ) +
  geom_label(
    data = vl,
    aes(
      x = 1.5, 
      y = max + 0.6, 
      label = pFormat, 
      group = interaction(`Cell line`, GOI)
    ),
    fill = "white",
    label.size = 0,
    label.padding = unit(0.01, "lines"),
    position = position_dodge(width = 0.9),
    show.legend = FALSE,
    fontface = "bold",
    family = "Arial Unicode MS",
    size = 1.5
  ) +
  geom_segment(
    data = vl,
    aes(
      x = x,
      y = max,
      xend = xend,
      yend = max
    ),
    colour = "grey43",
    size = 0.1,
    show.legend = FALSE
  ) +
  guides(
    fill = guide_legend(
      title = "Gene", 
      override.aes = list(size = 4)
    )
  )

figure <- plotPDF(pdf)

# path <- file.path("~/GitHub/miR34a_asRNA_project/inst", fileMap(type = "pdf")["figure3a"][[1]])
# ggsave(
#    plot = figure,
#    filename = path,
#    device = cairo_pdf,
#    height = 60,
#    width = 83,
#    units = "mm"
# )
```

<div class="caption">
miR34a asRNA and miR34a levels in PC3, Skov3, and Saos2 cell lines stably over-expressing miR34a asRNA compared to mock over-expressing control. The points represent the values obtained from each independant experiment (n >= 3) whereas, the gray shadow represents the distribution of those points. 95% confidence interval (vertical black lines), mean (horizontal black lines), and p-values are shown with the horizontal line under the p-value indicating the comparison that was tested.
</div>

<br></br>

#### _Conclusion_

miR34a asRNA over-expression causes a concominant up-regulation of miR34a in all 3 cell lines. This indicates that miR34a asRNA concordantly regulates miR34a.

<br></br><br></br>
